--- /home/glen/BUILD/kdepim-3.5.10/konsolekalendar/main.cpp~	2008-02-13 11:39:54.000000000 +0200
+++ /home/glen/BUILD/kdepim-3.5.10/konsolekalendar/main.cpp	2010-07-08 23:23:01.418447242 +0300
@@ -799,13 +799,13 @@
   if ( !args->isSet( "time" )     && !args->isSet( "epoch-start" ) &&
        !args->isSet( "end-time" ) && !args->isSet( "epoch-end" ) ) {
     // set default start date/time
-    startdatetime = QDateTime::QDateTime( startdate, starttime );
+    startdatetime = QDateTime( startdate, starttime );
     kdDebug() << "main | datetimestamp | "
               << "setting startdatetime from "
               << "default startdate (today) and starttime"
               << endl;
     // set default end date/time
-    enddatetime = QDateTime::QDateTime( enddate, endtime );
+    enddatetime = QDateTime( enddate, endtime );
     kdDebug() << "main | datetimestamp | "
               << "setting enddatetime from "
               << "default enddate (today) and endtime"
@@ -814,13 +814,13 @@
 
   // Set startdatetime, enddatetime if still necessary
   if ( startdatetime.isNull() ) {
-    startdatetime = QDateTime::QDateTime( startdate, starttime );
+    startdatetime = QDateTime( startdate, starttime );
     kdDebug() << "main | datetimestamp | "
               << "setting startdatetime from startdate and starttime"
               << endl;
   }
   if ( enddatetime.isNull() ) {
-    enddatetime = QDateTime::QDateTime( enddate, endtime );
+    enddatetime = QDateTime( enddate, endtime );
     kdDebug() << "main | datetimestamp | "
               << "setting enddatetime from enddate and endtime"
               << endl;
