--- kdepim-trinity-3.5.13.2/kmail/kmcomposewin.cpp~	2012-09-26 18:45:42.000000000 +0300
+++ kdepim-trinity-3.5.13.2/kmail/kmcomposewin.cpp	2014-06-06 23:20:27.869971174 +0300
@@ -1362,7 +1362,7 @@
   (void) new KAction (i18n("Cl&ean Spaces"), 0, TQT_TQOBJECT(this), TQT_SLOT(slotCleanSpace()),
                       actionCollection(), "clean_spaces");
 
-  mFixedFontAction = new KToggleAction( i18n("Use Fi&xed Font"), 0, TQT_TQOBJECT(this),
+  mFixedFontAction = new KToggleAction( i18n("Use Fi&xed Font"), "text", 0, TQT_TQOBJECT(this),
                       TQT_SLOT(slotUpdateFont()), actionCollection(), "toggle_fixedfont" );
   mFixedFontAction->setChecked( GlobalSettings::self()->useFixedFont() );
 
--- kdepim-trinity-3.5.13.2/kmail/kmmainwidget.cpp~	2014-06-06 23:22:04.000000000 +0300
+++ kdepim-trinity-3.5.13.2/kmail/kmmainwidget.cpp	2014-06-06 23:22:06.301032422 +0300
@@ -2773,7 +2773,7 @@
   mFindInMessageAction = new KAction( i18n("&Find in Message..."), "find", KStdAccel::shortcut(KStdAccel::Find), TQT_TQOBJECT(this),
 		      TQT_SLOT(slotFind()), actionCollection(), "find_in_messages" );
 
-  (void) new KAction( i18n("Select &All Messages"), KStdAccel::selectAll(), TQT_TQOBJECT(this),
+  (void) new KAction( i18n("Select &All Messages"), "ok", KStdAccel::selectAll(), TQT_TQOBJECT(this),
 		      TQT_SLOT(slotMarkAll()), actionCollection(), "mark_all_messages" );
 
   //----- Folder Menu
@@ -2817,13 +2817,13 @@
                                       TQT_SLOT( slotArchiveFolder() ), actionCollection(),
                                       "archive_folder" );
 
-  mPreferHtmlAction = new KToggleAction( i18n("Prefer &HTML to Plain Text"), 0, TQT_TQOBJECT(this),
+  mPreferHtmlAction = new KToggleAction( i18n("Prefer &HTML to Plain Text"), "html", 0, TQT_TQOBJECT(this),
 		      TQT_SLOT(slotOverrideHtml()), actionCollection(), "prefer_html" );
 
   mPreferHtmlLoadExtAction = new KToggleAction( i18n("Load E&xternal References"), 0, TQT_TQOBJECT(this),
 		      TQT_SLOT(slotOverrideHtmlLoadExt()), actionCollection(), "prefer_html_external_refs" );
 
-  mThreadMessagesAction = new KToggleAction( i18n("&Thread Messages"), 0, TQT_TQOBJECT(this),
+  mThreadMessagesAction = new KToggleAction( i18n("&Thread Messages"), "view_tree", 0, TQT_TQOBJECT(this),
 		      TQT_SLOT(slotOverrideThread()), actionCollection(), "thread_messages" );
 
   mThreadBySubjectAction = new KToggleAction( i18n("Thread Messages also by &Subject"), 0, TQT_TQOBJECT(this),
@@ -3053,7 +3053,7 @@
 		     TQT_SLOT(slotCollapseAllThreads()),
 		     actionCollection(), "collapse_all_threads" );
 
-  mViewSourceAction = new KAction( i18n("&View Source"), Key_V, TQT_TQOBJECT(this),
+  mViewSourceAction = new KAction( i18n("&View Source"), "source", Key_V, TQT_TQOBJECT(this),
                                    TQT_SLOT(slotShowMsgSrc()), actionCollection(),
                                    "view_source" );
 
--- kdepim-trinity-3.5.13.2/kmail/kmreaderwin.cpp~	2012-09-26 18:45:42.000000000 +0300
+++ kdepim-trinity-3.5.13.2/kmail/kmreaderwin.cpp	2014-06-06 23:23:02.023519306 +0300
@@ -611,7 +611,7 @@
   mUrlSaveAsAction = new KAction( i18n("Save Link As..."), 0, TQT_TQOBJECT(this),
                                   TQT_SLOT(slotUrlSave()), ac, "saveas_url" );
 
-  mToggleFixFontAction = new KToggleAction( i18n("Use Fi&xed Font"),
+  mToggleFixFontAction = new KToggleAction( i18n("Use Fi&xed Font"), "text",
                                             Key_X, TQT_TQOBJECT(this), TQT_SLOT(slotToggleFixedFont()),
                                             ac, "toggle_fixedfont" );
 
