Index: akregator/src/librss/loader.h
===================================================================
--- akregator/src/librss/loader.h	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ akregator/src/librss/loader.h	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -227,7 +227,7 @@
 	 * 'OutputRetriever' will make it execute the script
 	 * '/home/myself/some-script.py' and assume whatever that script prints to
 	 * stdout is RSS markup. This is e.g. handy for conversion scripts, which
-	 * download a HTML file and convert it's contents into RSS markup.
+	 * download an HTML file and convert it's contents into RSS markup.
 	 *
 	 * No matter what kind of retrieval algorithm you employ, your
 	 * 'slotLoadingComplete' method might look like this:
Index: akregator/src/eventsrc
===================================================================
--- akregator/src/eventsrc	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ akregator/src/eventsrc	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -15,7 +15,7 @@
 Name[es]=Origen añadido
 Name[et]=Kanal lisatud
 Name[eu]=Iturria gehituta
-Name[fa]=خوراندن اضافه شده‌
+Name[fa]=خوراندن اضافه‌شده
 Name[fi]=Syöte lisätty
 Name[fr]=Flux ajouté
 Name[ga]=Cuireadh fotha leis
Index: kmobile/devices/gammu/libkmobile_gammu.desktop
===================================================================
--- kmobile/devices/gammu/libkmobile_gammu.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmobile/devices/gammu/libkmobile_gammu.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -14,7 +14,7 @@
 Name[es]=Teléfono móvil u organizador (gammu)
 Name[et]=Mobiiltelefon või Organizer (gammu)
 Name[eu]=Mugikorra edo antolatzailea (gammu)
-Name[fa]=تلفن همراه یا سازمان دهنده (gammu)
+Name[fa]=تلفن همراه یا سازمان‌دهنده (gammu)
 Name[fi]=Matkapuhelin tai organisaattori (gammu)
 Name[fr]=Téléphone portable ou organiseur (gammu)
 Name[ga]=Fón Póca nó Eagraí (gammu)
Index: certmanager/lib/libkleopatrarc.desktop
===================================================================
--- certmanager/lib/libkleopatrarc.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ certmanager/lib/libkleopatrarc.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -99,7 +99,7 @@
 Name[es]=Clave revocada
 Name[et]=Tühistatud võti
 Name[eu]=Errebokatutako gakoa
-Name[fa]=کلید لغو شده‌
+Name[fa]=کلید لغو‌شده‌
 Name[fi]=Peruutettu avain
 Name[fr]=Clé révoquée
 Name[he]=מפתח לא קביל
Index: kmail/application_octetstream.desktop
===================================================================
--- kmail/application_octetstream.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmail/application_octetstream.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -39,7 +39,7 @@
 Comment[es]=Un accesorio de formato para el cuerpo de application/octet-stream
 Comment[et]=Põhiosa vormindamisplugin (MIME tüübile application/octet-stream)
 Comment[eu]=Gorputz-zati formateatzaile plugin bat aplikazio/zortikote-jarioentzat
-Comment[fa]=یک وصلۀ قالب دهندۀ بخش بدنه برای کاربرد/octet-stream
+Comment[fa]=یک وصلۀ قالب‌دهندۀ بخش بدنه برای کاربرد/octet-stream
 Comment[fi]=Muokkainliitännäinen application/octet-stream-muodolle
 Comment[fr]=Un module formateur de corps pour application (flux d'octets)
 Comment[hu]=Formázómodul application/octet-stream típusú adatfolyamokhoz
Index: kmail/kmail_config_composer.desktop
===================================================================
--- kmail/kmail_config_composer.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmail/kmail_config_composer.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -70,7 +70,7 @@
 Comment[es]=Frases y comportamiento general
 Comment[et]=Väljendid ja üldine käitumine
 Comment[eu]=Esaldiak eta portamolde orokorra
-Comment[fa]=عبارات و رفتار کلی
+Comment[fa]=عبارات و رفتار عمومی
 Comment[fi]=Ilmaukset ja yleiset asetukset
 Comment[fr]=Phrases et comportement général
 Comment[gl]=Frases e comportamento xeral
Index: kmail/kmkernel.cpp
===================================================================
--- kmail/kmkernel.cpp	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmail/kmkernel.cpp	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -2247,9 +2247,15 @@
   if ( !Wallet::isEnabled() || walletOpenFailed )
     return 0;
 
+  // find an appropriate parent window for the wallet dialog
+  WId window = 0;
+  if ( qApp->activeWindow() )
+    window = qApp->activeWindow()->winId();
+  else if ( getKMMainWidget() )
+    window = getKMMainWidget()->topLevelWidget()->winId();
+
   delete mWallet;
-  mWallet = Wallet::openWallet( Wallet::NetworkWallet(),
-        getKMMainWidget() ? getKMMainWidget()->topLevelWidget()->winId() : 0 );
+  mWallet = Wallet::openWallet( Wallet::NetworkWallet(), window );
 
   if ( !mWallet ) {
     walletOpenFailed = true;
Index: kmail/kmcomposewin.h
===================================================================
--- kmail/kmcomposewin.h	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmail/kmcomposewin.h	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -398,8 +398,8 @@
   void slotAttachRemove();
   void slotAttachSave();
   void slotAttachProperties();
+  void slotAttachOpenWith();
 
-
   /**
    * Select an email from the addressbook and add it to the line
    * the pressed button belongs to.
@@ -546,9 +546,9 @@
   virtual bool queryExit ();
 
   /**
-   * Open the attachment with the given index.
+   * Open the attachment with the given index and with ("Open with")
    */
-  void openAttach( int index );
+  void openAttach( int index, bool with );
 
   /**
    * View the attachment with the given index.
@@ -688,7 +688,7 @@
   QPtrList<QListViewItem> mAtmItemList;
   QPtrList<KMMessagePart> mAtmList;
   QPopupMenu *mAttachMenu;
-  int mOpenId, mViewId, mRemoveId, mSaveAsId, mPropertiesId;
+  int mOpenId, mOpenWithId, mViewId, mRemoveId, mSaveAsId, mPropertiesId;
   bool mAutoDeleteMsg;
   bool mSigningAndEncryptionExplicitlyDisabled;
   bool mLastSignActionState, mLastEncryptActionState;
Index: kmail/kmcomposewin.cpp
===================================================================
--- kmail/kmcomposewin.cpp	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmail/kmcomposewin.cpp	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -333,7 +333,7 @@
 
   connect( mAtmListView,
            SIGNAL( doubleClicked( QListViewItem* ) ),
-           SLOT( slotAttachProperties() ) );
+           SLOT( slotAttachOpen() ) );
   connect( mAtmListView,
            SIGNAL( rightButtonPressed( QListViewItem*, const QPoint&, int ) ),
            SLOT( slotAttachPopupMenu( QListViewItem*, const QPoint&, int ) ) );
@@ -2898,6 +2898,8 @@
 
      mOpenId = mAttachMenu->insertItem(i18n("to open", "Open"), this,
                              SLOT(slotAttachOpen()));
+     mOpenWithId = mAttachMenu->insertItem(i18n("Open with..."), this,
+                             SLOT(slotAttachOpenWith()));
      mViewId = mAttachMenu->insertItem(i18n("to view", "View"), this,
                              SLOT(slotAttachView()));
      mRemoveId = mAttachMenu->insertItem(i18n("Remove"), this, SLOT(slotAttachRemove()));
@@ -3142,12 +3144,23 @@
   int i = 0;
   for ( QPtrListIterator<QListViewItem> it(mAtmItemList); *it; ++it, ++i ) {
     if ( (*it)->isSelected() ) {
-      openAttach( i );
+      openAttach( i, false );
     }
   }
 }
 
 //-----------------------------------------------------------------------------
+void KMComposeWin::slotAttachOpenWith()
+{
+  int i = 0;
+  for ( QPtrListIterator<QListViewItem> it(mAtmItemList); *it; ++it, ++i ) {
+    if ( (*it)->isSelected() ) {
+      openAttach( i, true );
+    }
+  }
+}
+
+//-----------------------------------------------------------------------------
 bool KMComposeWin::inlineSigningEncryptionSelected() {
   if ( !mSignAction->isChecked() && !mEncryptAction->isChecked() )
     return false;
@@ -3175,7 +3188,7 @@
 }
 
 //-----------------------------------------------------------------------------
-void KMComposeWin::openAttach( int index )
+void KMComposeWin::openAttach( int index, bool with )
 {
   KMMessagePart* msgPart = mAtmList.at(index);
   const QString contentTypeStr =
@@ -3202,7 +3215,7 @@
   KService::Ptr offer =
     KServiceTypeProfile::preferredService( mimetype->name(), "Application" );
 
-  if ( !offer || mimetype->name() == "application/octet-stream" ) {
+  if ( with || !offer || mimetype->name() == "application/octet-stream" ) {
     if ( ( !KRun::displayOpenWithDialog( url, autoDelete ) ) && autoDelete ) {
       QFile::remove(url.path());
     }
Index: kmail/Mainpage.dox
===================================================================
--- kmail/Mainpage.dox	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kmail/Mainpage.dox	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -793,7 +793,7 @@
 
   - @em BodyPartFormatter
 	the formatter interface. Contains a single method format()
-	that takes a BodyPart to process and a HTMLWriter to write the
+	that takes a BodyPart to process and an HTMLWriter to write the
 	generated HTML to and returns a result code with which it can
 	request more information or delegate the formatting back to
 	some default processor. BodyPartFormatters are meant to be
Index: knode/articlewidget.h
===================================================================
--- knode/articlewidget.h	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ knode/articlewidget.h	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -145,7 +145,7 @@
       FancyFormatting = 2,
       AllowROT13 = 4
     };
-    /// convert the given string into a HTML string
+    /// convert the given string into an HTML string
     QString toHtmlString( const QString &line, int flags = ParseURL );
     /// convert the given image into a data:/ URL
     static QString imgToDataUrl( const QImage &image, const char* fmt );
Index: kaddressbook/features/distributionlistwidget.cpp
===================================================================
--- kaddressbook/features/distributionlistwidget.cpp	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kaddressbook/features/distributionlistwidget.cpp	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -327,6 +327,7 @@
 #else
   KABC::DistributionList *list = mManager->list( oldName );
   list->setName( newName );
+  mManager->save();
   updateNameCombo();
 #endif
 
Index: libkdepim/categoryeditdialog.cpp
===================================================================
--- libkdepim/categoryeditdialog.cpp	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ libkdepim/categoryeditdialog.cpp	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -76,6 +76,7 @@
   }
   mWidget->mButtonRemove->setEnabled( categoriesExist );
   mWidget->mCategories->setSelected( mWidget->mCategories->firstChild(), true );
+  mWidget->mEdit->setText( mWidget->mCategories->currentItem()->text( 0 ) );
 }
 
 void CategoryEditDialog::slotTextChanged(const QString &text)
Index: libkcal/libical/src/libical/sspm.c
===================================================================
--- libkcal/libical/src/libical/sspm.c	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ libkcal/libical/src/libical/sspm.c	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -1171,7 +1171,7 @@
 			     char *src,
 			     size_t *size)
 {
-    int cc;
+    int cc = 0;
     char buf[4] = {0,0,0,0};  
     int p = 0;
     int valid_data = 0;
Index: libkcal/htmlexport.h
===================================================================
--- libkcal/htmlexport.h	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ libkcal/htmlexport.h	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -37,7 +37,7 @@
 namespace KCal {
 
 /**
-  This class provides the functions to export a calendar as a HTML page.
+  This class provides the functions to export a calendar as an HTML page.
 */
 class KDE_EXPORT HtmlExport
 {
Index: kpilot/conduits/sysinfoconduit/sysinfo-setup_dialog.ui
===================================================================
--- kpilot/conduits/sysinfoconduit/sysinfo-setup_dialog.ui	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kpilot/conduits/sysinfoconduit/sysinfo-setup_dialog.ui	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -99,7 +99,7 @@
                                     <bool>true</bool>
                                 </property>
                                 <property name="whatsThis" stdset="0">
-                                    <string>&lt;qt&gt;Select this option to output the system information data as a HTML document.&lt;/qt&gt;</string>
+                                    <string>&lt;qt&gt;Select this option to output the system information data as an HTML document.&lt;/qt&gt;</string>
                                 </property>
                             </widget>
                             <widget class="QRadioButton" row="1" column="0" rowspan="1" colspan="2">
Index: kpilot/conduits/memofileconduit/memofile-conduit.desktop
===================================================================
--- kpilot/conduits/memofileconduit/memofile-conduit.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kpilot/conduits/memofileconduit/memofile-conduit.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -47,7 +47,7 @@
 Comment[es]=Este conducto sincroniza las notas de su agenda electrónica con el directorio local.
 Comment[et]=See kanal sünkroniseerib pihuarvutis ja arvutis olevad memod.
 Comment[eu]=Kanal honek zure agendako oharrak direktorio lokal batekin sinkronizatzen ditu.
-Comment[fa]=این لوله، memo‌های دستی خود را با یک فهرست راهنمای محلی همگام‌سازی کنید.
+Comment[fa]=این لوله، memo‌های دستی خود را با یک فهرست راهنمای محلی همگام‌سازی می‌کند.
 Comment[fi]=Tämä yhdyskäytävä synkronoi taskutietokoneen muistiot paikalliseen kansioon.
 Comment[fr]=Ce canal synchronise les mémos du Palm avec ceux de KDE.
 Comment[hu]=Ezzel a csatolóval egy kézi számítógép memóit lehet szinkronizálni a helyi címjegyzékkel.
Index: kontact/plugins/akregator/akregatorplugin.desktop
===================================================================
--- kontact/plugins/akregator/akregatorplugin.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kontact/plugins/akregator/akregatorplugin.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -65,7 +65,7 @@
 Name[es]=Orígenes
 Name[et]=Kanalid
 Name[eu]=Iturriak
-Name[fa]=خوراندن
+Name[fa]=خوراندنها
 Name[fi]=Syötteet
 Name[fr]=Flux
 Name[ga]=Fothaí
Index: kontact/plugins/akregator/akregatorplugin3.2.desktop
===================================================================
--- kontact/plugins/akregator/akregatorplugin3.2.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kontact/plugins/akregator/akregatorplugin3.2.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -64,7 +64,7 @@
 Name[es]=Orígenes
 Name[et]=Kanalid
 Name[eu]=Iturriak
-Name[fa]=خوراندن
+Name[fa]=خوراندنها
 Name[fi]=Syötteet
 Name[fr]=Flux
 Name[ga]=Fothaí
Index: kontact/interfaces/kontactplugin.desktop
===================================================================
--- kontact/interfaces/kontactplugin.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ kontact/interfaces/kontactplugin.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -15,7 +15,7 @@
 Name[es]=Plugin Kontact
 Name[et]=Kontacti plugin
 Name[eu]=Kontact plugin-a
-Name[fa]=وصلۀ 
+Name[fa]=وصلۀ Kontact
 Name[fi]=Kontact-liitännäinen
 Name[fr]=Module Kontact
 Name[ga]=Breiseán Kontact
Index: ktnef/gui/ktnef.desktop
===================================================================
--- ktnef/gui/ktnef.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ ktnef/gui/ktnef.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -57,7 +57,7 @@
 Comment[es]=A visor/extractor para archivos TNEF
 Comment[et]=TNEF-failide näitaja/ekstraktija
 Comment[eu]=TNEF fitxategien ikustaile/erauzlea
-Comment[fa]=یک مشاهده‌گر/استخراجگر برای پرونده‌های TNEF
+Comment[fa]=یک مشاهده‌گر/استخراج‌گر برای پرونده‌های TNEF
 Comment[fi]=Näytin/purkaja TNEF-tiedostoille
 Comment[fr]=Pour afficher / extraire des fichiers TNEF
 Comment[hi]=टीएनईएफ फ़ाइलों के लिए एक प्रदर्शक/एक्सट्रेक्टर
Index: korganizer/korganizer.kcfg
===================================================================
--- korganizer/korganizer.kcfg	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ korganizer/korganizer.kcfg	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -74,7 +74,7 @@
 
     <entry type="Bool" key="Html With Save">
       <label>Export to HTML with every save</label>
-      <whatsthis>Check this box to export the calendar to a HTML-file every time you save it. By default, this file will be called calendar.html and placed in the user home folder.</whatsthis>
+      <whatsthis>Check this box to export the calendar to an HTML-file every time you save it. By default, this file will be called calendar.html and placed in the user home folder.</whatsthis>
       <default>false</default>
     </entry>
     <entry type="Enum" key="Destination" name="Destination">

Property changes on: korganizer/pixmaps
___________________________________________________________________
Name: svn:ignore
   + Makefile
Makefile.in


Index: korganizer/korganizer_configgroupautomation.desktop
===================================================================
--- korganizer/korganizer_configgroupautomation.desktop	(.../tags/KDE/3.5.5/kdepim)	(revision 594434)
+++ korganizer/korganizer_configgroupautomation.desktop	(.../branches/KDE/3.5/kdepim)	(revision 594434)
@@ -25,7 +25,7 @@
 Name[es]=Automatización del grupo
 Name[et]=Grupitöö
 Name[eu]=Talde automatizazioa
-Name[fa]=خودکار سازی گروه
+Name[fa]=خودکارسازی گروه
 Name[fi]=Ryhmän automatisointi
 Name[fr]=Automatisation des groupes
 Name[gl]=Automatización de Grupo

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


