--- kdepim-030725/kontact/plugins/korganizer/Makefile.am~	2003-07-19 00:53:14.000000000 +0200
+++ kdepim-030725/kontact/plugins/korganizer/Makefile.am	2003-07-25 20:38:25.000000000 +0200
@@ -4,17 +4,17 @@
 
 kde_module_LTLIBRARIES = libkontact_korganizerplugin.la libkontact_todoplugin.la
 
-libkontact_korganizerplugin_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
+libkontact_korganizerplugin_la_LDFLAGS = \
+  -L$(top_builddir)/libkcal $(all_libraries) $(KDE_PLUGIN) 
 libkontact_korganizerplugin_la_LIBADD = \
-  $(top_builddir)/kontact/interfaces/libkpinterfaces.la $(LIB_KPARTS) \
-  -L$(top_builddir)/libkcal -lkcal
+  $(top_builddir)/kontact/interfaces/libkpinterfaces.la $(LIB_KPARTS) -lkcal
 libkontact_korganizerplugin_la_SOURCES = korganizerplugin.cpp kcalendariface.stub \
   summarywidget.cpp
 
-libkontact_todoplugin_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
+libkontact_todoplugin_la_LDFLAGS = \
+  -L$(top_builddir)/libkcal $(all_libraries) $(KDE_PLUGIN)  
 libkontact_todoplugin_la_LIBADD = \
-  $(top_builddir)/kontact/interfaces/libkpinterfaces.la $(LIB_KPARTS) \
-  -L$(top_builddir)/libkcal -lkcal
+  $(top_builddir)/kontact/interfaces/libkpinterfaces.la $(LIB_KPARTS) -lkcal
 libkontact_todoplugin_la_SOURCES = todoplugin.cpp kcalendariface.stub
 
 METASOURCES = AUTO
