--- kdepim-3.5.9/kmail/kmmainwin.rc~	2008-02-14 23:26:54.000000000 +0200
+++ kdepim-3.5.9/kmail/kmmainwin.rc	2008-02-14 23:33:11.645281237 +0200
@@ -175,7 +175,7 @@
  </MenuBar>
  <ToolBar noMerge="1" name="mainToolBar" fullWidth="true" >
   <text>Main Toolbar</text>
-  <Action name="new_message" />
+  <Action name="post_message" />
   <Action name="file_save_as" />
   <Action name="file_print" />
   <Separator/>
@@ -201,5 +201,7 @@
   <Action name="search_messages" />
   <ActionList name="toolbar_filter_actions" />
   <Action name="create_todo"/>
+  <Action name="status_flag" />
+  <Action name="mark_all_as_read" />
  </ToolBar>
 </kpartgui>
--- kdepim-3.5.9/kmail/kmail_part.rc~	2008-02-14 23:36:06.000000000 +0200
+++ kdepim-3.5.9/kmail/kmail_part.rc	2008-02-14 23:36:32.748879880 +0200
@@ -169,7 +169,7 @@
  </MenuBar>
  <ToolBar noMerge="1" name="mainToolBar" fullWidth="true" >
   <text>Main Toolbar</text>
-  <Action name="new_message" />
+  <Action name="post_message" />
   <Action name="file_save_as" />
   <Action name="file_print" />
   <Separator/>
@@ -195,5 +195,7 @@
   <Action name="search_messages" />
   <ActionList name="toolbar_filter_actions" />
   <Action name="create_todo"/>
+  <Action name="status_flag" />
+  <Action name="mark_all_as_read" />
  </ToolBar>
 </kpartgui>
--- kmail-3.5.9/kmcomposerui.rc/kmcomposerui.rc~	2008-02-13 11:39:37.000000000 +0200
+++ kmail-3.5.9/kmcomposerui.rc/kmcomposerui.rc	2008-02-19 14:23:51.807830875 +0200
@@ -78,4 +78,5 @@
   <Action name="sign_message" />
   <Action name="encrypt_message" />
+  <Action name="wordwrap" />
   <Action name="options_select_crypto" />
  </ToolBar>
@@ -112,4 +113,7 @@
   <Action name="align_center" />
   <Action name="align_right" />
  </ToolBar>
+ <ActionProperties>
+  <Action icon="view_text" name="wordwrap" />
+ </ActionProperties>
 </kpartgui>
