Index: kresources/exchange/exchange.desktop
===================================================================
--- kresources/exchange/exchange.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/exchange/exchange.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -17,6 +17,7 @@
 Name[et]=Exchange 2000 server
 Name[eu]=Exchange 2000 zerbitzaria
 Name[fr]=Serveur Exchange2000
+Name[ga]=Freastalaí Exchange 2000
 Name[hi]=एक्सचेंज2000 सर्वर
 Name[hu]=Exchange 2000-kiszolgáló
 Name[is]=Exchange 2000 þjónn
Index: kresources/slox/kcal_slox.desktop
===================================================================
--- kresources/slox/kcal_slox.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/slox/kcal_slox.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -11,6 +11,7 @@
 Name[et]=SUSE LINUX Openexchange server
 Name[eu]=SUSE LINUX Openexchange zerbitzaria
 Name[fr]=Serveur SUSE Linux Openexchange
+Name[ga]=Freastalaí Openexchange SUSE LINUX
 Name[hu]=SUSE LINUX Openexchange-kiszolgáló
 Name[is]=SUSE LINUX Openexchange þjónn
 Name[it]=Server SUSE LINUX Openexchange
Index: kresources/slox/webdavhandler.cpp
===================================================================
--- kresources/slox/webdavhandler.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/slox/webdavhandler.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -28,6 +28,8 @@
 #endif
 #endif
 
+#include <stdlib.h>
+
 #include <libkcal/incidence.h>
 
 #include <libkdepim/kpimprefs.h>
@@ -150,7 +152,16 @@
 {
   QDateTime utc = KPimPrefs::localTimeToUtc( dt, timeZoneId );
 
-  uint ticks = -utc.secsTo( QDateTime( QDate( 1970, 1, 1 ), QTime( 0, 0 ) ) );
+  // secsTo and toTime_t etc also perform a timezone conversion using the system timezone,
+  // but we want to use the calendar timezone, so we have to convert ourself and spoof the tz to UTC before 
+  // converting to ticks to prevent this 
+  QCString origTz = getenv("TZ");
+  setenv( "TZ", "UTC", 1 );
+  uint ticks = utc.toTime_t();
+  if ( origTz.isNull() )
+    unsetenv( "TZ" );
+  else
+    setenv( "TZ", origTz, 1 );
 
   return QString::number( ticks ) + "000";
 }
Index: kresources/slox/kabc_slox.desktop
===================================================================
--- kresources/slox/kabc_slox.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/slox/kabc_slox.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -11,6 +11,7 @@
 Name[et]=SUSE LINUX Openexchange server
 Name[eu]=SUSE LINUX Openexchange zerbitzaria
 Name[fr]=Serveur SUSE Linux Openexchange
+Name[ga]=Freastalaí Openexchange SUSE LINUX
 Name[hu]=SUSE LINUX Openexchange-kiszolgáló
 Name[is]=SUSE LINUX Openexchange þjónn
 Name[it]=Server SUSE LINUX Openexchange
Index: kresources/groupdav/kabc_groupdav.desktop
===================================================================
--- kresources/groupdav/kabc_groupdav.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/groupdav/kabc_groupdav.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -12,6 +12,7 @@
 Name[et]=GroupDAV server (nt. OpenGroupware)
 Name[eu]=GroupDAV zerbitzaria (adib. OpenGroupware)
 Name[fr]=GroupDAV Serveur (ex. OpenGroupware)
+Name[ga]=Freastalaí GroupDAV (m.sh. OpenGroupware)
 Name[hu]=GroupDAV-kiszolgáló (pl. OpenGroupware)
 Name[is]=GroupDAV þjónn (t.d. OpenGroupware)
 Name[it]=Server GroupDAV (per es. OpenGroupware)
Index: kresources/groupdav/kcal_groupdav.desktop
===================================================================
--- kresources/groupdav/kcal_groupdav.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/groupdav/kcal_groupdav.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -12,6 +12,7 @@
 Name[et]=GroupDAV server (nt. OpenGroupware)
 Name[eu]=GroupDAV zerbitzaria (adib. OpenGroupware)
 Name[fr]=GroupDAV Serveur (ex. OpenGroupware)
+Name[ga]=Freastalaí GroupDAV (m.sh. OpenGroupware)
 Name[hu]=GroupDAV-kiszolgáló (pl. OpenGroupware)
 Name[is]=GroupDAV þjónn (t.d. OpenGroupware)
 Name[it]=Server GroupDAV (per es. OpenGroupware)
Index: kresources/kolab/kcal/resourcekolab.cpp
===================================================================
--- kresources/kolab/kcal/resourcekolab.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/kolab/kcal/resourcekolab.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -487,10 +487,18 @@
       mUidMap[ uid ] = StorageReference( subResource, sernum );
     } else {
       /* This is a real add, from KMail, we didn't trigger this ourselves.
-         If this uid already exists in this folder, do conflict resolution */
+       * If this uid already exists in this folder, do conflict resolution, 
+       * unless the folder is read-only, in which case the user should not be
+       * offered a means of putting mails in a folder she'll later be unable to
+       * upload. Skip the incidence, in this case. */
       if ( mUidMap.contains( uid )
           && ( mUidMap[ uid ].resource() == subResource ) ) {
-        resolveConflict( incidence, subResource, sernum );
+        if ( (*map)[ subResource ].writable() ) {
+          resolveConflict( incidence, subResource, sernum );
+        } else {
+          kdWarning( 5650 ) << "Duplicate event in a read-only folder detected! "
+            "Please inform the owner of the folder. " << endl;
+        }
         return true;
       }
       /* Add to the cache if the add didn't come from KOrganizer, in which case
Index: kresources/groupware/kabc_groupware.desktop
===================================================================
--- kresources/groupware/kabc_groupware.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/groupware/kabc_groupware.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -10,6 +10,7 @@
 Name[et]=Grupitöö server
 Name[eu]=Groupware zerbitzaria
 Name[fr]=Serveur de travail collaboratif
+Name[ga]=Freastalaí Groupware
 Name[hi]=ग्रुपवेयर सर्वर
 Name[hu]=Groupware kiszolgáló
 Name[is]=Groupware þjónn
Index: kresources/groupware/kcal_groupware.desktop
===================================================================
--- kresources/groupware/kcal_groupware.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/groupware/kcal_groupware.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -10,6 +10,7 @@
 Name[et]=Grupitöö server
 Name[eu]=Groupware zerbitzaria
 Name[fr]=Serveur de travail collaboratif
+Name[ga]=Freastalaí Groupware
 Name[hi]=ग्रुपवेयर सर्वर
 Name[hu]=Groupware kiszolgáló
 Name[is]=Groupware þjónn
Index: kresources/remote/remote.desktop
===================================================================
--- kresources/remote/remote.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kresources/remote/remote.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -15,6 +15,7 @@
 Name[eu]=Egutegia urruneko fitxategi batean
 Name[fi]=Kalenteri etätiedostossa
 Name[fr]=Calendrier dans un fichier distant
+Name[ga]=Féilire i gCianchomhad
 Name[hi]=रिमोट फ़ाइल में कैलेन्डर
 Name[hu]=Távoli fájlban tárolt naptár
 Name[is]=Dagatal í fjarlægri skrá
Index: kmail/kmailicalifaceimpl.cpp
===================================================================
--- kmail/kmailicalifaceimpl.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmail/kmailicalifaceimpl.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -848,7 +848,7 @@
 void KMailICalIfaceImpl::slotIncidenceAdded( KMFolder* folder,
                                              Q_UINT32 sernum )
 {
-  if( !mUseResourceIMAP )
+  if( mResourceQuiet || !mUseResourceIMAP )
     return;
 
 //  kdDebug(5006) << "KMailICalIfaceImpl::slotIncidenceAdded" << endl;
Index: kmail/partNode.cpp
===================================================================
--- kmail/partNode.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmail/partNode.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -36,6 +36,7 @@
 #include "kmmimeparttree.h"
 #include <mimelib/utility.h>
 #include <qregexp.h>
+#include <kasciistricmp.h>
 
 /*
   ===========================================================================
@@ -414,7 +415,7 @@
 partNode* partNode::findNodeForDwPart( DwBodyPart* part )
 {
     partNode* found = 0;
-    if( dwPart()->partId() == part->partId() )
+    if( kasciistricmp( dwPart()->partId(), part->partId() ) == 0 )
         return this;
     if( mChild )
         found = mChild->findNodeForDwPart( part );
Index: kmail/keyresolver.cpp
===================================================================
--- kmail/keyresolver.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmail/keyresolver.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -1171,6 +1171,17 @@
     for ( uint i = 0; i < items.size(); ++i ) {
       ContactPreferences& pref = lookupContactPreferences( items[i].address );
       pref.encryptionPreference = items[i].pref;
+      pref.pgpKeyFingerprints.clear();
+      pref.smimeCertFingerprints.clear();
+      for ( std::vector<GpgME::Key>::const_iterator it = items[i].keys.begin(), end = items[i].keys.end() ; it != end ; ++it ) {
+        if ( it->protocol() == GpgME::Context::OpenPGP ) {
+          if ( const char * fpr = it->primaryFingerprint() )
+            pref.pgpKeyFingerprints.push_back( fpr );
+        } else if ( it->protocol() == GpgME::Context::CMS ) {
+          if ( const char * fpr = it->primaryFingerprint() )
+            pref.smimeCertFingerprints.push_back( fpr );
+        }
+      }
       saveContactPreference( items[i].address, pref );
     }
   }
Index: kmail/kmsender.cpp
===================================================================
--- kmail/kmsender.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmail/kmsender.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -519,8 +519,10 @@
 	       .arg(mCurrentMsg->subject()));
   if (!mSendProc->send(mCurrentMsg))
   {
-    mCurrentMsg->setTransferInProgress( false );
-    mOutboxFolder->unGetMsg( mFailedMessages );
+    if ( mCurrentMsg )
+      mCurrentMsg->setTransferInProgress( false );
+    if ( mOutboxFolder )
+      mOutboxFolder->unGetMsg( mFailedMessages );
     mCurrentMsg = 0;
     cleanup();
     setStatusMsg(i18n("Failed to send (some) queued messages."));
@@ -596,7 +598,8 @@
     // sending of message aborted
     if ( mCurrentMsg ) {
       mCurrentMsg->setTransferInProgress( false );
-      mOutboxFolder->unGetMsg( mFailedMessages );
+      if ( mOutboxFolder )
+        mOutboxFolder->unGetMsg( mFailedMessages );
       mCurrentMsg = 0;
     }
     msg = i18n("Sending aborted:\n%1\n"
@@ -610,8 +613,10 @@
     setStatusMsg( i18n( "Sending aborted." ) );
   } else {
     if (!mSendProc->sendOk()) {
-      mCurrentMsg->setTransferInProgress( false );
-      mOutboxFolder->unGetMsg( mFailedMessages );
+      if ( mCurrentMsg )
+        mCurrentMsg->setTransferInProgress( false );
+      if ( mOutboxFolder )
+        mOutboxFolder->unGetMsg( mFailedMessages );
       mCurrentMsg = 0;
       mFailedMessages++;
       // Sending of message failed.
Index: kalarm/alarmcalendar.h
===================================================================
--- kalarm/alarmcalendar.h	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kalarm/alarmcalendar.h	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -46,7 +46,7 @@
 		void                  startUpdate();
 		void                  endUpdate();
 		KCal::Event*          event(const QString& uniqueID)      { return mCalendar ? mCalendar->event(uniqueID) : 0; }
-		KCal::Event::List     events()                            { return mCalendar->events(); }
+		KCal::Event::List     events()                            { return mCalendar->rawEvents(); }
 		KCal::Event::List     events(const QDate& d, bool sorted = false) { return mCalendar->events(d, sorted); }
 		KCal::Event::List     eventsWithAlarms(const QDateTime& from, const QDateTime& to);
 		KCal::Event*          addEvent(const KAEvent&, bool useEventID = false);
Index: Makefile.cvs
===================================================================
--- Makefile.cvs	(.../tags/KDE/3.4.2/kdepim)	(revision 0)
+++ Makefile.cvs	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -0,0 +1,16 @@
+
+all: 
+	@echo "This Makefile is only for the CVS repository"
+	@echo "This will be deleted before making the distribution"
+	@echo ""
+	@if test ! -d admin; then \
+	   echo "Please recheckout this module!" ;\
+	   echo "for cvs: use checkout once and after that update again" ;\
+	   echo "for cvsup: checkout kde-common from cvsup and" ;\
+	   echo "   link kde-common/admin to ./admin" ;\
+	   exit 1 ;\
+	fi
+	$(MAKE) -f admin/Makefile.common cvs
+
+.SILENT:
+

Property changes on: Makefile.cvs
___________________________________________________________________
Name: svn:keywords
   + Author Date Id Revision

Index: libemailfunctions/email.cpp
===================================================================
--- libemailfunctions/email.cpp	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ libemailfunctions/email.cpp	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -697,14 +697,24 @@
     //          and outside of quoted text behind the trailing '>'.
     bInComment = false;
     bInQuotesOutsideOfEmail = false;
+    int parenthesesNesting = 0;
     for( i = iAd+1; len > i; ++i ) {
       c = aStr[i];
       if( bInComment ){
         if( ')' == c ){
-          if( !name.isEmpty() )
-            name.append( ' ' );
-          bInComment = false;
-        }else{
+          if ( --parenthesesNesting == 0 ) {
+            bInComment = false;
+            if( !name.isEmpty() )
+              name.append( ' ' );
+          } else {
+            // nested ")", add it
+            name.append( ')' ); // name can't be empty here
+          }
+        } else {
+          if( '(' == c ) {
+            // nested "("
+            ++parenthesesNesting;
+          }
           name.append( c ); // all comment stuff is part of the name
         }
       }else if( bInQuotesOutsideOfEmail ){
@@ -730,7 +740,8 @@
             case '(':
               if( !name.isEmpty() )
                 name.append( ' ' );
-              bInComment = true;
+              if ( ++parenthesesNesting > 0 )
+                bInComment = true;
               break;
             default:
               if( ' ' != c )
Index: kmailcvt/filters.hxx
===================================================================
--- kmailcvt/filters.hxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filters.hxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -85,6 +85,57 @@
 };
 
 
+
+/** 
+* Glorified QString[N] for (a) understandability (b) older gcc compatibility. 
+*/
+template <unsigned int size> class FolderStructureBase
+{
+public:
+	typedef QString NString[size];
+	/** Constructor. Need a default constructor for QValueList. */
+	FolderStructureBase() {} ;
+
+	/** Constructor. Turn N QStrings into a folder structure 
+	*   description. 
+	*/
+	FolderStructureBase(const NString &s)
+	{
+	    for(unsigned int i=0; i<size; i++) d[i]=s[i];
+	} ;
+
+	/** Copy Constructor. */
+	FolderStructureBase(const FolderStructureBase &s)
+	{
+	    for(unsigned int i=0; i<size; i++) d[i]=s[i];
+	} ;
+
+	/** Assignment operator. Does the same thing as 
+	*   the copy constructor.
+	*/
+	FolderStructureBase &operator =(const FolderStructureBase &s)
+	{
+	    for(unsigned int i=0; i<size; i++) d[i]=s[i];
+	    return *this;
+	} ;
+
+	/** Access the different fields. There doesn't seem to
+	*   be a real semantics for the fields.
+	*/
+	const QString operator [](unsigned int i) const
+	{
+	    if (i<size) return d[i]; else return QString::null;
+	} ;
+
+	/** Access the different fields, for writing. */
+	QString &operator [](unsigned int i)
+	{
+	    Q_ASSERT(i<size);
+	    if (i<size) return d[i]; else return d[0];
+	} ;
+private:
+	QString d[size];
+} ;
+
 #endif
 
-// vim: ts=2 sw=2 et
Index: kmailcvt/filter_thunderbird.cxx
===================================================================
--- kmailcvt/filter_thunderbird.cxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_thunderbird.cxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -198,7 +198,7 @@
       tmp.close();
       first_msg = false;
 
-      QString destFolder = "";
+      QString destFolder;
       QString _targetDir = targetDir;
       if(!targetDir.isNull()){
          if(_targetDir.contains(".sbd")) _targetDir.remove(".sbd");
Index: kmailcvt/filter_evolution_v2.cxx
===================================================================
--- kmailcvt/filter_evolution_v2.cxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_evolution_v2.cxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -201,7 +201,7 @@
       tmp.close();
       first_msg = false;
 
-      QString destFolder = "";
+      QString destFolder;
       QString _targetDir = targetDir;
       if(!targetDir.isNull()){
          if(_targetDir.contains(".sbd")) _targetDir.remove(".sbd");
Index: kmailcvt/filter_outlook.cxx
===================================================================
--- kmailcvt/filter_outlook.cxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_outlook.cxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -52,7 +52,7 @@
   info->addLog(i18n("Importing new mail files..."));
   info->addLog(i18n("No files found for import."));
   
-  QString outlookDir = "";
+  QString outlookDir;
   outlookDir = KFileDialog::getExistingDirectory(QDir::homeDirPath(), info->parent());
   QDir importDir (outlookDir);
   QStringList files = importDir.entryList("*.[pP][sS][tT]", QDir::Files, QDir::Name);
Index: kmailcvt/filter_pmail.cxx
===================================================================
--- kmailcvt/filter_pmail.cxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_pmail.cxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -320,13 +320,13 @@
 QString FilterPMail::getFolderName(QString ID) 
 {
     bool found = false;
-    QString folder = "";
+    QString folder;
     QString search = ID;
     
     while (!found)
     {
-        for ( QValueList<QString[5]>::Iterator it = folderMatrix.begin(); it != folderMatrix.end(); it++) {
-            QString tmp[5] = *it;
+        for ( FolderStructureIterator it = folderMatrix.begin(); it != folderMatrix.end(); it++) {
+            FolderStructure tmp = *it;
             
             QString _ID = tmp[2];
             if(_ID == search) {
Index: kmailcvt/filter_oe.cxx
===================================================================
--- kmailcvt/filter_oe.cxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_oe.cxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -138,7 +138,7 @@
                 folderName = "OE-" + mailfileinfo.baseName(TRUE);
                 if(parsedFolder) {
                     QString _tmpFolder = getFolderName(_nameOfFile);
-                    if(_tmpFolder != "") folderName = "OE" + _tmpFolder;
+                    if(!_tmpFolder.isEmpty()) folderName = "OE" + _tmpFolder;
                 }
                 info->addLog(i18n("Importing OE5+ Mailbox %1").arg( "../" + _nameOfFile));
                 info->setTo(folderName);
@@ -368,7 +368,7 @@
 QString FilterOE::parseFolderString( QDataStream& ds, int filePos )
 {
     char tmp;
-    QString returnString = "";
+    QString returnString;
     int wasAt = ds.device()->at();
     ds.device()->at(filePos);
     
@@ -389,15 +389,15 @@
 {
     bool found = false;
     bool foundFilename = false;
-    QString folder = "";
+    QString folder;
     // we must do this because folder with more than one upper letter
     // at start have maybe not a file named like the folder !!!
     QString search = filename.lower();
     
     while (!found)
     {
-        for ( QValueList<QString[4]>::Iterator it = folderStructure.begin(); it != folderStructure.end(); it++) {
-            QString tmp[4] = *it;
+        for ( FolderStructureIterator it = folderStructure.begin(); it != folderStructure.end(); it++) {
+            FolderStructure tmp = *it;
             if(foundFilename == false) {
                 QString _tmpFileName = tmp[1];
                 _tmpFileName = _tmpFileName.lower();
@@ -410,7 +410,7 @@
                 QString _currentID = tmp[2];
                 QString _parentID = tmp[3];
                 if(_currentID == search) {
-                    if(_parentID == "") { // this is the root of the folder
+                    if(_parentID.isEmpty()) { // this is the root of the folder
                         found = true;
                         break;
                     } else {
@@ -421,7 +421,7 @@
             }
         }
         // need to break the while loop maybe in some cases
-        if((foundFilename == false) && (folder == "")) return folder;
+        if((foundFilename == false) && (folder.isEmpty())) return folder;
     }
     return folder;
 }
Index: kmailcvt/filter_mbox.cxx
===================================================================
--- kmailcvt/filter_mbox.cxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_mbox.cxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -69,8 +69,9 @@
 
       while ( ! mbox.atEnd() ) {
 	KTempFile tmp;
+	QIODevice::Offset filepos = 0;
         /* comment by Danny:
-        * Don't use QTextStream to read from mbox, etter use QDataStream. QTextStream only 
+        * Don't use QTextStream to read from mbox, better use QDataStream. QTextStream only 
         * support Unicode/Latin1/Locale. So you lost information from emails with 
         * charset!=Unicode/Latin1/Locale (e.g. KOI8-R) and Content-Transfer-Encoding != base64 
         * (e.g. 8Bit). It also not help to convert the QTextStream to Unicode. By this you
@@ -83,7 +84,14 @@
 	tmp.file()->writeBlock( input, l );
 	
 	while ( ! mbox.atEnd() &&  (l = mbox.readLine(input.data(),MAX_LINE)) && ((seperate = input.data()).left(5) != "From ")) {
-	  tmp.file()->writeBlock( input, l );
+          tmp.file()->writeBlock( input, l );
+
+          // workaround to fix hang if a corrupted mbox contains some 
+          // binary data, for more see bug #106796
+          if (mbox.at() == filepos)
+            mbox.at(mbox.size());
+          else 
+            filepos = mbox.at();
 	}
 	tmp.close();
         first_msg = false;
Index: kmailcvt/filter_pmail.hxx
===================================================================
--- kmailcvt/filter_pmail.hxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_pmail.hxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -50,8 +50,18 @@
     QDir dir;
     /**  pointer to the info */
     FilterInfo * inf;
-    /** QStringList with the foldernames, First String contains the ID, the second the folder */
-    QValueList<QString[5]> folderMatrix;
+
+    /** Folder structure here has 5 entries. */
+    typedef FolderStructureBase<5> FolderStructure;
+    /** List with the folder matrix, which contains following strings:
+	1. type (2 for root-folder, 1 for folder, 0 for mailarchiv)
+	2. type (1 for root-folder, 3 for folder, 0 for mailarchiv)  
+	3. "ID:flag:filename" of folder/archiv   
+	4. "ID:name" of parent folder
+	5. name of folder/archiv
+    */
+    QValueList<FolderStructure> folderMatrix;
+    typedef QValueList<FolderStructure>::Iterator FolderStructureIterator;
     
     /** true, if the folderfile is parsed **/
     bool folderParsed;
Index: kmailcvt/filter_oe.hxx
===================================================================
--- kmailcvt/filter_oe.hxx	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kmailcvt/filter_oe.hxx	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -61,8 +61,17 @@
     bool parsedFolder;
     /** true if the current parsing file is the folder file */
     bool currentIsFolderFile;
+
+    /** Folder structure with following  4 entries:
+        1. descriptive folder name 
+        2. filename 
+        3. ID of current folder
+        4. ID of parent folder 
+    */
+    typedef FolderStructureBase<4> FolderStructure;
     /** matrix with information about the folder structure*/
-    QValueList<QString[4]> folderStructure;
+    QValueList<FolderStructure> folderStructure;
+    typedef QValueList<FolderStructure>::Iterator FolderStructureIterator;
 
     /** name of the current folder */
     QString folderName;
Index: libkdepim/komposer/plugins/default/defaulteditor.desktop
===================================================================
--- libkdepim/komposer/plugins/default/defaulteditor.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ libkdepim/komposer/plugins/default/defaulteditor.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -28,6 +28,7 @@
 Name[et]=Komposeri redaktor
 Name[eu]=Komposer editorea
 Name[fr]=Éditeur Komposer
+Name[ga]=Eagarthóir Komposer
 Name[hi]=कम्पोज़र एडिटर
 Name[hu]=Komposer szerkesztő
 Name[is]=Komposer ritill
@@ -57,6 +58,7 @@
 Comment[et]=Komposeri vaikeredaktor
 Comment[eu]=Komposer editore lehenetsia
 Comment[fr]=Éditeur Komposer par défaut
+Comment[ga]=Eagarthóir réamhshocraithe Komposer
 Comment[he]=עורך ברירת מחדל של Kompoer
 Comment[hi]=कम्पोज़र डिफ़ॉल्ट संपादक
 Comment[hu]=A Komposer alapértelmezett szerkesztője
Index: libkdepim/komposer/core/komposerplugin.desktop
===================================================================
--- libkdepim/komposer/core/komposerplugin.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ libkdepim/komposer/core/komposerplugin.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -17,6 +17,7 @@
 Name[eu]=Komposer plugin-a
 Name[fi]=Komposer-liitännäinen
 Name[fr]=Module de Komposer
+Name[ga]=Breiseán Komposer
 Name[gl]=Extensión Komposer
 Name[he]=תןסף Kompoer
 Name[hi]=कम्पोज़र प्लगइन
Index: libkdepim/komposer/core/komposereditor.desktop
===================================================================
--- libkdepim/komposer/core/komposereditor.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ libkdepim/komposer/core/komposereditor.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -16,6 +16,7 @@
 Comment[eu]=Komposer editorea
 Comment[fi]=Komposer-editori
 Comment[fr]=Éditeur Komposer
+Comment[ga]=Eagarthóir Komposer
 Comment[gl]=Editor Komposer
 Comment[he]=עורך של Komposer
 Comment[hi]=कम्पोज़र एडिटर
Index: libkcal/localdir.desktop
===================================================================
--- libkcal/localdir.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ libkcal/localdir.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -16,6 +16,7 @@
 Name[eu]=Egutegia direktorio lokalean
 Name[fi]=Kalenteri paikallisessa hakemistossa
 Name[fr]=Calendrier dans un dossier local
+Name[ga]=Féilire i gComhadlann Logánta
 Name[hi]=स्थानीय डिरेक्ट्री में कैलेन्डर
 Name[hu]=Helyi könyvtárban tárolt naptár
 Name[is]=Dagatal í staðbundinni möppu
Index: libkcal/local.desktop
===================================================================
--- libkcal/local.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ libkcal/local.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -16,6 +16,7 @@
 Name[eu]=Egutegia fitxategi lokalean
 Name[fi]=Kalenteri paikallisessa tiedostossa
 Name[fr]=Calendrier dans un fichier local
+Name[ga]=Féilire i gComhad Logánta
 Name[hi]=स्थानीय फ़ाइल में कैलेन्डर
 Name[hu]=Helyi fájlban tárolt naptár
 Name[is]=Dagatal í staðbundinni skrá
Index: kpilot/ChangeLog
===================================================================
--- kpilot/ChangeLog	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/ChangeLog	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -13,6 +13,19 @@
 TODO: give the knotes conduit a decent test mode.
 TODO: only re-write a local database if it is changed.
 
+2005-07-20  Jason 'vanRijn' Kasper
+* D'oh!  Version number should change from 4.5.2 to 4.5.3 since we actually
+  did do stuff since KDE 3.4.1, and being that 3.4.2 just got tagged 8
+  hours ago, here's hoping we can get this fixed before 3.4.2 is declared
+  officially official.  =:/
+
+2005-05-26  Jason 'vanRijn' Kasper
+* Fixing nasty little bug (#106324) causing data loss in memofile conduit
+* isFirstSync() is true for both copyHHtoPC and copyPCtoHH.  memofile
+  conduit only wants to do copyHHtoPC if we're either told to, or we don't
+  have any local memos in the filesystem
+  
+
 2005-05-22  Adriaan de Groot
 * Gratuitously update the version number, call this one "kind" (Dutch for
   "child", or otherwise short for "kind of liable to eat your data").
Index: kpilot/conduits/perlconduit/perl-conduit.desktop
===================================================================
--- kpilot/conduits/perlconduit/perl-conduit.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/conduits/perlconduit/perl-conduit.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -13,6 +13,7 @@
 Name[et]=Perl (näidis)
 Name[eu]=Perl (adibidea)
 Name[fr]=Perl (exemple)
+Name[ga]=Perl (Sampla)
 Name[gl]=Perl (Mostra)
 Name[hi]=पर्ल (सेंपल)
 Name[hu]=Perl (példa)
Index: kpilot/conduits/pythonconduit/python-conduit.desktop
===================================================================
--- kpilot/conduits/pythonconduit/python-conduit.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/conduits/pythonconduit/python-conduit.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -13,6 +13,7 @@
 Name[et]=Python (näidis)
 Name[eu]=Python (adibidea)
 Name[fr]=Python (exemple)
+Name[ga]=Python (Sampla)
 Name[gl]=Python (Mostra)
 Name[hi]=पायथन (सेम्पल)
 Name[hu]=Python (példa)
Index: kpilot/conduits/vcalconduit/vcal-conduit.desktop
===================================================================
--- kpilot/conduits/vcalconduit/vcal-conduit.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/conduits/vcalconduit/vcal-conduit.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -15,6 +15,7 @@
 Name[eu]=Egutegia (KOrganizer)
 Name[fi]=Kalenteri (KOrganizer)
 Name[fr]=Agenda (KOrganizer)
+Name[ga]=Féilire (KOrganizer)
 Name[gl]=Calendario (KOrganizer)
 Name[hi]=कैलेण्डर (के-आर्गेनाइज़र)
 Name[hu]=Naptár (KOrganizer)
Index: kpilot/kpilot/kpilot_config.desktop
===================================================================
--- kpilot/kpilot/kpilot_config.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/kpilot/kpilot_config.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -63,6 +63,7 @@
 Comment[eu]=KPilot-en konfigurazio nagusia
 Comment[fi]=KPilotin pääasetukset
 Comment[fr]=Configuration principale de KPilot
+Comment[ga]=Príomhchumraíocht KPilot
 Comment[gl]=Configuración Principal de KPilot 
 Comment[hi]=के-पायलट मुख्य कॉन्फ़िगरेशन
 Comment[hu]=A KPilot legfontosabb beállításai
@@ -99,6 +100,7 @@
 Keywords[eu]=kpilot,nagusia
 Keywords[fi]=kpilot
 Keywords[fr]=kpilot,principal
+Keywords[ga]=kpilot,príomh
 Keywords[gl]=kpilot,principal
 Keywords[hi]=के-पायलट,मुख्य
 Keywords[hu]=kpilot,alap
Index: kpilot/kpilot/kpilot.desktop
===================================================================
--- kpilot/kpilot/kpilot.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/kpilot/kpilot.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -31,6 +31,7 @@
 GenericName[eu]=PalmPilot tresna
 GenericName[fi]=Palm Pilot -työkalu
 GenericName[fr]=Outil pour le Palm Pilot
+GenericName[ga]=Uirlis PalmPilot
 GenericName[gl]=Ferramenta de PalmPilot
 GenericName[he]=כלי פאלם-פיילוט
 GenericName[hi]=पाम-पायलट औज़ार
Index: kpilot/lib/options.h
===================================================================
--- kpilot/lib/options.h	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kpilot/lib/options.h	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -224,7 +224,7 @@
 #define FUNCTIONSETUPL(a) const int fname = a; Q_UNUSED(fname);
 #endif
 
-#define KPILOT_VERSION	"4.5.2 (kind)"
+#define KPILOT_VERSION	"4.5.3 (peuter)"
 
 
 // Function to expand newlines in rich text to <br>\n
Index: kontact/plugins/specialdates/kcmsdsummary.desktop
===================================================================
--- kontact/plugins/specialdates/kcmsdsummary.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kontact/plugins/specialdates/kcmsdsummary.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -86,7 +86,7 @@
 Keywords[hu]=születésnap,évforduló,szabadság,konfigurálás,beállítások
 Keywords[is]=afmæli, frídagar, stillingar, stilla
 Keywords[it]=compleanno, anniversario, vacanze, configura, impostazioni
-Keywords[lt]=birthday, anniversary, holiday, configure, settings, konfigūruoti, nustatymai, gimtadieniai, išeiginės, jubiliejai
+Keywords[lt]=birthday, anniversary, holiday, configure, settings, konfigūruoti, nustatymai, gimtadieniai, išeiginės,sukaktys
 Keywords[nb]=fødselsdag, jubileum, ferie, sette opp, innstillinger
 Keywords[nl]=verjaardag,jubileum,vakantie,instellingen,configuratie,feestdag
 Keywords[nn]=fødselsdag,bursdag,gebursdag,jubileum,ferie,helgedag,merkedag,oppsett,innstillingar
Index: kontact/plugins/kaddressbook/kcmkabsummary.desktop
===================================================================
--- kontact/plugins/kaddressbook/kcmkabsummary.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kontact/plugins/kaddressbook/kcmkabsummary.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -107,7 +107,7 @@
 Keywords[is]=afmæli, stillingar, stilla
 Keywords[it]=compleanno, anniversario, configura, impostazioni
 Keywords[ja]=誕生日,記念日,設定,設定
-Keywords[lt]=birthday, anniversary, configure, settings, gimtadienis, jubiliejus, konfigūruoti, nustatymai
+Keywords[lt]=birthday, anniversary, configure, settings, gimtadienis,sukaktys, konfigūruoti, nustatymai
 Keywords[nb]=fødselsdag, jubileum, sette opp, innstillinger
 Keywords[nl]=verjaardag,instellingen,configuratie
 Keywords[nn]=fødselsdag,bursdag,gebursdag,jubileum,oppsett,innstillingar
Index: kontact/plugins/korganizer/todoplugin.desktop
===================================================================
--- kontact/plugins/korganizer/todoplugin.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kontact/plugins/korganizer/todoplugin.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -30,7 +30,7 @@
 Comment[hu]=Kontact-bővítőmodul a KOrganizer feladatlistához
 Comment[is]=Kontact KOrganizer tossalista íforrit
 Comment[it]=Plugin Kontact per le cose da fare di KOrganizer
-Comment[lt]=Kontact KOrganizer užduočių sąrašo priedas
+Comment[lt]=Kontact KOrganizer darbų sąrašo priedas
 Comment[nb]=Kontact programtillegg for KOrganizer gjøreliste
 Comment[nl]=Kontact KOrganizer takenlijstplugin
 Comment[nn]=Kontact-programtillegg for KOrganizer-hugselister
@@ -62,7 +62,7 @@
 Name[hu]=Feladatok
 Name[is]=Tossalisti
 Name[it]=Cose da fare
-Name[lt]=Užduočių sąrašas
+Name[lt]=Darbų sąrašas
 Name[nb]=Gjøreliste
 Name[nl]=Takenlijst
 Name[nn]=Hugseliste
Index: kontact/src/kontact.setdlg
===================================================================
--- kontact/src/kontact.setdlg	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ kontact/src/kontact.setdlg	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -522,7 +522,7 @@
 Comment[nl]=Het weer
 Comment[nn]=Vêrinformasjon
 Comment[pl]=Informacje o pogodzie
-Comment[pt]=Informações Metereológicas
+Comment[pt]=Informações Meteorológicas
 Comment[pt_BR]=Informações sobre o Clima
 Comment[ru]=Информация о погоде
 Comment[sl]=Podatki o vremenu
Index: korganizer/plugins/projectview/projectview.desktop
===================================================================
--- korganizer/plugins/projectview/projectview.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ korganizer/plugins/projectview/projectview.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -67,7 +67,7 @@
 Comment[hu]=Ezzel a modullal projekttervező nézet alakítható ki a KOrganizerben (például a feladatokhoz vagy a havi nézetekhez). Ha aktiválja ezt a modult, átválthat projektnézetbe, hogy a feladatok projekttervként legyenek megtekinthetők.
 Comment[is]=Þetta íforrit veitir verkskipulagssýn fyrir KOrganizer (svipað og verkþátta eða mánaðarsýn). Ef þú virkjar þetta íforrit, getur þú á einfaldan hátt skipt yfir í verkefnasýn og skoðað verkþáttalistann þinn í skipuleggjara.
 Comment[it]=Questo plugin fornisce una vista di pianificazione progetto per KOrganizer (come la vista delle cose da fare o quella mensile). Se abiliti questo plugin, puoi passare dalla vista progetto alla vista delle cose da fare come in un pianificatore di progetti.
-Comment[lt]=Šis priedas pateikia KOrganizaer projekto planavimo vaizdą (panašiai kaip užduočių ar mėnesio vaizdai). Jei įjungsite šį įskiepį, galėsite peršokti į projekto vaizdą ir peržiūrėti savo užduočių sąrašą kaip projektą.
+Comment[lt]=Šis priedas pateikia KOrganizaer projekto planavimo vaizdą (panašiai kaip darbų ar mėnesio vaizdai). Jei įjungsite šį įskiepį, galėsite peršokti į projekto vaizdą ir peržiūrėti savo darbų sąrašą kaip projektą.
 Comment[nb]=Dette tilleggsprogrammet lager en prosjektplan-visning for KOrganizer (som i gjørelister og månedsvisninger). Slår du på dette tilleggsprogrammet kan du bytte til prosjektvisning og se på gjørelista som i en prosjektplan.
 Comment[nl]=Deze plugin biedt een projectweergave voor KOrganizer (net zoals een takenlijst of maandweergave). Wanneer u deze plugin aanzet, kunt u deze weergave aanzetten om uw takenlijst te bekijken als een projectplanner.
 Comment[pl]=Ta wtyczka tworzy widok planowania projektu w KOrganizerze (tak jak widok zadań do zrobienia lub widok miesiąca). Po włączeniu tej wtyczki możliwe jest przełączenie się na widok projektu zadań do zrobienia.
Index: korganizer/plugins/printing/whatsnext/whatsnextprint.desktop
===================================================================
--- korganizer/plugins/printing/whatsnext/whatsnextprint.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ korganizer/plugins/printing/whatsnext/whatsnextprint.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -41,7 +41,7 @@
 Comment[hu]=Ezzel a modullal kinyomtathatók a rövidesen aktuálissá váló feladatok és események.
 Comment[is]=Þetta íforrit gerir þér kleyft að prenta út lista yfir alla væntanlega atburði og verkþætti.
 Comment[it]=Questo plugin vi permette di stampare una lista dei prossimi eventi e cose da fare.
-Comment[lt]=Šis priedas leidžia spausdinti sąrašą visų artėjančių įvykių ir užduočių.
+Comment[lt]=Šis priedas leidžia spausdinti sąrašą visų artėjančių įvykių ir darbų.
 Comment[nb]=Med dette tilleggsprogrammet kan du skrive ut en liste over kommende hendelser og ting som skal gjøres.
 Comment[nl]=Deze plugin maakt het mogelijk om een lijst van alle komende evenementen en taken uit te printen.
 Comment[nn]=Dette programtillegget lèt deg skriva ut ei liste over alle dei komande hendingane og oppføringane i hugselista.
Index: korganizer/plugins/printing/list/listprint.desktop
===================================================================
--- korganizer/plugins/printing/list/listprint.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ korganizer/plugins/printing/list/listprint.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -43,7 +43,7 @@
 Comment[hu]=Ezzel a modullal listaként kinyomtathatók a feladatok és események.
 Comment[is]=Þetta íforrit gerir þér kleyft að prenta út lista með atburðum og verkþáttum.
 Comment[it]=Questo plugin ti permette di stampare eventi e cose da fare in modalità elenco.
-Comment[lt]=Šis priedas leidžia spausdinti įvykius ir užduotis sąrašo forma.
+Comment[lt]=Šis priedas leidžia spausdinti įvykius ir darbus sąrašo forma.
 Comment[nb]=Med dette tilleggsprogrammet kan du skrive ut hendelser og gjørelister på liste-form.
 Comment[nl]=Deze plugin maakt het mogelijk om evenementen en taken in lijstvorm uit te printen.
 Comment[nn]=Dette programtillegget lèt deg skriva ut hendingar og hugselister på listeform.
Index: korganizer/plugins/timespanview/timespanview.desktop
===================================================================
--- korganizer/plugins/timespanview/timespanview.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ korganizer/plugins/timespanview/timespanview.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -40,7 +40,7 @@
 Comment[hu]=Ezzel modullal egy időszakot lehet áttekinteni a KOrganizerben (például a feladatokat vagy a havi nézeteket). Ha aktiválja ezt a modult, átválthat időszakos nézetbe, hogy az események Gantt-diagramon legyenek megtekinthetők.
 Comment[is]=Þetta íforrit veitir tímabilssýn fyrir KOrganizer (svipað og verkþátta eða mánaðarsýn). Ef þú virkjar þetta íforrit getur þú skipt yfir í tímabilssýn og skoðað atburðina þína eins og á Gantt skýringarmynd.
 Comment[it]=Questo plugin fornisce una vista a intervalli temporali. Se abiliti questo plugin, potrai passare alla vista a intervalli temporali e vedere i tuoi eventi come in un diagramma di Gantt.
-Comment[lt]=Šis priedas korganizer programoje sudaro galimybę apžvelgti laiko tarpą (pvz., užduočių arba mėnesio peržiūra). Įgalinus šį priedą galėsite persijungti į laiko tarpo peržiūrą ir žiūrėti įvykius tarsi Gantt diagramoje.
+Comment[lt]=Šis priedas korganizer programoje sudaro galimybę apžvelgti laiko tarpą (pvz., darbų arba mėnesio peržiūra). Įgalinus šį priedą galėsite persijungti į laiko tarpo peržiūrą ir žiūrėti įvykius tarsi Gantt diagramoje.
 Comment[nb]=Dette tilleggsprogrammet lager en periodevisning for korganizer (slik som gjøreliste og månedsvisning). Hvis du slår på dette tilleggsprogrammet kan du bytte til periodevisning og se hendelser som i et Gantt-diagram.
 Comment[nl]=Deze plugin biedt een weergave voor KOrganizer (zoals de takenlijst en de maandweergave). Wanneer u deze plugin inschakelt kunt u kiezen voor een weergave waarin u uw evenementen in een Gantt-diagram kunt bekijken.
 Comment[pl]=Ta wtyczka tworzy widok zakresu czasu w KOrganizerze (tak jak widok zadań do zrobienia lub widok miesiąca). Po włączeniu tej wtyczki możliwe jest przełączenie się na widok zakresu czasu i przeglądanie zdarzeń jak na diagramie Gantta.
Index: korganizer/korganizer-import.desktop
===================================================================
--- korganizer/korganizer-import.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ korganizer/korganizer-import.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -21,6 +21,7 @@
 Name[et]=Kalendri import
 Name[eu]=Inportatu egutegia
 Name[fr]=Importer un calendrier
+Name[ga]=Iompórtáil Féilire
 Name[hi]=कैलेन्डर आयात करें
 Name[hu]=Naptár importálása
 Name[is]=Flytja inn dagatal
Index: korganizer/korganizer_configmain.desktop
===================================================================
--- korganizer/korganizer_configmain.desktop	(.../tags/KDE/3.4.2/kdepim)	(revision 442272)
+++ korganizer/korganizer_configmain.desktop	(.../branches/KDE/3.4/kdepim)	(revision 442272)
@@ -16,6 +16,7 @@
 Name[ar]=شخصي
 Name[be]=Пэрсанальныя
 Name[bg]=Персоналност
+Name[br]=Diouzhoc'h :
 Name[bs]=Lično
 Name[cs]=Osobní
 Name[cy]=Personol

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.4/kde-common/admin


