--- kdepim-3.5.9/akregator/src/mk4storage/metakit/include/mk4.h.orig	2008-02-28 11:35:22.000000000 +0000
+++ kdepim-3.5.9/akregator/src/mk4storage/metakit/include/mk4.h	2008-02-28 11:36:46.000000000 +0000
@@ -102,7 +102,7 @@
 #if !defined (_WIN32) && !defined (q4_LONG64)
 #if (defined (_PA_RISC2_0) && defined(__hpux)) || defined (__powerpc64__) || defined(__sparcv9) || \
     defined (__x86_64__) || defined (__s390x__) || defined (__alpha) || \
-    (defined (__ia64) && (!defined (__HP_aCC) || defined(__LP64__)))
+    (defined (__ia64) && (!defined (__HP_aCC) || defined(__LP64__)) || (defined (__sparc__) && defined(__arch64__)))
 #define q4_LONG64 1
 #endif
 #endif
