--- kdepim-3.5.5/kmail/kmmessage.cpp~	2006-10-01 20:30:30.000000000 +0300
+++ kdepim-3.5.5/kmail/kmmessage.cpp	2007-02-25 19:22:15.495706384 +0200
@@ -1616,7 +1616,7 @@
   setSubject("");
   setDateToday();
 
-  setHeaderField("User-Agent", "KMail/" KMAIL_VERSION );
+  setHeaderField("User-Agent", "PLD Linux KMail/" KMAIL_VERSION );
   // This will allow to change Content-Type:
   setHeaderField("Content-Type","text/plain");
 }
