--- kdepim-030317/kmail/kmcomposewin.cpp~	Mon Mar 10 01:20:13 2003
+++ kdepim-030317/kmail/kmcomposewin.cpp	Thu Mar 20 08:36:27 2003
@@ -871,7 +871,7 @@
   (void) new KAction (i18n("Cl&ean Spaces"), 0, this, SLOT(slotCleanSpace()),
                       actionCollection(), "clean_spaces");
 
-  fixedFontAction = new KToggleAction( i18n("Use Fi&xed Font"), 0, this,
+  fixedFontAction = new KToggleAction( i18n("Use Fi&xed Font"), "text", 0, this,
                       SLOT(slotUpdateFont()), actionCollection(), "toggle_fixedfont" );
 
   //these are checkable!!!
--- kdepim-030908/kmail/kmmainwidget.cpp~	2003-09-07 19:41:22.000000000 +0200
+++ kdepim-030908/kmail/kmmainwidget.cpp	2003-09-08 14:34:53.000000000 +0200
@@ -1828,7 +1828,7 @@
   findInMessageAction = new KAction( i18n("&Find in Message..."), "find", KStdAccel::shortcut(KStdAccel::Find), this,
 		      SLOT(slotFind()), actionCollection(), "find_in_messages" );
 
-  (void) new KAction( i18n("Select &All Messages"), KStdAccel::selectAll(), this,
+  (void) new KAction( i18n("Select &All Messages"), "ok", KStdAccel::selectAll(), this,
 		      SLOT(slotMarkAll()), actionCollection(), "mark_all_messages" );
 
   (void) new KAction( i18n("Select Message &Text"),
@@ -1861,10 +1861,10 @@
   removeFolderAction = new KAction( i18n("&Delete Folder"), "editdelete", 0, this,
 		      SLOT(slotRemoveFolder()), actionCollection(), "delete_folder" );
 
-  preferHtmlAction = new KToggleAction( i18n("Prefer &HTML to Plain Text"), 0, this,
+  preferHtmlAction = new KToggleAction( i18n("Prefer &HTML to Plain Text"), "html", 0, this,
 		      SLOT(slotOverrideHtml()), actionCollection(), "prefer_html" );
 
-  threadMessagesAction = new KToggleAction( i18n("&Thread Messages"), 0, this,
+  threadMessagesAction = new KToggleAction( i18n("&Thread Messages"), "view_tree", 0, this,
 		      SLOT(slotOverrideThread()), actionCollection(), "thread_messages" );
 
   threadBySubjectAction = new KToggleAction( i18n("Thread Messages also by &Subject"), 0, this,
--- kdepim-030911/kmail/kmreaderwin.cpp~	2003-09-09 23:22:40.000000000 +0200
+++ kdepim-030911/kmail/kmreaderwin.cpp	2003-09-11 20:15:02.000000000 +0200
@@ -537,10 +537,10 @@
                                      ac, "add_bookmarks" );
   mUrlSaveAsAction = new KAction( i18n("Save Link As..."), 0, this,
 			     SLOT(slotUrlSave()), ac, "saveas_url" );
-  mViewSourceAction = new KAction( i18n("&View Source"), Key_V, this,
+  mViewSourceAction = new KAction( i18n("&View Source"), "source", Key_V, this,
 		      SLOT(slotShowMsgSrc()), ac, "view_source" );
 
-  mToggleFixFontAction = new KToggleAction( i18n("Use Fi&xed Font"),
+  mToggleFixFontAction = new KToggleAction( i18n("Use Fi&xed Font"), "text",
  			Key_X, this, SLOT(slotToggleFixedFont()),
  			ac, "toggle_fixedfont" );
 
