--- kdepim-3.5.6/kpilot/conduits/Makefile.am.org	2007-01-17 01:00:41.486539750 +0100
+++ kdepim-3.5.6/kpilot/conduits/Makefile.am	2007-01-17 01:01:02.759869250 +0100
@@ -5,12 +5,15 @@
 
 if include_malconduit
 MAL_SUBDIR = malconduit
+else
+MAL_SUBDIR =
 endif
 
 SUBDIRS = \
 	  abbrowserconduit \
 	  docconduit \
 	  knotes \
+	  $(MAL_SUBDIR) \
 	  memofileconduit \
 	  notepadconduit \
 	  popmail \
