--- kdepim-3.1.1/kpilot/conduits/vcalconduit/vcal-conduitbase.cc~	Mon Mar  3 00:49:15 2003
+++ kdepim-3.1.1/kpilot/conduits/vcalconduit/vcal-conduitbase.cc	Wed Mar  5 17:28:51 2003
@@ -30,7 +30,7 @@
 ** Bug reports and questions can be sent to kde-pim@kde.org
 */
 
-static const char *vcalconduitbase_id = "$Id$";
+static const char *vcalconduitbase_id = "$Id$";
 
 #include <options.h>
 #include <unistd.h>
@@ -478,7 +478,7 @@
 		DEBUGCONDUIT<<fname<<": Description: "<<e->summary()<<endl;
 		DEBUGCONDUIT<<fname<<": Time: "
 			<< e->dtStart().toString()
-#if KDE_VERSION > 0x030100
+#if KDE_VERSION > 0x030180
 			<< " until "
 			<< e->dtEnd()
 #endif
