--- kdepim-3.5.4/kmail/kmcomposewin.cpp.orig	2006-07-22 10:14:16.000000000 +0200
+++ kdepim-3.5.4/kmail/kmcomposewin.cpp	2006-10-01 15:08:38.000000000 +0200
@@ -1220,7 +1220,7 @@
   (void) new KAction (i18n("Cl&ean Spaces"), 0, this, SLOT(slotCleanSpace()),
                       actionCollection(), "clean_spaces");
 
-  mFixedFontAction = new KToggleAction( i18n("Use Fi&xed Font"), 0, this,
+  mFixedFontAction = new KToggleAction( i18n("Use Fi&xed Font"), "text", 0, this,
                       SLOT(slotUpdateFont()), actionCollection(), "toggle_fixedfont" );
   mFixedFontAction->setChecked( GlobalSettings::self()->useFixedFont() );
 
--- kdepim-3.5.4/kmail/kmmainwidget.cpp.orig	2006-07-22 10:14:16.000000000 +0200
+++ kdepim-3.5.4/kmail/kmmainwidget.cpp	2006-10-01 15:20:48.000000000 +0200
@@ -2394,7 +2394,7 @@
   mFindInMessageAction = new KAction( i18n("&Find in Message..."), "find", KStdAccel::shortcut(KStdAccel::Find), this,
 		      SLOT(slotFind()), actionCollection(), "find_in_messages" );
 
-  (void) new KAction( i18n("Select &All Messages"), KStdAccel::selectAll(), this,
+  (void) new KAction( i18n("Select &All Messages"), "ok", KStdAccel::selectAll(), this,
 		      SLOT(slotMarkAll()), actionCollection(), "mark_all_messages" );
 
   //----- Folder Menu
@@ -2434,13 +2434,13 @@
   mRemoveFolderAction = new KAction( "foo", "editdelete", 0, this,
 		      SLOT(slotRemoveFolder()), actionCollection(), "delete_folder" );
 
-  mPreferHtmlAction = new KToggleAction( i18n("Prefer &HTML to Plain Text"), 0, this,
+  mPreferHtmlAction = new KToggleAction( i18n("Prefer &HTML to Plain Text"), "html", 0, this,
 		      SLOT(slotOverrideHtml()), actionCollection(), "prefer_html" );
 
   mPreferHtmlLoadExtAction = new KToggleAction( i18n("Load E&xternal References"), 0, this,
 		      SLOT(slotOverrideHtmlLoadExt()), actionCollection(), "prefer_html_external_refs" );
 
-  mThreadMessagesAction = new KToggleAction( i18n("&Thread Messages"), 0, this,
+  mThreadMessagesAction = new KToggleAction( i18n("&Thread Messages"), "view_tree", 0, this,
 		      SLOT(slotOverrideThread()), actionCollection(), "thread_messages" );
 
   mThreadBySubjectAction = new KToggleAction( i18n("Thread Messages also by &Subject"), 0, this,
@@ -2696,7 +2696,7 @@
 		     SLOT(slotCollapseAllThreads()),
 		     actionCollection(), "collapse_all_threads" );
 
-  mViewSourceAction = new KAction( i18n("&View Source"), Key_V, this,
+  mViewSourceAction = new KAction( i18n("&View Source"), "source", Key_V, this,
                                    SLOT(slotShowMsgSrc()), actionCollection(),
                                    "view_source" );
 
--- kdepim-3.5.4/kmail/kmreaderwin.cpp.orig	2006-05-22 20:10:53.000000000 +0200
+++ kdepim-3.5.4/kmail/kmreaderwin.cpp	2006-10-01 15:19:19.000000000 +0200
@@ -665,7 +665,7 @@
   mUrlSaveAsAction = new KAction( i18n("Save Link As..."), 0, this,
                                   SLOT(slotUrlSave()), ac, "saveas_url" );
 
-  mToggleFixFontAction = new KToggleAction( i18n("Use Fi&xed Font"),
+  mToggleFixFontAction = new KToggleAction( i18n("Use Fi&xed Font"), "text",
                                             Key_X, this, SLOT(slotToggleFixedFont()),
                                             ac, "toggle_fixedfont" );
 
