--- kdepim-3.4.0/kontact/src/iconsidepane.cpp.orig	2005-03-19 23:26:33.000000000 +0100
+++ kdepim-3.4.0/kontact/src/iconsidepane.cpp	2005-03-24 01:00:06.000000000 +0100
@@ -124,12 +125,24 @@
 int EntryItem::width( const QListBox *listbox ) const
 {
   int w;
-  if ( text().isEmpty() )
+  if ( text().isEmpty() && !showText() )
+    // dont show text or text empty: only icon's width
     w = mPixmap.width();
   else if (navigator()->viewMode() > SmallIcons)
-    w = QMAX( (int)navigator()->viewMode(), listbox->fontMetrics().width( text() ) );
+  {
+    // icon above text, take max of text's and icon's widths
+    // we actually dont wnat the text's width when we dont show it
+    if (showText())
+      w = QMAX( (int)navigator()->viewMode(), listbox->fontMetrics().width( text() ) );
+    else
+      w = (int)navigator()->viewMode();
+  }
   else
-    w = (int)navigator()->viewMode() + 4 + listbox->fontMetrics().width( text() );
+  {
+    // text next to icon: sum their widhts 
+    w = (int)navigator()->viewMode() + 4;
+    if (showText()) w += listbox->fontMetrics().width( text() );
+  }
 
   return w + ( KDialog::marginHint() * 2 );
 }
@@ -137,14 +150,28 @@
 int EntryItem::height( const QListBox *listbox ) const
 {
   int h;
-  if ( text().isEmpty() )
+  if ( text().isEmpty() && !showText() )
+    // text is empty or we dont show text - only height of pixmap (+4 margin)
     h = mPixmap.height() + 4;
   else if (navigator()->viewMode() > SmallIcons)
-    h = (int)navigator()->viewMode() + listbox->fontMetrics().lineSpacing() + 4;
+  {
+    // text not null but icon is over the text - we have icons > small icons
+    // returns height of icon + distance ebtween baselines + 4 margin
+    // we dont want the distance when we dont show text
+    h = (int)navigator()->viewMode() + 4;
+    if (showText()) h+=listbox->fontMetrics().lineSpacing();
+  }
   else
-    h = QMAX( (int)navigator()->viewMode(),
-              listbox->fontMetrics().lineSpacing() ) +
-              KDialog::spacingHint() * 2;
+  {
+    // icon next to text, height is (max of icon height and 
+    // baseline distance) + 
+    if (showText)
+      h = QMAX( (int)navigator()->viewMode(),
+                listbox->fontMetrics().lineSpacing() ) +
+                KDialog::spacingHint() * 2;
+    else
+      h = (int)navigator()->viewMode() + KDialog::spacingHint() * 2;
+  }
 
   return h;
 }
@@ -192,7 +219,7 @@
     p->setPen( box->colorGroup().highlightedText() );
   }
 
-  if ( !text().isEmpty() ) {
+  if ( !text().isEmpty() && showText() ) {
     QFontMetrics fm = p->fontMetrics();
 
     int x = 0;
@@ -218,6 +245,7 @@
     else
       p->setPen( box->colorGroup().text() );
 
+  if (showText())
     p->drawText( x, y, text() );
   }
 
@@ -241,6 +269,7 @@
   mMouseOn = 0;
   mHighlightItem = 0;
   mViewMode = sizeIntToEnum( Prefs::self()->sidePaneIconSize() );
+  mHaveText=Prefs::self()->SidePaneShowText();
   setSelectionMode( KListBox::Single );
   viewport()->setBackgroundMode( PaletteBackground );
   setFrameStyle( QFrame::NoFrame );
@@ -438,13 +467,39 @@
   menu.insertItem( i18n( "Small" ), (int)SmallIcons );
   menu.insertItem( i18n( "Text Only" ), (int)TextOnly );
   menu.setItemChecked( (int)mViewMode, true );
+ 
+  // needed to check if we show any icons at all, we cant disable
+  // both icons and text right?
+  mViewMode = sizeIntToEnum( Prefs::self()->sidePaneIconSize() );
+  // needed to know whether to ask user for showing or hiding the text
+  mHaveText=Prefs::self()->SidePaneShowText();
+  
+  if (mViewMode != (int)TextOnly)
+  {
+      menu.insertSeparator();
+      // the current seting is no-text
+      if (mHaveText != IconShowText)
+          menu.insertItem( i18n( "Show Text" ), IconShowText);
+      else
+          menu.insertItem( i18n( "Hide Text" ), IconShowText-1);
+  }
+      
   int choice = menu.exec( pos );
 
   if ( choice == -1 )
     return;
-
-  mViewMode = sizeIntToEnum( choice );
-  Prefs::self()->setSidePaneIconSize( choice );
+// choice is not from an icon size menu part and it is either
+// IconShowText=showtext or the IconShowText-1 = dont show text
+  if ( choice < 0 ) 
+  {
+      if (choice==IconShowText || choice==IconShowText-1) 
+         Prefs::self()->setSidePaneShowText(choice) ;
+  }
+  else
+  {
+        mViewMode = sizeIntToEnum( choice );
+        Prefs::self()->setSidePaneIconSize( choice );
+  }
 
   int maxWidth = 0;
   QListBoxItem* it = 0;
--- kdepim-3.4.0/kontact/src/iconsidepane.h.orig	2005-03-19 23:54:23.000000000 +0100
+++ kdepim-3.4.0/kontact/src/iconsidepane.h	2005-03-24 01:01:55.000000000 +0100
@@ -37,6 +37,7 @@
 class Navigator;
 
 enum IconViewMode { LargeIcons = 48, NormalIcons = 32, SmallIcons = 22, TextOnly = 0 };
+const int IconShowText = -2;
 
 /**
   A @see QListBoxPixmap Square Box with an optional icon and a text
@@ -96,6 +97,7 @@
     void highlightItem( EntryItem* item );
 
     IconViewMode viewMode() { return mViewMode; }
+    bool showText() { return mHaveText==IconShowText && viewMode() != TextOnly  ;  }
     IconViewMode sizeIntToEnum(int size) const;
     const QPtrList<KAction> & actions() { return mActions; }
   signals:
@@ -123,6 +125,7 @@
   private:
     SidePaneBase *mSidePane;
     IconViewMode mViewMode;
+    int mHaveText;
 
     QListBoxItem* mMouseOn;
 
--- kdepim-3.4.0/kontact/src/kontact.kcfg.orig	2005-03-19 23:49:52.000000000 +0100
+++ kdepim-3.4.0/kontact/src/kontact.kcfg	2005-03-24 01:03:18.000000000 +0100
@@ -20,7 +20,9 @@
     <entry type="IntList" name="SidePaneSplitter">
       <default>1</default>
     </entry>
-
+    <entry type="Int" name="SidePaneShowText">
+      <default>-2</default>
+    </entry>
     <entry type="Int" name="SidePaneIconSize">
       <default>32</default>
     </entry>
