diff -urN -x CVS kdepim.orig/kaddressbook/csv-templates/kaddressbook.desktop kdepim/kaddressbook/csv-templates/kaddressbook.desktop
--- kdepim.orig/kaddressbook/csv-templates/kaddressbook.desktop	Sun Feb 29 16:47:43 2004
+++ kdepim/kaddressbook/csv-templates/kaddressbook.desktop	Tue Apr 27 15:19:28 2004
@@ -18,6 +18,7 @@
 Name[fo]=KAdressubók (KDE 3.1)
 Name[gl]=Caderno de Enderezos (KDE 3.1)
 Name[hi]=केडीईएड्रेसबुक (केडीई 3.1)
+Name[hr]=KAdressBook (KDE 3.1)
 Name[hu]=Címjegyzék (KDE 3.1)
 Name[is]=KPóstfangaskrá (KDE 3.1)
 Name[it]=Rubrica indirizzi (KDE 3.1)
@@ -26,6 +27,7 @@
 Name[nn]=KDE-adressebok (KDE 3.1)
 Name[nso]=KBuka ya Diaterese (KDE 3.1)
 Name[pl]=Książka adresowa (KDE 3.1)
+Name[pt_BR]=Livro de Endereços (KDE 3.1)
 Name[ro]=Cartea de adrese (KDE 3.1)
 Name[ru]=Адресная книга (KDE 3.1)
 Name[sv]=Kadressbok (KDE 3.1)
diff -urN -x CVS kdepim.orig/kaddressbook/imagewidget.cpp kdepim/kaddressbook/imagewidget.cpp
--- kdepim.orig/kaddressbook/imagewidget.cpp	Sat Apr  3 00:34:54 2004
+++ kdepim/kaddressbook/imagewidget.cpp	Wed Apr 14 18:25:31 2004
@@ -1,25 +1,25 @@
-/*                                                                      
-    This file is part of KAddressBook.                                  
-    Copyright (c) 2003 Tobias Koenig <tokoe@kde.org>                   
-                                                                        
+/*
+    This file is part of KAddressBook.
+    Copyright (c) 2003 Tobias Koenig <tokoe@kde.org>
+
     This program is free software; you can redistribute it and/or modify
     it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or   
-    (at your option) any later version.                                 
-                                                                        
-    This program is distributed in the hope that it will be useful,     
-    but WITHOUT ANY WARRANTY; without even the implied warranty of      
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        
-    GNU General Public License for more details.                        
-                                                                        
-    You should have received a copy of the GNU General Public License   
-    along with this program; if not, write to the Free Software         
-    Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.           
-                                                                        
-    As a special exception, permission is given to link this program    
-    with any edition of Qt, and distribute the resulting executable,    
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+
+    As a special exception, permission is given to link this program
+    with any edition of Qt, and distribute the resulting executable,
     without including the source code for Qt in the source distribution.
-*/                                                                      
+*/
 
 #include <kabc/picture.h>
 #include <kdebug.h>
@@ -114,7 +114,7 @@
   mUseImageUrl->setEnabled( false );
   boxLayout->addWidget( mUseImageUrl, 1, 1 );
 
-  mClearButton = new QPushButton( i18n( "Clear" ), box );
+  mClearButton = new KPushButton( KStdGuiItem::clear(), box );
   mClearButton->setEnabled( false );
   boxLayout->addMultiCellWidget( mClearButton, 3, 3, 0, 1 );
 
@@ -215,7 +215,7 @@
 void ImageWidget::imageChanged()
 {
   updateGUI();
-  
+
   emit changed();
 }
 
diff -urN -x CVS kdepim.orig/kaddressbook/kabcore.cpp kdepim/kaddressbook/kabcore.cpp
--- kdepim.orig/kaddressbook/kabcore.cpp	Sat Feb 28 11:40:02 2004
+++ kdepim/kaddressbook/kabcore.cpp	Sun Apr 11 13:00:04 2004
@@ -945,7 +945,7 @@
   mActionMail->setWhatsThis( i18n( "Send a mail to all selected contacts." ) );
   action->setWhatsThis( i18n( "Print a special number of contacts." ) );
 
-  mActionSave = new KAction(  KStdGuiItem::save().text(), "filesave", CTRL+Key_S, this,
+  mActionSave = KStdAction::save( this,
                              SLOT( save() ), actionCollection(), "file_sync" );
   mActionSave->setWhatsThis( i18n( "Save all changes of the address book to the storage backend." ) );
 
diff -urN -x CVS kdepim.orig/kaddressbook/kcmconfigs/kabconfig.desktop kdepim/kaddressbook/kcmconfigs/kabconfig.desktop
--- kdepim.orig/kaddressbook/kcmconfigs/kabconfig.desktop	Thu Mar  4 14:13:07 2004
+++ kdepim/kaddressbook/kcmconfigs/kabconfig.desktop	Sun Apr 25 01:30:31 2004
@@ -31,6 +31,7 @@
 Name[ga]=Leabhar Seoltaí
 Name[gl]=Libro de Enderezos
 Name[hi]=पता पुस्तिका
+Name[hr]=Adresar
 Name[hu]=Címjegyzék
 Name[id]=Buku Alamat
 Name[is]=Póstfangaskrá
@@ -72,7 +73,7 @@
 Comment[fi]=Muokkaa osoitekirjan asetuksia
 Comment[fr]=Configurer le carnet d'adresses
 Comment[gl]=Configura-lo Caderno de Enderezos
-Comment[hi]=पता पुस्तिका कॉन्फिगर करें
+Comment[hi]=पता पुस्तिका कॉन्फ़िगर करें
 Comment[hu]=A címjegyzék beállítása
 Comment[is]=Stilla póstfangaskrána
 Comment[it]=Configura la rubrica indirizzi
@@ -112,7 +113,7 @@
 Keywords[fi]=kaddressbook, aseta, asetukset
 Keywords[fr]=carnet d'adresses,adresses,kab,kaddressbook,configurer,paramètres,paramètre
 Keywords[gl]=kaddressbook, configurar, opcións
-Keywords[hi]=केएड्रेसबुक, कॉन्फिगर, विन्यास 
+Keywords[hi]=केएड्रेसबुक, कॉन्फ़िगर, विन्यास 
 Keywords[hu]=kaddressbook, konfigurálás, beállítások
 Keywords[is]=kaddressbook, stillingar, stilla
 Keywords[it]=kaddressbook, configura, impostazioni
diff -urN -x CVS kdepim.orig/kaddressbook/kcmconfigs/kabldapconfig.desktop kdepim/kaddressbook/kcmconfigs/kabldapconfig.desktop
--- kdepim.orig/kaddressbook/kcmconfigs/kabldapconfig.desktop	Sat Mar 27 14:13:33 2004
+++ kdepim/kaddressbook/kcmconfigs/kabldapconfig.desktop	Sun Apr 25 01:30:33 2004
@@ -62,7 +62,7 @@
 Comment[fi]=Aseta osoitekirjan LDAP asetuksia
 Comment[fr]=Configurer les paramètres du carnet d'adresses LDAP
 Comment[gl]=Configura-las Opcións LDAP do Caderno de Enderezos
-Comment[hi]=पता पुस्तिका एलडीएपी विन्यास कॉन्फिगर करें
+Comment[hi]=पता पुस्तिका एलडीएपी विन्यास कॉन्फ़िगर करें
 Comment[hu]=A címjegyzék LDAP-beállításainak megváltoztatása
 Comment[is]=Breyta LDAP stillingum póstfangaskrárinnar
 Comment[it]=Configurare le impostazioni LDAP della rubrica indirizzi
@@ -101,7 +101,7 @@
 Keywords[fi]=kaddressbook, aseta, asetukset, LDAP
 Keywords[fr]=carnet d'adresses,adresses,kab,kaddressbook,configurer,paramètres,paramètre,LDAP
 Keywords[gl]=kaddressbook, configurar,opcións, LDAP
-Keywords[hi]=केएड्रेसबुक, कॉन्फिगर, विन्यास,एलडीएपी
+Keywords[hi]=केएड्रेसबुक, कॉन्फ़िगर, विन्यास,एलडीएपी
 Keywords[hu]=kaddressbook, konfiguráció, beállítások, LDAP
 Keywords[is]=kaddressbook, stillingar, stilla, LDAP
 Keywords[it]=kaddressbook, configura, impostazioni, LDAP
diff -urN -x CVS kdepim.orig/kaddressbook/thumbnailcreator/ldifvcardthumbnail.desktop kdepim/kaddressbook/thumbnailcreator/ldifvcardthumbnail.desktop
--- kdepim.orig/kaddressbook/thumbnailcreator/ldifvcardthumbnail.desktop	Sun Feb 29 16:48:07 2004
+++ kdepim/kaddressbook/thumbnailcreator/ldifvcardthumbnail.desktop	Sun Apr 25 01:30:35 2004
@@ -14,7 +14,7 @@
 Name[fi]=Sähköiset käyntikorttitiedostot
 Name[fr]=Fichiers de cartes de commerce électroniques
 Name[gl]=Ficheiros de Tarxeta Electrónica de Negocios
-Name[hi]=इलेक्ट्रानिक व्यापार कार्ड फाइलें
+Name[hi]=इलेक्ट्रानिक व्यापार कार्ड फ़ाइलें
 Name[hu]=Elektronikus névjegykártyák
 Name[is]=Skrár með rafrænum nafnspjöldum
 Name[it]=Biglietto da visita elettronico
diff -urN -x CVS kdepim.orig/kaddressbook/xxport/eudora_xxport.desktop kdepim/kaddressbook/xxport/eudora_xxport.desktop
--- kdepim.orig/kaddressbook/xxport/eudora_xxport.desktop	Sun Feb 29 16:48:20 2004
+++ kdepim/kaddressbook/xxport/eudora_xxport.desktop	Sun Apr 18 14:40:37 2004
@@ -51,6 +51,7 @@
 Comment[et]=Plugin Eudora kontaktide importimiseks ja eksportimiseks
 Comment[fi]=Liitännäinen Eudora-kontaktien tuomiseen ja viemiseen
 Comment[fr]=Module d'import / export de contacts Eudora
+Comment[gl]=Plugin para importar e exportar contactos de Eudora
 Comment[hi]=यूडोरा कान्टेक्ट्स को आयात और निर्यात करने का प्लगइन
 Comment[hu]=Bővítőmodul Eudora névjegyek importálásához/exportálásához
 Comment[is]=Íforrit til að flytja inn og út Eudora tengiliði
diff -urN -x CVS kdepim.orig/kalarm/Changelog kdepim/kalarm/Changelog
--- kdepim.orig/kalarm/Changelog	Sat Mar 27 16:17:02 2004
+++ kdepim/kalarm/Changelog	Sun Apr 25 19:57:00 2004
@@ -1,6 +1,13 @@
 KAlarm
 
-=== Version 1.0.6 --- 26 Mar 2004 ===
+=== Version 1.0.7 --- 18 Apr 2004 ===
+Fix scheduleCommand() and scheduleEmail() DCOP handling.
+Make KAlarm build for "./configure --without-arts".
+Fix email body text not being saved in email alarms.
+Fix loss of --exec command line arguments.
+Show all audio file types in Preferences sound file chooser dialogue.
+
+=== Version 1.0.6 (KDE 3.2.2) --- 26 Mar 2004 ===
 Make the Quit menu item in main window quit the program.
 Update time entry field after editing as soon as mouse cursor leaves it.
 Cancel deferral if reminder is set before it, to prevent it becoming stuck.
diff -urN -x CVS kdepim.orig/kalarm/Makefile.am kdepim/kalarm/Makefile.am
--- kdepim.orig/kalarm/Makefile.am	Sat Feb 21 17:00:31 2004
+++ kdepim/kalarm/Makefile.am	Sat Apr 17 17:44:16 2004
@@ -1,5 +1,10 @@
 SUBDIRS = widgets pixmaps
 
+if include_ARTS
+else
+AM_CPPFLAGS = -DWITHOUT_ARTS
+endif
+
 INCLUDES= -I$(top_srcdir) -I$(top_srcdir)/kalarm \
 	-I$(top_srcdir)/kalarm/widgets \
 	-I$(top_builddir)/libical/src \
diff -urN -x CVS kdepim.orig/kalarm/dcophandler.cpp kdepim/kalarm/dcophandler.cpp
--- kdepim.orig/kalarm/dcophandler.cpp	Fri Jan 30 20:43:12 2004
+++ kdepim/kalarm/dcophandler.cpp	Sat Apr 17 15:50:57 2004
@@ -93,11 +93,11 @@
 	||       func == "scheduleFile(QString,QDateTime,QColor,Q_UINT32,QString,Q_UINT32,QString)")
 		function = SCHEDULE | FILE;
 	//                scheduleCommand(commandLine, dateTime, flags, recurrence)
-	else if (func == "scheduleCommand(const QString&,const QDateTime&,Q_UINT32,Q_INT32,const QString&)"
+	else if (func == "scheduleCommand(const QString&,const QDateTime&,Q_UINT32,const QString&)"
 	||       func == "scheduleCommand(QString,QDateTime,Q_UINT32,QString)")
 		function = SCHEDULE | COMMAND;
 	//                scheduleEmail(addresses, subject, message, attachments, dateTime, flags, recurrence)
-	else if (func == "scheduleEmail(const QString&,const QString&,const QString&,const QString&,const QDateTime&,Q_UINT32,Q_INT32,const QString&)"
+	else if (func == "scheduleEmail(const QString&,const QString&,const QString&,const QString&,const QDateTime&,Q_UINT32,const QString&)"
 	||       func == "scheduleEmail(QString,QString,QString,QString,QDateTime,Q_UINT32,QString)")
 		function = SCHEDULE | EMAIL;
 
@@ -290,9 +290,11 @@
 				flags |= KAlarmEvent::DEFAULT_FONT;
 			}
 			if ((action == KAlarmEvent::MESSAGE  ||  action == KAlarmEvent::FILE)  &&  !(function & PRE_070))
+			{
 				arg >> audioFile;
-			if (!(function & PRE_091))
-				arg >> reminderMinutes;
+				if (!(function & PRE_091))
+					arg >> reminderMinutes;
+			}
 			if (function & (REP_COUNT | REP_END))
 			{
 				KAlarmEvent::RecurType recurType;
diff -urN -x CVS kdepim.orig/kalarm/kalarm.h kdepim/kalarm/kalarm.h
--- kdepim.orig/kalarm/kalarm.h	Tue Mar  9 09:52:34 2004
+++ kdepim/kalarm/kalarm.h	Sat Apr 17 15:50:57 2004
@@ -26,7 +26,7 @@
 #include <config.h>
 #endif
 
-#define KALARM_VERSION "1.0.6"
+#define KALARM_VERSION "1.0.7"
 
 #include <kdeversion.h>
 extern int  marginKDE2;   // KDE2 compatibility
diff -urN -x CVS kdepim.orig/kalarm/kalarmapp.cpp kdepim/kalarm/kalarmapp.cpp
--- kdepim.orig/kalarm/kalarmapp.cpp	Fri Mar 26 14:40:28 2004
+++ kdepim/kalarm/kalarmapp.cpp	Sun May  2 13:20:18 2004
@@ -39,6 +39,7 @@
 #include <kfileitem.h>
 #include <kaction.h>
 #include <kstdaction.h>
+#include <kstdguiitem.h>
 #include <kstaticdeleter.h>
 #include <kdebug.h>
 
@@ -60,8 +61,6 @@
 
 #include <netwm.h>
 
-extern QCString execArguments;
-
 #define       DAEMON_APP_NAME_DEF    "kalarmd"
 const char*   DCOP_OBJECT_NAME     = "display";
 const char*   GUI_DCOP_OBJECT_NAME = "tray";
@@ -403,7 +402,7 @@
 					kdDebug(5950)<<"KAlarmApp::newInstance(): exec\n";
 					if (args->isSet("mail"))
 						USAGE(i18n("%1 incompatible with %2").arg(QString::fromLatin1("--mail")).arg(QString::fromLatin1("--exec")))
-					alMessage = execArguments;
+					alMessage = args->getOption("exec");
 					action = KAlarmEvent::COMMAND;
 				}
 				else if (args->isSet("mail"))
diff -urN -x CVS kdepim.orig/kalarm/main.cpp kdepim/kalarm/main.cpp
--- kdepim.orig/kalarm/main.cpp	Wed Dec 10 20:39:09 2003
+++ kdepim/kalarm/main.cpp	Sun Apr 18 16:48:34 2004
@@ -1,7 +1,7 @@
 /*
  *  main.cpp
  *  Program:  kalarm
- *  (C) 2001, 2002, 2003 by David Jarvie <software@astrojar.org.uk>
+ *  (C) 2001 - 2004 by David Jarvie <software@astrojar.org.uk>
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -31,8 +31,6 @@
 
 #define PROGRAM_NAME "kalarm"
 
-QCString execArguments;    // argument to --exec option
-
 static KCmdLineOptions options[] =
 {
 	{ "a", 0, 0 },
@@ -94,25 +92,38 @@
 		"(c) 2001 - 2003, David Jarvie", 0, "http://www.astrojar.org.uk/linux/kalarm.html");
 	aboutData.addAuthor("David Jarvie", 0, "software@astrojar.org.uk");
 
-	// Fetch all command line options/arguments after --exec and concatenate
-	// them into a single argument. Then change the leading '-'.
+	// Fetch all command line options/arguments after --exec and
+	// concatenate them into a single argument.
 	// This is necessary because the "!" indicator in the 'options'
 	// array above doesn't work (on KDE2, at least)
+	int    newargc = argc;
+	char** newargv = argv;
 	for (int i = 0;  i < argc;  ++i)
 	{
 		if (!strcmp(argv[i], "-e") || !strcmp(argv[i], "--exec"))
 		{
+			// --exec has been found. Create a new args list.
+			newargc = i + 1;
+			newargv = (char**)malloc(newargc * sizeof(char*));
+			for (int j = 0;  j < newargc;  ++j)
+				newargv[j] = argv[j];
+			// Concatenate the --exec arguments into one
+			QCString execArguments;
 			while (++i < argc)
 			{
 				execArguments += argv[i];
 				if (i < argc - 1)
 					execArguments += ' ';
-				argv[i][0] = 'x';     // in case it's an option
 			}
+			i = execArguments.length();
+			newargv[newargc] = (char*)malloc(i + 1);
+			memcpy(newargv[newargc], static_cast<const char*>(execArguments), i);
+			newargv[newargc][i] = 0;
+			++newargc;
 			break;
 		}
 	}
-	KCmdLineArgs::init(argc, argv, &aboutData);
+	KCmdLineArgs::init(newargc, newargv, &aboutData);
 	KCmdLineArgs::addCmdLineOptions(options);
 	KUniqueApplication::addCmdLineOptions();
 
diff -urN -x CVS kdepim.orig/kalarm/mainwindow.cpp kdepim/kalarm/mainwindow.cpp
--- kdepim.orig/kalarm/mainwindow.cpp	Fri Mar 26 14:40:28 2004
+++ kdepim/kalarm/mainwindow.cpp	Sun Apr 11 13:00:04 2004
@@ -648,7 +648,7 @@
 	{
 		int n = items.count();
 		if (KMessageBox::warningContinueCancel(this, i18n("Do you really want to delete the selected alarm?", "Do you really want to delete the %n selected alarms?", n),
-						       i18n("Delete Alarm", "Delete Alarms", n), i18n("&Delete"))
+						       i18n("Delete Alarm", "Delete Alarms", n), KGuiItem( i18n("&Delete"), "editdelete"))
 		    != KMessageBox::Continue)
 			return;
 	}
diff -urN -x CVS kdepim.orig/kalarm/messagewin.cpp kdepim/kalarm/messagewin.cpp
--- kdepim.orig/kalarm/messagewin.cpp	Sun Feb 15 12:42:24 2004
+++ kdepim/kalarm/messagewin.cpp	Tue Apr 20 22:51:08 2004
@@ -49,6 +49,7 @@
 #include <knotifyclient.h>
 #include <kaudioplayer.h>
 #include <kdebug.h>
+#include <kpushbutton.h>
 
 #include "kalarmapp.h"
 #include "alarmcalendar.h"
@@ -69,8 +70,8 @@
 			setReadOnly(true);
 			setWordWrap(QTextEdit::NoWrap);
 		}
-		virtual QSize sizeHint() const
-		{ return QSize(contentsWidth(), contentsHeight() + horizontalScrollBar()->height()); }
+		int scrollBarHeight() const     { return horizontalScrollBar()->height(); }
+		virtual QSize sizeHint() const  { return QSize(contentsWidth(), contentsHeight() + scrollBarHeight()); }
 };
 
 
@@ -308,6 +309,9 @@
 			text->setPaper(mBgColour);
 			text->setPaletteForegroundColor(mFgColour);
 			text->setFont(font);
+			int h = text->sizeHint().height();
+			text->setMinimumHeight(h - text->scrollBarHeight());
+			text->setMaximumHeight(h);
 			QWhatsThis::add(text, i18n("The alarm message"));
 			int lineSpacing = text->fontMetrics().lineSpacing();
 			int vspace = lineSpacing/2 - marginKDE2;
@@ -355,7 +359,7 @@
 	grid->setColStretch(0, 1);     // keep the buttons right-adjusted in the window
 
 	// Close button
-	QPushButton* okButton = new QPushButton(KStdGuiItem::close().text(), topWidget);
+	KPushButton* okButton = new KPushButton(KStdGuiItem::close(), topWidget);
 	// Prevent accidental acknowledgement of the message if the user is typing when the window appears
 	okButton->clearFocus();
 	okButton->setFocusPolicy(QWidget::ClickFocus);    // don't allow keyboard selection
diff -urN -x CVS kdepim.orig/kalarm/prefdlg.cpp kdepim/kalarm/prefdlg.cpp
--- kdepim.orig/kalarm/prefdlg.cpp	Tue Feb 10 23:27:06 2004
+++ kdepim/kalarm/prefdlg.cpp	Mon Apr 26 16:36:01 2004
@@ -54,6 +54,7 @@
 #include "traywindow.h"
 #include "kamail.h"
 #include "kalarmapp.h"
+#include "soundpicker.h"
 #include "prefdlg.moc"
 
 
@@ -795,8 +796,7 @@
 
 void DefaultPrefTab::slotBrowseSoundFile()
 {
-	QString	defaultDir = KGlobal::dirs()->findResourceDir("sound", "KDE_Notify.wav");
-	KURL url = KFileDialog::getOpenURL(defaultDir, i18n("*.wav|Wav Files"), 0, i18n("Choose Sound File"));
+	KURL url = SoundPicker::browseFile(mDefaultSoundFile->text());
 	if (!url.isEmpty())
 		mDefaultSoundFile->setText(url.prettyURL());
 }
diff -urN -x CVS kdepim.orig/kalarm/soundpicker.cpp kdepim/kalarm/soundpicker.cpp
--- kdepim.orig/kalarm/soundpicker.cpp	Sat Feb 21 17:00:31 2004
+++ kdepim/kalarm/soundpicker.cpp	Mon Apr 26 21:39:26 2004
@@ -1,7 +1,7 @@
 /*
  *  soundpicker.cpp  -  widget to select a sound file or a beep
  *  Program:  kalarm
- *  (C) 2002 by David Jarvie  software@astrojar.org.uk
+ *  (C) 2002, 2004 by David Jarvie  software@astrojar.org.uk
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -23,13 +23,16 @@
 #include <qlayout.h>
 #include <qtooltip.h>
 #include <qwhatsthis.h>
+#include <qregexp.h>
 
 #include <kglobal.h>
 #include <klocale.h>
 #include <kfiledialog.h>
 #include <kstandarddirs.h>
 #include <kiconloader.h>
+#ifndef WITHOUT_ARTS
 #include <arts/kplayobjectfactory.h>
+#endif
 
 #include "checkbox.h"
 #include "pushbutton.h"
@@ -117,9 +120,7 @@
 {
 	if (mFilePicker->isOn())
 	{
-		if (mDefaultDir.isEmpty())
-			mDefaultDir = KGlobal::dirs()->findResourceDir("sound", "KDE_Notify.wav");
-		KURL url = KFileDialog::getOpenURL(mDefaultDir, KDE::PlayObjectFactory::mimeTypes().join(" "), 0, i18n("Choose Sound File"));
+		KURL url = browseFile(mFile, mDefaultDir);
 		if (!url.isEmpty())
 		{
 			mFile = url.prettyURL();
@@ -137,6 +138,30 @@
 }
 
 /******************************************************************************
+ * Display a dialogue to choose a sound file, initially highlighting any
+ * specified file. 'initialFile' must be a full path name or URL.
+ * Reply = URL selected. If none is selected, URL.isEmpty() is true.
+ */
+KURL SoundPicker::browseFile(const QString& initialFile, const QString& defaultDir)
+{
+	QString initialDir = !initialFile.isEmpty() ? QString(initialFile).remove(QRegExp("/[^/]*$"))
+	                   : !defaultDir.isEmpty()  ? defaultDir
+	                   : KGlobal::dirs()->findResourceDir("sound", "KDE_Notify.wav");
+#ifdef WITHOUT_ARTS
+	KFileDialog fileDlg(initialDir, QString::fromLatin1("*.wav *.mp3 *.ogg|%1\n*|%2").arg(i18n("Sound Files")).arg(i18n("All Files")), 0, "soundFileDlg", true);
+#else
+	KFileDialog fileDlg(initialDir, KDE::PlayObjectFactory::mimeTypes().join(" "), 0, "soundFileDlg", true);
+#endif
+	fileDlg.setCaption(i18n("Choose Sound File"));
+	fileDlg.setMode(KFile::File | KFile::ExistingOnly);
+	if (!initialFile.isEmpty())
+		fileDlg.setSelection(initialFile);
+	if (fileDlg.exec() == QDialog::Accepted)
+		return fileDlg.selectedURL();
+	return KURL();
+}
+
+/******************************************************************************
  * Set the sound picker button according to whether a sound file is selected.
  */
 void SoundPicker::setFilePicker()
diff -urN -x CVS kdepim.orig/kalarm/soundpicker.h kdepim/kalarm/soundpicker.h
--- kdepim.orig/kalarm/soundpicker.h	Wed Dec 10 20:39:09 2003
+++ kdepim/kalarm/soundpicker.h	Sun Apr 25 19:57:00 2004
@@ -1,7 +1,7 @@
 /*
  *  soundpicker.h  -  widget to select a sound file or a beep
  *  Program:  kalarm
- *  (C) 2002 by David Jarvie  software@astrojar.org.uk
+ *  (C) 2002, 2004 by David Jarvie  software@astrojar.org.uk
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -37,6 +37,7 @@
 		void           setChecked(bool);
 		bool           beep() const;
 		QString        file() const;
+		static KURL    browseFile(const QString& initialFile = QString::null, const QString& initialDir = QString::null);
 
 	protected slots:
 		void           slotSoundToggled(bool on);
diff -urN -x CVS kdepim.orig/kandy/src/main.cpp kdepim/kandy/src/main.cpp
--- kdepim.orig/kandy/src/main.cpp	Sat Feb 28 12:10:35 2004
+++ kdepim/kandy/src/main.cpp	Sun Apr  4 17:19:50 2004
@@ -89,7 +89,7 @@
 {
   KAboutData about("kandy", I18N_NOOP("Kandy"), version, description,
                    KAboutData::License_GPL, "(C) 2001 Cornelius Schumacher",0,
-                   "http://devel-home.kde.org/~kandy");
+                   "http://kandy.kde.org/");
   about.addAuthor( "Cornelius Schumacher", 0, "schumacher@kde.org" );
   KCmdLineArgs::init(argc,argv,&about);
   KCmdLineArgs::addCmdLineOptions(options);
diff -urN -x CVS kdepim.orig/kitchensync/devicemimetypes/cellphone.desktop kdepim/kitchensync/devicemimetypes/cellphone.desktop
--- kdepim.orig/kitchensync/devicemimetypes/cellphone.desktop	Sun Feb 29 16:48:50 2004
+++ kdepim/kitchensync/devicemimetypes/cellphone.desktop	Sat Apr 17 14:47:44 2004
@@ -21,6 +21,7 @@
 Comment[gl]=Teléfono celular conectado
 Comment[he]=טלפון סלולרי מחובר
 Comment[hi]=कनेक्टेड सेलुलर फोन
+Comment[hr]=Spojeni mobilni telefon
 Comment[hu]=Csatlakoztatott mobiltelefon
 Comment[is]=Tengdur farsími
 Comment[it]=Telefono cellulare connesso
diff -urN -x CVS kdepim.orig/kitchensync/devicemimetypes/pda.desktop kdepim/kitchensync/devicemimetypes/pda.desktop
--- kdepim.orig/kitchensync/devicemimetypes/pda.desktop	Sun Feb 29 16:48:52 2004
+++ kdepim/kitchensync/devicemimetypes/pda.desktop	Sat Apr 17 14:47:46 2004
@@ -20,6 +20,7 @@
 Comment[gl]=PDA Conectado
 Comment[he]=אירגונית ממוחשבת מחוברת
 Comment[hi]=कनेक्टेड पीडीए
+Comment[hr]=Spojeni PDA
 Comment[hu]=Csatlakoztatott PDA
 Comment[is]=Tengd lófatölva
 Comment[it]=PDA connesso
diff -urN -x CVS kdepim.orig/kitchensync/kitchensync/fileview/fileview.desktop kdepim/kitchensync/kitchensync/fileview/fileview.desktop
--- kdepim.orig/kitchensync/kitchensync/fileview/fileview.desktop	Sat Mar 27 14:13:42 2004
+++ kdepim/kitchensync/kitchensync/fileview/fileview.desktop	Sun Apr 25 01:30:38 2004
@@ -8,7 +8,7 @@
 Name[fr]=Composant d'aperçu de fichier
 Name[gl]=Parte de Vista de Ficheiros
 Name[he]=רכיב תצוגת קבצים
-Name[hi]=फाइलव्यू-पार्ट
+Name[hi]=फ़ाइलव्यू-पार्ट
 Name[is]=Skráarsýnar-íforrit
 Name[ja]=ファイルプレビューパート
 Name[lt]=Bylų peržiūros dalis
diff -urN -x CVS kdepim.orig/kitchensync/klockfile/ksharedfile.desktop kdepim/kitchensync/klockfile/ksharedfile.desktop
--- kdepim.orig/kitchensync/klockfile/ksharedfile.desktop	Tue Mar  2 16:37:16 2004
+++ kdepim/kitchensync/klockfile/ksharedfile.desktop	Sun Apr 25 01:30:40 2004
@@ -19,7 +19,7 @@
 Name[fr]=Module KDED de partage/verrouillage de fichier
 Name[gl]=Módulo de Compartición/Bloqueo de Ficheiros de KDED
 Name[he]=מודול שיתוף\נעילת קבצים של KDED
-Name[hi]=केडीईडी फाइल साझा/देखना मॉड्यूल
+Name[hi]=केडीईडी फ़ाइल साझा/देखना मॉड्यूल
 Name[hr]=KDED modul za dijeljenje datoteka/zaključavanje
 Name[hu]=KDED fájlmegosztási és -zárolási modul
 Name[is]=KDED skráar læsingar og miðlunar íforrit
@@ -73,7 +73,7 @@
 Comment[fr]=Pour partager des fichiers avec d'autres personnes
 Comment[gl]=Compartir ficheiros entre outros
 Comment[he]=שיתוף קבצים בין משתמשים
-Comment[hi]=अन्यों के बीच फाइलें साझा करें
+Comment[hi]=अन्यों के बीच फ़ाइलें साझा करें
 Comment[hr]=Dijeli datoteke sa ostalima
 Comment[hu]=fájlmegosztás a hálózaton
 Comment[is]=Miðla skrám til annarra notanda
diff -urN -x CVS kdepim.orig/kmail/eventsrc kdepim/kmail/eventsrc
--- kdepim.orig/kmail/eventsrc	Sun Feb 29 16:49:26 2004
+++ kdepim/kmail/eventsrc	Sat Apr 17 14:47:48 2004
@@ -24,6 +24,7 @@
 Comment[fr]=Un nouveau message est arrivé
 Comment[gl]=Chegou Novo Correo
 Comment[hi]=नई डाक आई
+Comment[hr]=Stigla je nova pošta
 Comment[hu]=Új e-mail érkezett
 Comment[is]=Nýr póstur
 Comment[it]=Arrivata nuova posta
diff -urN -x CVS kdepim.orig/kmail/imapprogressdialog.cpp kdepim/kmail/imapprogressdialog.cpp
--- kdepim.orig/kmail/imapprogressdialog.cpp	Sun Jan 11 22:57:07 2004
+++ kdepim/kmail/imapprogressdialog.cpp	Sun Apr 11 13:00:04 2004
@@ -157,7 +157,6 @@
     bottomLayout->addStretch();
 
     KPushButton* pbClose = new KPushButton( KStdGuiItem::close(), this );
-    pbClose->setText( i18n( "Close" ) );
     bottomLayout->addWidget( pbClose );
 
     connect(pbClose, SIGNAL(clicked()), this, SLOT(close())  );
diff -urN -x CVS kdepim.orig/kmail/kfileio.cpp kdepim/kmail/kfileio.cpp
--- kdepim.orig/kmail/kfileio.cpp	Sun Jan 11 22:57:07 2004
+++ kdepim/kmail/kfileio.cpp	Sun Apr 11 13:00:04 2004
@@ -196,7 +196,7 @@
 	rc = KMessageBox::warningContinueCancel(0,
 	     i18n("Failed to make a backup copy of %1.\nContinue anyway?")
 	     .arg(aFileName),
-             i18n("Save to File"), i18n("&Save"));
+             i18n("Save to File"), KStdGuiItem::save());
 	if (rc != KMessageBox::Continue) return FALSE;
       }
     }
diff -urN -x CVS kdepim.orig/kmail/kmcomposewin.cpp kdepim/kmail/kmcomposewin.cpp
--- kdepim.orig/kmail/kmcomposewin.cpp	Tue Mar 30 00:12:01 2004
+++ kdepim/kmail/kmcomposewin.cpp	Wed Apr 14 18:25:31 2004
@@ -70,6 +70,7 @@
 #include <kaction.h>
 #include <kdirwatch.h>
 #include <kstdguiitem.h>
+#include <kiconloader.h>
 
 #include <kspell.h>
 #include <kspelldlg.h>
@@ -3023,7 +3024,7 @@
                                                i18n("<qt>Not all characters fit into the chosen"
                                                     " encoding.<br><br>Send the message anyway?</qt>"),
                                                i18n("Some characters will be lost"),
-                                               i18n("Yes"), i18n("No, let me change the encoding") ) == KMessageBox::Yes);
+                                               KStdGuiItem::yes(), i18n("No, let me change the encoding") ) == KMessageBox::Yes);
       if (!anyway)
       {
         mEditor->setText(oldText);
@@ -4343,7 +4344,7 @@
      mAttachMenu->insertItem(i18n("to view", "View"), this,
                              SLOT(slotAttachView()));
      mAttachMenu->insertItem(i18n("Remove"), this, SLOT(slotAttachRemove()));
-     mSaveAsId = mAttachMenu->insertItem( i18n("Save As..."), this,
+     mSaveAsId = mAttachMenu->insertItem( SmallIcon("filesaveas"), i18n("Save As..."), this,
                                           SLOT( slotAttachSave() ) );
      mPropertiesId = mAttachMenu->insertItem( i18n("Properties"), this,
                                               SLOT( slotAttachProperties() ) );
@@ -5281,7 +5282,7 @@
 void KMComposeWin::slotEditToolbars()
 {
   saveMainWindowSettings(KMKernel::config(), "Composer");
-  KEditToolbar dlg(actionCollection(), "kmcomposerui.rc");
+  KEditToolbar dlg(guiFactory(), this);
 
   connect( &dlg, SIGNAL(newToolbarConfig()),
            SLOT(slotUpdateToolbars()) );
diff -urN -x CVS kdepim.orig/kmail/kmfldsearch.cpp kdepim/kmail/kmfldsearch.cpp
--- kdepim.orig/kmail/kmfldsearch.cpp	Tue Mar 30 19:03:29 2004
+++ kdepim/kmail/kmfldsearch.cpp	Sun Apr 11 13:00:04 2004
@@ -24,6 +24,7 @@
 #include <kstatusbar.h>
 #include <kwin.h>
 #include <kconfig.h>
+#include <kstdaction.h>
 
 #include <qcheckbox.h>
 #include <qlayout.h>
@@ -259,13 +260,11 @@
 					SLOT(slotForwardAttachedMsg()), ac,
 					"search_message_forward_as_attachment" );
   mForwardActionMenu->insert( mForwardAttachedAction );
-  mSaveAsAction = new KAction( i18n("Save &As..."), "filesave", 0,
-                               this, SLOT(slotSaveMsg()), ac, "search_file_save_as" );
+  mSaveAsAction = KStdAction::saveAs( this, SLOT(slotSaveMsg()), ac, "search_file_save_as" );
   mSaveAtchAction = new KAction( i18n("Save Attachments..."), "attach", 0,
                                  this, SLOT(slotSaveAttachments()), ac, "search_save_attachments" );
 
-  mPrintAction = new KAction( i18n( "&Print..." ), "fileprint", 0, this,
-			      SLOT(slotPrintMsg()), ac, "search_print" );
+  mPrintAction = KStdAction::print( this, SLOT(slotPrintMsg()), ac, "search_print" );
   mClearAction = new KAction( i18n("Clear Selection"), 0, 0, this,
 			      SLOT(slotClearSelection()), ac, "search_clear_selection" );
   connect(mTimer, SIGNAL(timeout()), this, SLOT(updStatus()));
diff -urN -x CVS kdepim.orig/kmail/kmfolderindex.cpp kdepim/kmail/kmfolderindex.cpp
--- kdepim.orig/kmail/kmfolderindex.cpp	Sat Jan 17 20:43:26 2004
+++ kdepim/kmail/kmfolderindex.cpp	Thu Apr 29 08:47:27 2004
@@ -181,6 +181,8 @@
 
   mIndexStream = fopen(QFile::encodeName(indexName), "r+"); // index file
   assert( mIndexStream );
+  fcntl(fileno(mIndexStream), F_SETFD, FD_CLOEXEC);
+
   updateIndexStreamPtr();
 
   writeMsgDict();
diff -urN -x CVS kdepim.orig/kmail/kmfoldermaildir.cpp kdepim/kmail/kmfoldermaildir.cpp
--- kdepim.orig/kmail/kmfoldermaildir.cpp	Sun Jan 25 23:51:16 2004
+++ kdepim/kmail/kmfoldermaildir.cpp	Thu Apr 29 08:47:27 2004
@@ -31,6 +31,8 @@
 #include <unistd.h>
 #include <assert.h>
 #include <limits.h>
+#include <unistd.h>
+#include <fcntl.h>
 
 #ifndef MAX_LINE
 #define MAX_LINE 4096
@@ -104,6 +106,8 @@
       emit statusMsg(str);
     } else {
       mIndexStream = fopen(QFile::encodeName(indexLocation()), "r+"); // index file
+      if ( mIndexStream )
+        fcntl(fileno(mIndexStream), F_SETFD, FD_CLOEXEC);
       updateIndexStreamPtr();
     }
 
@@ -174,6 +178,7 @@
     umask(old_umask);
 
     if (!mIndexStream) return errno;
+    fcntl(fileno(mIndexStream), F_SETFD, FD_CLOEXEC);
   }
   else
   {
diff -urN -x CVS kdepim.orig/kmail/kmfoldermbox.cpp kdepim/kmail/kmfoldermbox.cpp
--- kdepim.orig/kmail/kmfoldermbox.cpp	Sat Jan 31 14:34:02 2004
+++ kdepim/kmail/kmfoldermbox.cpp	Thu Apr 29 08:47:27 2004
@@ -154,6 +154,10 @@
 
   mChanged = false;
 
+  fcntl(fileno(mStream), F_SETFD, FD_CLOEXEC);
+  if (mIndexStream)
+     fcntl(fileno(mIndexStream), F_SETFD, FD_CLOEXEC);
+
   return rc;
 }
 
@@ -193,6 +197,7 @@
   umask(old_umask);
 
   if (!mStream) return errno;
+  fcntl(fileno(mStream), F_SETFD, FD_CLOEXEC);
 
   if (!path().isEmpty())
   {
@@ -202,6 +207,7 @@
     umask(old_umask);
 
     if (!mIndexStream) return errno;
+    fcntl(fileno(mIndexStream), F_SETFD, FD_CLOEXEC);
   }
   else
   {
diff -urN -x CVS kdepim.orig/kmail/kmmainwidget.cpp kdepim/kmail/kmmainwidget.cpp
--- kdepim.orig/kmail/kmmainwidget.cpp	Mon Mar 29 22:30:11 2004
+++ kdepim/kmail/kmmainwidget.cpp	Sat May  1 16:34:16 2004
@@ -755,11 +755,11 @@
 
   if(kmkernel->xmlGuiInstance()) {
     KNotifyClient::Instance instance(kmkernel->xmlGuiInstance());
-    KNotifyClient::event(topLevelWidget()->winId(), "new-mail-arrived", 
+    KNotifyClient::event(topLevelWidget()->winId(), "new-mail-arrived",
       i18n("New mail arrived"));
   }
   else
-    KNotifyClient::event(topLevelWidget()->winId(), "new-mail-arrived", 
+    KNotifyClient::event(topLevelWidget()->winId(), "new-mail-arrived",
       i18n("New mail arrived"));
   if (mBeepOnNew) {
     KNotifyClient::beep();
@@ -877,7 +877,7 @@
       i18n("<qt>Are you sure you want to move all messages from "
            "folder <b>%1</b> to the trash?</qt>").arg(mFolder->label());
 
-    if (KMessageBox::warningContinueCancel(this, text, title, title)
+    if (KMessageBox::warningContinueCancel(this, text, title, KGuiItem( title, "edittrash"))
       != KMessageBox::Continue) return;
   }
 
@@ -971,7 +971,7 @@
   }
 
   if (KMessageBox::warningContinueCancel(this, str, i18n("Delete Folder"),
-                                         i18n("&Delete"))
+                                         KGuiItem( i18n("&Delete"), "editdelete"))
       == KMessageBox::Continue)
   {
     if (mFolder->hasAccounts())
@@ -1084,7 +1084,7 @@
         "\"spam\" and may increase the likelihood that your system will be "
         "compromised by other present and anticipated security exploits." ),
       i18n( "Security Warning" ),
-      i18n( "Continue" ),
+      KStdGuiItem::cont(),
       "OverrideHtmlWarning", false);
     if( result == KMessageBox::Cancel ) {
       mPreferHtmlAction->setChecked( false );
diff -urN -x CVS kdepim.orig/kmail/kmmessage.cpp kdepim/kmail/kmmessage.cpp
--- kdepim.orig/kmail/kmmessage.cpp	Sun Feb  1 15:04:51 2004
+++ kdepim/kmail/kmmessage.cpp	Wed Apr 14 18:25:32 2004
@@ -1216,7 +1216,7 @@
 	     "%1, is valid.\n"
              "The failing address will be reported to be %2.")
         .arg(senderStr).arg(receiver),
-	i18n("Bounce Message"), i18n("Continue")) == KMessageBox::Cancel)
+	i18n("Bounce Message"), KStdGuiItem::cont()) == KMessageBox::Cancel)
     {
       return 0;
     }
diff -urN -x CVS kdepim.orig/kmail/kmmimeparttree.cpp kdepim/kmail/kmmimeparttree.cpp
--- kdepim.orig/kmail/kmmimeparttree.cpp	Sun Mar 28 14:24:14 2004
+++ kdepim/kmail/kmmimeparttree.cpp	Wed Apr 14 18:25:32 2004
@@ -96,7 +96,7 @@
         kdDebug(5006) << "\n**\n** KMMimePartTree::itemRightClicked() **\n**" << endl;
 
         QPopupMenu* popup = new QPopupMenu;
-        popup->insertItem( i18n( "Save &As..." ), this, SLOT( slotSaveAs() ) );
+        popup->insertItem( SmallIcon("filesaveas"),i18n( "Save &As..." ), this, SLOT( slotSaveAs() ) );
         popup->insertItem( i18n( "Save as &Encoded..." ), this,
                            SLOT( slotSaveAsEncoded() ) );
         popup->insertItem( i18n( "Save All Attachments..." ), this,
@@ -152,7 +152,7 @@
     QPtrList<partNode> parts;
     parts.append( static_cast<KMMimePartTreeItem *>(item)->node() );
     mReaderWin->setUpdateAttachment();
-    KMSaveAttachmentsCommand *command = new KMSaveAttachmentsCommand( this, parts, 
+    KMSaveAttachmentsCommand *command = new KMSaveAttachmentsCommand( this, parts,
             mReaderWin->message(), encoded );
     command->start();
 }
@@ -167,7 +167,7 @@
         ++it;
     }
     mReaderWin->setUpdateAttachment();
-    KMSaveAttachmentsCommand *command = new KMSaveAttachmentsCommand( this, parts, 
+    KMSaveAttachmentsCommand *command = new KMSaveAttachmentsCommand( this, parts,
             mReaderWin->message(), encoded );
     command->start();
 }
@@ -203,7 +203,7 @@
 
   KIO::filesize_t totalSize = 0;
   QListViewItem * myChild = item->firstChild();
-  while ( myChild ) 
+  while ( myChild )
   {
     totalSize += static_cast<KMMimePartTreeItem*>(myChild)->origSize();
     myChild = myChild->nextSibling();
@@ -230,7 +230,7 @@
   if( node )
     node->setMimePartTreeItem( this );
   setIconAndTextForType( mimetype );
-  if ( parent ) 
+  if ( parent )
     parent->correctSize(this);
 }
 
@@ -256,7 +256,7 @@
   if( node )
     node->setMimePartTreeItem( this );
   setIconAndTextForType( mimetype );
-  if ( listView() ) 
+  if ( listView() )
     static_cast<KMMimePartTree*>(listView())->correctSize(this);
 }
 
diff -urN -x CVS kdepim.orig/kmail/kmreadermainwin.cpp kdepim/kmail/kmreadermainwin.cpp
--- kdepim.orig/kmail/kmreadermainwin.cpp	Sun Jan 11 22:57:07 2004
+++ kdepim/kmail/kmreadermainwin.cpp	Wed Apr 14 18:25:32 2004
@@ -18,6 +18,7 @@
 #include <kstdaccel.h>
 #include <kwin.h>
 #include <kaction.h>
+#include <kiconloader.h>
 
 #include "kmcommands.h"
 #include "kmenubar.h"
@@ -307,7 +308,7 @@
     mReaderWin->viewSourceAction()->plug( menu );
 
     mPrintAction->plug( menu );
-    menu->insertItem(  i18n( "Save &As..." ), mReaderWin, SLOT( slotSaveMsg() ) );
+    menu->insertItem(  SmallIcon("filesaveas"), i18n( "Save &As..." ), mReaderWin, SLOT( slotSaveMsg() ) );
     menu->insertItem( i18n("Save Attachments..."), mReaderWin, SLOT(slotSaveAttachments()) );
   }
   menu->exec(aPoint, 0);
diff -urN -x CVS kdepim.orig/kmail/kmreaderwin.cpp kdepim/kmail/kmreaderwin.cpp
--- kdepim.orig/kmail/kmreaderwin.cpp	Sat Apr  3 00:37:28 2004
+++ kdepim/kmail/kmreaderwin.cpp	Wed Apr 14 18:25:32 2004
@@ -94,6 +94,7 @@
 #include <kprocess.h>
 #include <kdialog.h>
 #include <kaction.h>
+#include <kiconloader.h>
 
 #include <qclipboard.h>
 #include <qhbox.h>
@@ -1523,10 +1524,10 @@
   mAtmCurrent = id;
   mAtmCurrentName = name;
   KPopupMenu *menu = new KPopupMenu();
-  menu->insertItem(i18n("Open..."), 1);
+  menu->insertItem(SmallIcon("fileopen"),i18n("Open..."), 1);
   menu->insertItem(i18n("Open With..."), 2);
   menu->insertItem(i18n("View..."), 3);
-  menu->insertItem(i18n("Save As..."), 4);
+  menu->insertItem(SmallIcon("filesaveas"), i18n("Save As..."), 4);
   menu->insertItem(i18n("Properties..."), 5);
   connect(menu, SIGNAL(activated(int)), this, SLOT(slotAtmLoadPart(int)));
   menu->exec( p ,0 );
diff -urN -x CVS kdepim.orig/kmail/pics/Makefile.am kdepim/kmail/pics/Makefile.am
--- kdepim.orig/kmail/pics/Makefile.am	Sun Nov 16 18:01:27 2003
+++ kdepim/kmail/pics/Makefile.am	Sat May  1 14:57:45 2004
@@ -8,7 +8,7 @@
 	kmmsgpartiallyencrypted.png \
 	kmmsgpartiallysigned.png kmmsgfiller.png kmmsgfullyencrypted.png \
 	kmmsgfullysigned.png kmmsgundefinedencrypted.png \
-	kmmsgundefinedsigned.png kmmsgspam.png \
+	kmmsgundefinedsigned.png kmmsgspam.png kmmsgham.png \
 	kmailwindowlayout1.png \
 	kmailwindowlayout2.png kmailwindowlayout3.png kmailwindowlayout4.png \
 	kmailwindowlayout5.png kmailwindowlayout1_smart_mime.png \
Binary files kdepim.orig/kmail/pics/icons/cr16-action-mark_as_ham.png and kdepim/kmail/pics/icons/cr16-action-mark_as_ham.png differ
Binary files kdepim.orig/kmail/pics/icons/cr22-action-mark_as_ham.png and kdepim/kmail/pics/icons/cr22-action-mark_as_ham.png differ
Binary files kdepim.orig/kmail/pics/icons/cr32-action-mark_as_ham.png and kdepim/kmail/pics/icons/cr32-action-mark_as_ham.png differ
Binary files kdepim.orig/kmail/pics/kmmsgham.png and kdepim/kmail/pics/kmmsgham.png differ
diff -urN -x CVS kdepim.orig/kmail/profiles/profile-default-rc.desktop kdepim/kmail/profiles/profile-default-rc.desktop
--- kdepim.orig/kmail/profiles/profile-default-rc.desktop	Sun Feb 29 16:49:28 2004
+++ kdepim/kmail/profiles/profile-default-rc.desktop	Sun Apr 25 01:30:42 2004
@@ -18,8 +18,9 @@
 Name[fi]=Oletus
 Name[fr]=Défaut
 Name[gl]=Por Omisión
-Name[hi]=डिफाल्ट
+Name[hi]=डिफ़ॉल्ट
 Name[hr]=Uobičajeno
+Name[hsb]=Standard
 Name[hu]=Alapértelmezés
 Name[id]=Standar
 Name[is]=Sjálfgefið
@@ -71,7 +72,7 @@
 Comment[fi]=Normaali profiili
 Comment[fr]=Profil standard
 Comment[gl]=Perfil Estándar
-Comment[hi]=मानक प्रोफाइल
+Comment[hi]=मानक प्रोफ़ाइल
 Comment[hr]=Standardni profil
 Comment[hu]=Standard profil
 Comment[is]=Venjulegt snið
diff -urN -x CVS kdepim.orig/kmail/profiles/profile-high-contrast-rc.desktop kdepim/kmail/profiles/profile-high-contrast-rc.desktop
--- kdepim.orig/kmail/profiles/profile-high-contrast-rc.desktop	Sun Feb 29 16:49:30 2004
+++ kdepim/kmail/profiles/profile-high-contrast-rc.desktop	Sun Apr 25 01:30:44 2004
@@ -14,6 +14,7 @@
 Name[fr]=Contraste élevé
 Name[gl]=Alto Contraste
 Name[hi]=उच्च कंट्रास्ट
+Name[hr]=Visoki kontrast
 Name[hu]=Kontrasztos
 Name[is]=Mikil birtuskil
 Name[it]=Alto contrasto
@@ -54,7 +55,7 @@
 Comment[fi]=Suurennettu kirjasinkoko näkövammaisille käyttäjille
 Comment[fr]=Des polices de caractères plus grandes pour les mal-voyants
 Comment[gl]=Tamaños de fonte aumentados para usuarios con discapacidade visual
-Comment[hi]=कम दिखाई देने वाले उपयोगकर्ताओं के लिए फॉन्ट आकार बढ़ाएँ
+Comment[hi]=कम दिखाई देने वाले उपयोगकर्ताओं के लिए फ़ॉन्ट आकार बढ़ाएँ
 Comment[hr]=Povećana pisma za korisnike sa slabijim vidom
 Comment[hu]=Nagyobb betűméretek látáscsökkent felhasználóknak
 Comment[is]=Stærra letur fyrir notendur með slæma sjón
diff -urN -x CVS kdepim.orig/kmail/profiles/profile-html-rc.desktop kdepim/kmail/profiles/profile-html-rc.desktop
--- kdepim.orig/kmail/profiles/profile-html-rc.desktop	Sun Feb 29 16:49:32 2004
+++ kdepim/kmail/profiles/profile-html-rc.desktop	Sun Apr 25 01:30:46 2004
@@ -19,7 +19,7 @@
 Comment[fi]=Normaali profiili HTML esikatselua käyttäville - vähemmän turvallinen!
 Comment[fr]=Profil standard avec l'aperçu HTML activé - Moins sécurisé !
 Comment[gl]=Perfil estándar con previsualización HTML activada - menos seguro!
-Comment[hi]=एचटीएमएल पूर्वावलोकन के साथ मानक प्रोफाइल सक्षम है - कम सुरक्षित!
+Comment[hi]=एचटीएमएल पूर्वावलोकन के साथ मानक प्रोफ़ाइल सक्षम है - कम सुरक्षित!
 Comment[hr]=Standardni profil sa uključenim pregledom HTML sadržaja - manje sigurno!
 Comment[hu]=Standard profil, HTML-előnézettel (kevésbé biztonságos)
 Comment[is]=Staðlað snið með HTML forsýn - minna öryggi!
diff -urN -x CVS kdepim.orig/kmail/profiles/profile-secure-rc.desktop kdepim/kmail/profiles/profile-secure-rc.desktop
--- kdepim.orig/kmail/profiles/profile-secure-rc.desktop	Sun Feb 29 16:49:36 2004
+++ kdepim/kmail/profiles/profile-secure-rc.desktop	Sat Apr 17 14:47:52 2004
@@ -14,6 +14,7 @@
 Name[fr]=Le plus sécurisé
 Name[gl]=Máis Segura
 Name[hi]=अत्यंत सुरक्षित
+Name[hr]=Najsigurnije
 Name[hu]=Maximális biztonság
 Name[is]=Öruggast
 Name[it]=Massima sicurezza
@@ -52,6 +53,7 @@
 Comment[fr]=Règle toutes les options nécessaires pour atteindre une sécurité maximale
 Comment[gl]=Estabelece tódalas opcións precisas para acada-la máxima seguridade
 Comment[hi]=अधिकतम सुरक्षा हेतु सभी आवश्यक विकल्पों को नियत करे
+Comment[hr]=Podešava sve opcije da postigne najbolju sigurnost
 Comment[hu]=Az összes beállítás a legbiztonságosabb értékre állítva
 Comment[is]=Setur allar stillingar þannig að öryggið sé mest
 Comment[it]=Imposta tutte le opzioni necessario per ottenere la massima sicurezza
diff -urN -x CVS kdepim.orig/kmobile/devices/digicam/libkmobile_digicam.desktop kdepim/kmobile/devices/digicam/libkmobile_digicam.desktop
--- kdepim.orig/kmobile/devices/digicam/libkmobile_digicam.desktop	Sun Feb 29 16:49:38 2004
+++ kdepim/kmobile/devices/digicam/libkmobile_digicam.desktop	Sun Apr 25 01:30:48 2004
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Digital Camera
+Name[ar]=كاميرا رقمية
 Name[bg]=Цифрова камера
 Name[bs]=Digitalna kamera
 Name[ca]=Càmera digital
diff -urN -x CVS kdepim.orig/knode/knarticlefactory.cpp kdepim/knode/knarticlefactory.cpp
--- kdepim.orig/knode/knarticlefactory.cpp	Mon Nov 24 00:05:42 2003
+++ kdepim/knode/knarticlefactory.cpp	Sun Apr 11 13:00:04 2004
@@ -21,6 +21,8 @@
 #include <kwin.h>
 #include <kseparator.h>
 #include <kapplication.h>
+#include <kstdguiitem.h>
+#include <kpushbutton.h>
 
 #include "knarticlefactory.h"
 #include "knglobals.h"
@@ -1080,7 +1082,7 @@
   topL->addSpacing(10);
   topL->addWidget(sep);   
 
-  c_loseBtn=new QPushButton(i18n("&Close"), this);
+  c_loseBtn=new KPushButton(KStdGuiItem::close(), this);
   c_loseBtn->setDefault(true);
   topL->addWidget(c_loseBtn, 0, Qt::AlignRight);
 
diff -urN -x CVS kdepim.orig/knode/kncomposer.cpp kdepim/knode/kncomposer.cpp
--- kdepim.orig/knode/kncomposer.cpp	Sun Jan  4 17:14:54 2004
+++ kdepim/knode/kncomposer.cpp	Sun Apr 11 13:00:04 2004
@@ -837,7 +837,7 @@
   }
   else {
     switch ( KMessageBox::warningYesNoCancel( this, i18n("Do you want to save this article in the draft folder?"),
-                                              QString::null, i18n("&Save"), i18n("&Discard"))) {
+                                              QString::null, KStdGuiItem::save(), KStdGuiItem::discard())) {
       case KMessageBox::Yes :
         r_esult=CRsave;
         break;
diff -urN -x CVS kdepim.orig/knode/knconfig.cpp kdepim/knode/knconfig.cpp
--- kdepim.orig/knode/knconfig.cpp	Tue Oct 21 23:00:12 2003
+++ kdepim/knode/knconfig.cpp	Tue Apr 20 13:58:17 2004
@@ -256,8 +256,6 @@
   f_onts[articleList]=c->readFontEntry("articleListFont",&defFont);
   f_ontNames[articleList]=i18n("Article List");
 
-  updateHexcodes();
-
   //icons
   recreateLVIcons();
   i_cons[newFups]         = UserIcon("newsubs");
@@ -490,12 +488,17 @@
 }
 
 
-void KNConfig::Appearance::updateHexcodes()
+QString KNConfig::Appearance::quotedTextHexcode(int i)
 {
-  sprintf(h_excodes[quoted1Hex], "#%2x%2x%2x", quoteColor1().red(), quoteColor1().green(), quoteColor1().blue());
-  sprintf(h_excodes[quoted2Hex], "#%2x%2x%2x", quoteColor2().red(), quoteColor2().green(), quoteColor2().blue());
-  sprintf(h_excodes[quoted3Hex], "#%2x%2x%2x", quoteColor3().red(), quoteColor3().green(), quoteColor3().blue());
-  sprintf(h_excodes[headerHex], "#%2x%2x%2x", headerDecoColor().red(), headerDecoColor().green(), headerDecoColor().blue());
+  switch(i) {
+    case quoted1Hex:
+      return quoteColor1().name();
+    case quoted2Hex:
+      return quoteColor2().name();
+    case quoted3Hex:
+      return quoteColor3().name();
+  }
+  return textColor().name();
 }
 
 
diff -urN -x CVS kdepim.orig/knode/knconfig.h kdepim/knode/knconfig.h
--- kdepim.orig/knode/knconfig.h	Sat Jul 26 16:51:51 2003
+++ kdepim/knode/knconfig.h	Tue Apr 20 13:58:17 2004
@@ -320,7 +320,7 @@
                         quoted2=5, quoted3=6, url=7, unreadThread=8, readThread=9,
                         unreadArticle=10, readArticle=11, activeItem=12, selectedItem=13 };
 
-    enum HexIndex     { quoted1Hex=0, quoted2Hex=1, quoted3Hex=2, headerHex=3 };
+    enum HexIndex     { quoted1Hex=0, quoted2Hex=1, quoted3Hex=2 };
 
     enum FontIndex    { article=0, articleFixed=1, composer=2, groupList=3, articleList=4 };
 
@@ -350,9 +350,8 @@
     QColor activeItemColor();
     QColor selectedItemColor();
 
-    const char* headerDecoHexcode()       { return h_excodes[headerHex]; }
-    const char* quotedTextHexcode(int i)  { return h_excodes[i]; }
-    void updateHexcodes();
+    QString headerDecoHexcode()       { return headerDecoColor().name(); }
+    QString quotedTextHexcode(int i);
 
     QFont articleFont();
     QFont articleFixedFont();
@@ -381,7 +380,6 @@
     QString c_olorNames[COL_CNT];
     QFont   f_onts[FNT_CNT];
     QString f_ontNames[FNT_CNT];
-    char    h_excodes[HEX_CNT][8];
     QPixmap i_cons[ICON_CNT];
 
 };
diff -urN -x CVS kdepim.orig/knode/knconfigwidgets.cpp kdepim/knode/knconfigwidgets.cpp
--- kdepim.orig/knode/knconfigwidgets.cpp	Wed Mar  3 13:15:58 2004
+++ kdepim/knode/knconfigwidgets.cpp	Tue Apr 20 13:58:17 2004
@@ -791,7 +791,6 @@
 
   d_ata->setDirty(true);
 
-  d_ata->updateHexcodes();
   d_ata->recreateLVIcons();
 }
 
diff -urN -x CVS kdepim.orig/knode/knconvert.cpp kdepim/knode/knconvert.cpp
--- kdepim.orig/knode/knconvert.cpp	Sun Aug 10 20:10:23 2003
+++ kdepim/knode/knconvert.cpp	Sun Apr 11 13:00:04 2004
@@ -27,12 +27,12 @@
 #include <klineedit.h>
 #include <kprocess.h>
 #include <kapplication.h>
+#include <kpushbutton.h>
 
 #include <kmime_util.h>
 
 #include "knconvert.h"
 #include "resource.h"
-#include <qpushbutton.h>
 
 
 bool KNConvert::needToConvert(const QString &oldVersion)
@@ -60,7 +60,7 @@
   s_tartBtn->setDefault(true);
   btnL->addStretch(1);
   btnL->addWidget(s_tartBtn);
-  c_ancelBtn=new QPushButton(i18n("Cancel"), this);
+  c_ancelBtn=new KPushButton(KStdGuiItem::cancel(), this);
   btnL->addWidget(c_ancelBtn);
 
   connect(s_tartBtn, SIGNAL(clicked()), this, SLOT(slotStart()));
diff -urN -x CVS kdepim.orig/knode/knmainwidget.cpp kdepim/knode/knmainwidget.cpp
--- kdepim.orig/knode/knmainwidget.cpp	Tue Mar 23 21:24:08 2004
+++ kdepim/knode/knmainwidget.cpp	Tue Apr 20 13:58:17 2004
@@ -21,7 +21,6 @@
 #include <kxmlguiclient.h>
 #include <kconfig.h>
 #include <kmessagebox.h>
-#include <kkeydialog.h>
 #include <kedittoolbar.h>
 #include <kstdaction.h>
 #include <kdebug.h>
@@ -662,10 +661,6 @@
                        "configure", 0, this,
                        SLOT(slotSettings()), actionCollection(),
                        "knode_configure_knode" );
-   (void) new KAction( i18n("Configure S&hortcuts..."),
-                       "configure_shortcuts", 0, this,
-                       SLOT(slotConfKeys()), actionCollection(),
-                       "knode_configure_shortcuts" );
 
   //collection-view - folder
   a_ctFolNew                = new KAction(i18n("&New Folder"), "folder_new", 0, this,
@@ -1012,7 +1007,6 @@
 
 void KNMainWidget::paletteChange( const QPalette & )
 {
-  c_fgManager->appearance()->updateHexcodes();
   KNArticleWidget::configChanged();
   configChanged();
 }
@@ -2267,14 +2261,6 @@
 }
 
 
-void KNMainWidget::slotConfKeys()
-{
-  //KKeyDialog::configureKeys(actionCollection(), xmlFile(), true, this);
-  KKeyDialog::configure( actionCollection(),
-			 true /*allow one-letter shortcuts*/
-    );
-}
-
 void KNMainWidget::slotSettings()
 {
   c_fgManager->configure();
diff -urN -x CVS kdepim.orig/knode/knmainwidget.h kdepim/knode/knmainwidget.h
--- kdepim.orig/knode/knmainwidget.h	Tue Jan  6 13:09:42 2004
+++ kdepim/knode/knmainwidget.h	Mon Apr 19 13:56:52 2004
@@ -396,7 +396,6 @@
   void slotSwitchToGroupView();
   void slotSwitchToHeaderView();
   void slotSwitchToArticleViewer();
-  void slotConfKeys();
   void slotSettings();
 
   //--------------------------- </Actions> -----------------------------
diff -urN -x CVS kdepim.orig/knode/knode.cpp kdepim/knode/knode.cpp
--- kdepim.orig/knode/knode.cpp	Fri Jan  2 23:31:02 2004
+++ kdepim/knode/knode.cpp	Mon Apr 19 13:56:52 2004
@@ -45,6 +45,8 @@
   KStdAction::quit(kapp, SLOT(closeAllWindows()), actionCollection());
   KStdAction::configureToolbars(this, SLOT(slotConfToolbar()),
                                 actionCollection(), "knode_configure_toolbars" );
+  KStdAction::keyBindings(this, SLOT(slotConfKeys()), 
+                          actionCollection(), "knode_configure_shortcuts");
 
   m_mainWidget = new KNMainWidget( this, true, this, 0 );
   connect( m_mainWidget, SIGNAL(signalCaptionChangeRequest(const QString&)),
@@ -82,6 +84,11 @@
   applyMainWindowSettings(knGlobals.config(),"mainWindow_options");
 }
 
+void KNMainWindow::slotConfKeys()
+{
+  KKeyDialog::configure(actionCollection(), true);
+}
+
 bool KNMainWindow::queryClose()
 {
   return m_mainWidget->queryClose();
diff -urN -x CVS kdepim.orig/knode/knode.h kdepim/knode/knode.h
--- kdepim.orig/knode/knode.h	Wed Jan  7 22:33:06 2004
+++ kdepim/knode/knode.h	Mon Apr 19 13:56:52 2004
@@ -46,6 +46,7 @@
 public slots:
   void slotConfToolbar();
   void slotNewToolbarConfig();
+  void slotConfKeys();
 protected:
   bool queryClose();
 private:
diff -urN -x CVS kdepim.orig/knode/knsearchdialog.cpp kdepim/knode/knsearchdialog.cpp
--- kdepim.orig/knode/knsearchdialog.cpp	Thu Oct 11 17:19:08 2001
+++ kdepim/knode/knsearchdialog.cpp	Sun Apr 11 13:00:04 2004
@@ -15,12 +15,13 @@
 */
 
 #include <qlayout.h>
-#include <qpushbutton.h>
 #include <qcheckbox.h>
 
 #include <klocale.h>
 #include <kapplication.h>
 #include <kiconloader.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include "knfilterconfigwidget.h"
 #include "knarticlefilter.h"
@@ -38,7 +39,7 @@
   startBtn=new QPushButton(i18n("Sta&rt Search"), bg);
   startBtn->setDefault(true); 
   newBtn=new QPushButton(i18n("&New Search"), bg);
-  closeBtn=new QPushButton(i18n("&Close"), bg);
+  closeBtn=new KPushButton(KStdGuiItem::close(), bg);
 
   completeThreads=new QCheckBox(i18n("Sho&w complete threads"),this);
   fcw=new KNFilterConfigWidget(this);
diff -urN -x CVS kdepim.orig/knotes/knote.cpp kdepim/knotes/knote.cpp
--- kdepim.orig/knotes/knote.cpp	Wed Mar 31 12:40:28 2004
+++ kdepim/knotes/knote.cpp	Sun Apr 11 13:00:05 2004
@@ -31,6 +31,7 @@
 #include <kapplication.h>
 #include <kdebug.h>
 #include <kaction.h>
+#include <kstdaction.h>
 #include <kcombobox.h>
 #include <ktoolbar.h>
 #include <kpopupmenu.h>
@@ -95,8 +96,7 @@
         this, SLOT(slotInsDate()), actionCollection(), "insert_date" );
     new KAction( i18n("Mail..."), "mail_send", 0,
         this, SLOT(slotMail()), actionCollection(), "mail_note" );
-    new KAction( i18n("Print..."), "fileprint", 0,
-        this, SLOT(slotPrint()), actionCollection(), "print_note" );
+    KStdAction::print( this, SLOT(slotPrint()), actionCollection(), "print_note" );
     new KAction( i18n("Preferences..."), "configure", 0,
         this, SLOT(slotPreferences()), actionCollection(), "configure_note" );
 
diff -urN -x CVS kdepim.orig/knotes/knoteedit.cpp kdepim/knotes/knoteedit.cpp
--- kdepim.orig/knotes/knoteedit.cpp	Wed Mar 31 12:40:28 2004
+++ kdepim/knotes/knoteedit.cpp	Sun Apr 11 13:00:05 2004
@@ -67,7 +67,7 @@
     connect( this, SIGNAL(copyAvailable(bool)), m_cut, SLOT(setEnabled(bool)) );
     connect( this, SIGNAL(copyAvailable(bool)), m_copy, SLOT(setEnabled(bool)) );
 
-    new KAction( i18n("Clear"), "editclear", 0, this, SLOT(clear()), actions, "edit_clear" );
+    new KAction( KStdGuiItem::clear(), 0, this, SLOT(clear()), actions, "edit_clear" );
     KStdAction::selectAll( this, SLOT(selectAll()), actions );
 
     // create the actions modifying the text format
diff -urN -x CVS kdepim.orig/kontact/plugins/korganizer/todoplugin.desktop kdepim/kontact/plugins/korganizer/todoplugin.desktop
--- kdepim.orig/kontact/plugins/korganizer/todoplugin.desktop	Mon Mar  1 14:53:32 2004
+++ kdepim/kontact/plugins/korganizer/todoplugin.desktop	Thu Apr 29 14:54:08 2004
@@ -70,7 +70,7 @@
 Name[hi]=टूडू लिस्ट
 Name[hu]=Feladatok
 Name[is]=Tossalisti
-Name[it]=Lista cose da-fare
+Name[it]=Cose da fare
 Name[ja]=Todo リスト
 Name[lt]=Planuojamų darbų sąrašas
 Name[nb]=Gjøreliste
diff -urN -x CVS kdepim.orig/kontact/src/kontact.setdlg kdepim/kontact/src/kontact.setdlg
--- kdepim.orig/kontact/src/kontact.setdlg	Sat Mar 27 14:14:07 2004
+++ kdepim/kontact/src/kontact.setdlg	Sat Apr 17 14:47:54 2004
@@ -96,6 +96,7 @@
 Name[fr]=Organiseur
 Name[gl]=Axenda
 Name[hi]=आर्गेनाइज़र
+Name[hr]=Rokovnik
 Name[hu]=Határidőnapló
 Name[is]=Skipuleggjari
 Name[ja]=オーガナイザ
diff -urN -x CVS kdepim.orig/korganizer/actionmanager.cpp kdepim/korganizer/actionmanager.cpp
--- kdepim.orig/korganizer/actionmanager.cpp	Thu Mar 25 15:04:09 2004
+++ kdepim/korganizer/actionmanager.cpp	Sun Apr 11 13:00:05 2004
@@ -200,19 +200,18 @@
     if ( mMainWindow->hasDocument() ) {
       new KAction( i18n("&New"), "filenew", CTRL+Key_N, this,
                    SLOT( file_new() ), mACollection, "korganizer_openNew" );
-      new KAction( i18n("&Open"), "fileopen", CTRL+Key_O, this,
-                   SLOT( file_open() ), mACollection, "korganizer_open" );
+      KStdAction::open( this, SLOT( file_open() ), mACollection, "korganizer_open" );
       mRecent = new KRecentFilesAction( i18n("Open &Recent"), 0, 0, this,
                                         SLOT( file_openRecent( const KURL & ) ),
                                         mACollection, "korganizer_openRecent" );
       new KAction( i18n("Re&vert"), "revert", 0, this,
                    SLOT( file_revert() ), mACollection, "korganizer_revert" );
-      new KAction( i18n("Save &As..."), "filesaveas", 0, this,
+      KStdAction::saveAs( this,
                    SLOT( file_saveas() ), mACollection, "korganizer_saveAs" );
-      new KAction( i18n("&Close"), "fileclose", CTRL+Key_W, this,
+      KStdAction::close( this,
                    SLOT( file_close() ), mACollection, "korganizer_close" );
     }
-    new KAction( i18n("&Save"), "filesave", CTRL+Key_S, this,
+    KStdAction::save( this,
                  SLOT( file_save() ), mACollection, "korganizer_save" );
   } else {
     KStdAction::openNew(this, SLOT(file_new()), mACollection);
@@ -269,16 +268,14 @@
 #endif
 
   if (mIsPart) {
-    new KAction(i18n("&Print..."), "fileprint", CTRL+Key_P, mCalendarView,
-                SLOT(print()), mACollection, "korganizer_print" );
+    KStdAction::print(mCalendarView, SLOT(print()), mACollection, "korganizer_print" );
   } else {
     KStdAction::print(mCalendarView, SLOT(print()), mACollection);
   }
 
 #if 1
   if (mIsPart) {
-    new KAction(i18n("Print Previe&w..."), "filequickprint", 0, mCalendarView,
-                SLOT(printPreview()), mACollection, "korganizer_quickprint" );
+    KStdAction::printPreview(mCalendarView, SLOT(printPreview()), mACollection, "korganizer_quickprint" );
   } else {
     KStdAction::printPreview(mCalendarView, SLOT(printPreview()),
                              mACollection);
@@ -452,7 +449,7 @@
   action->setEnabled(false);
   connect(mCalendarView,SIGNAL(groupEventsSelected(bool)),
           action,SLOT(setEnabled(bool)));
-  action = new KAction(i18n("Cancel"),0,
+  action = new KAction(KStdGuiItem::cancel(),0,
                        mCalendarView,SLOT(schedule_cancel()),
                        mACollection,"cancel");
   action->setEnabled(false);
diff -urN -x CVS kdepim.orig/korganizer/calendarview.cpp kdepim/korganizer/calendarview.cpp
--- kdepim.orig/korganizer/calendarview.cpp	Tue Nov 25 22:35:39 2003
+++ kdepim/korganizer/calendarview.cpp	Mon Apr 19 12:59:17 2004
@@ -561,7 +561,7 @@
 {
   if (dynamic_cast<KOMonthView*>(mViewManager->currentView() ) )
     mNavigator->selectNextWeek();
-  else 
+  else
     mNavigator->selectNext();
 }
 
@@ -569,7 +569,7 @@
 {
   if (dynamic_cast<KOMonthView*>(mViewManager->currentView() ) )
     mNavigator->selectPreviousWeek();
-  else 
+  else
     mNavigator->selectPrevious();
 }
 
@@ -711,7 +711,7 @@
 {
   return KMessageBox::warningContinueCancel(this,
       i18n("This item will be permanently deleted."),
-      i18n("KOrganizer Confirmation"),i18n("Delete"));
+      i18n("KOrganizer Confirmation"),KGuiItem(i18n("Delete"),"editdelete"));
 }
 
 
@@ -756,6 +756,8 @@
   } else {
     KNotifyClient::beep();
   }
+  
+  mTodoList->clearSelection();
 }
 
 void CalendarView::edit_paste()
@@ -807,6 +809,9 @@
 
   } else if ( pastedIncidence->type() == "Todo" ) {
     Todo* pastedTodo = static_cast<Todo*>(pastedIncidence);
+    Todo* _selectedTodo = selectedTodo();
+    if ( _selectedTodo )
+      pastedTodo->setRelatedTo( _selectedTodo );
     todoAdded( pastedTodo );
   }
 }
@@ -1485,9 +1490,9 @@
   createPrinter();
 
   KOrg::BaseView *currentView = mViewManager->currentView();
-  
+
   CalPrinter::PrintType printType = CalPrinter::Month;
-  
+
   if ( currentView ) printType = currentView->printType();
 
   DateList tmpDateList = mNavigator->selectedDates();
@@ -1707,7 +1712,7 @@
 {
   Incidence::List incidences = mCalendar->rawIncidences();
   Incidence::List::Iterator it;
-  
+
   for ( it = incidences.begin(); it != incidences.end(); it++ ) {
     (*it)->setOrganizer(KOPrefs::instance()->email());
     (*it)->recreate();
diff -urN -x CVS kdepim.orig/korganizer/calprinter.cpp kdepim/korganizer/calprinter.cpp
--- kdepim.orig/korganizer/calprinter.cpp	Wed Dec 10 16:13:49 2003
+++ kdepim/korganizer/calprinter.cpp	Sun Apr 11 13:00:05 2004
@@ -255,7 +255,7 @@
 void CalPrintDialog::setPreview(bool preview)
 {
 #if KDE_IS_VERSION( 3, 1, 93 )
-  setButtonOK( preview ? i18n("&Preview") : i18n("&Print...") );
+  setButtonOK( preview ? i18n("&Preview") : KStdGuiItem::print() );
 #else
   setButtonOKText( preview ? i18n("&Preview") : i18n("&Print...") );
 #endif
diff -urN -x CVS kdepim.orig/korganizer/configcolors.desktop kdepim/korganizer/configcolors.desktop
--- kdepim.orig/korganizer/configcolors.desktop	Sun Feb 29 16:50:33 2004
+++ kdepim/korganizer/configcolors.desktop	Wed Apr 28 14:39:34 2004
@@ -34,6 +34,7 @@
 Name[gl]=Cores
 Name[hi]=रंग
 Name[hr]=Boje
+Name[hsb]=Barby
 Name[hu]=Színek
 Name[id]=Warna
 Name[is]=Litir
@@ -82,7 +83,7 @@
 Comment[fi]=KOrganizer värien asetukset
 Comment[fr]=Configuration des couleurs de KOrganizer
 Comment[gl]=Configuración das Cores de Korganizer
-Comment[hi]=के-आर्गेनाइज़र रंग कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र रंग कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer színbeállítások
 Comment[is]=KSkipulag Litastillingar
 Comment[it]=Configurazione dei colori di Korganizer
diff -urN -x CVS kdepim.orig/korganizer/configfonts.desktop kdepim/korganizer/configfonts.desktop
--- kdepim.orig/korganizer/configfonts.desktop	Sun Feb 29 16:50:35 2004
+++ kdepim/korganizer/configfonts.desktop	Sun Apr 25 01:30:53 2004
@@ -77,7 +77,7 @@
 Comment[fi]=Korganizer kirjasinten asetukset
 Comment[fr]=Configuration des polices de KOrganizer
 Comment[gl]=Configuración de Fontes de Korganizer
-Comment[hi]=के-आर्गेनाइज़र फॉन्ट्स कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र फॉन्ट्स कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer betűtípus-beállítások
 Comment[is]=KSkipulag leturstillingar
 Comment[it]=Configurazione font di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/configfreebusy.desktop kdepim/korganizer/configfreebusy.desktop
--- kdepim.orig/korganizer/configfreebusy.desktop	Sun Feb 29 16:50:37 2004
+++ kdepim/korganizer/configfreebusy.desktop	Sun Apr 25 01:30:55 2004
@@ -62,7 +62,7 @@
 Comment[fi]=KOrganizer vapaa/varattu asetukset
 Comment[fr]=Configuration de la disponibilité pour KOrganizer
 Comment[gl]=Configuración de Dispoñibilidade/Indispoñibilidade de Korganizer
-Comment[hi]=के-आर्गेनाइज़र फ्री/बिज़ी कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र फ्री/बिज़ी कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer foglaltsági beállítások
 Comment[is]=KSkipulag Stillingar fyrir tímaráðstöfun
 Comment[it]=Configurazione Libero/Occupato di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/configgroupautomation.desktop kdepim/korganizer/configgroupautomation.desktop
--- kdepim.orig/korganizer/configgroupautomation.desktop	Sun Feb 29 16:50:39 2004
+++ kdepim/korganizer/configgroupautomation.desktop	Sun Apr 25 01:30:57 2004
@@ -61,7 +61,7 @@
 Comment[fi]=Korganizer ryhmän automatisoinnin asetukset
 Comment[fr]=Configuration de l'automatisation des groupes de KOrganizer
 Comment[gl]=Configuración de Automatización de Grupo de KOrganizer
-Comment[hi]=के-आर्गेनाइज़र फॉन्ट्स कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र फॉन्ट्स कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer csoportmunka-beállítások
 Comment[is]=KSkipulag Hópsjálfvirkni
 Comment[it]=Configurazione automazione gruppi di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/configgroupscheduling.desktop kdepim/korganizer/configgroupscheduling.desktop
--- kdepim.orig/korganizer/configgroupscheduling.desktop	Sun Feb 29 16:50:41 2004
+++ kdepim/korganizer/configgroupscheduling.desktop	Sun Apr 25 01:30:59 2004
@@ -62,7 +62,7 @@
 Comment[fi]=KOrganizer ryhmäajastuksen asetukset
 Comment[fr]=Configuration de la planification des groupes de KOrganizer
 Comment[gl]=Configuración de Horarios en Grupo de KOrganizer
-Comment[hi]=के-आर्गेनाइज़र समूह समय-सारणी कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र समूह समय-सारणी कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer szervezési beállítások
 Comment[is]=KSkipulag fyrir Hópáætlun
 Comment[it]=Configurazione del programmatore gruppi di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/configmain.desktop kdepim/korganizer/configmain.desktop
--- kdepim.orig/korganizer/configmain.desktop	Tue Mar 30 14:27:51 2004
+++ kdepim/korganizer/configmain.desktop	Sun Apr 25 01:31:01 2004
@@ -64,7 +64,7 @@
 Comment[fi]=KOrganizerin pääasetukset
 Comment[fr]=Configuration principale de KOrganizer
 Comment[gl]=Configuración Principal de KOrganizer
-Comment[hi]=के-आर्गेनाइज़र मुख्य कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र मुख्य कॉन्फ़िगरेशन
 Comment[hu]=A KOrganizer alapbeállításai
 Comment[is]=KSkipulag Aðalstillingar
 Comment[it]=Configurazione principale di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/configprinting.desktop kdepim/korganizer/configprinting.desktop
--- kdepim.orig/korganizer/configprinting.desktop	Sun Feb 29 16:50:45 2004
+++ kdepim/korganizer/configprinting.desktop	Sun Apr 25 01:31:03 2004
@@ -67,7 +67,7 @@
 Comment[fi]=KOrganizer tulostusasetukset
 Comment[fr]=Configuration de l'impression de KOrganizer
 Comment[gl]=Configuración de Impresión de KOrganizer
-Comment[hi]=के-आर्गेनाइज़र मुद्रण कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र मुद्रण कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer nyomtatási beállítások
 Comment[is]=KSkipulag Prentstillingar
 Comment[it]=KOrganizer configurazione stampa
diff -urN -x CVS kdepim.orig/korganizer/configtime.desktop kdepim/korganizer/configtime.desktop
--- kdepim.orig/korganizer/configtime.desktop	Sun Feb 29 16:50:47 2004
+++ kdepim/korganizer/configtime.desktop	Sun Apr 25 01:31:05 2004
@@ -67,7 +67,7 @@
 Comment[fi]=KOrganizerin aika-asetukset
 Comment[fr]=Configuration de l'heure de KOrganizer
 Comment[gl]=Configuración Horaria de KOrganizer
-Comment[hi]=के-आर्गेनाइज़र समय कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र समय कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer dátum- és időbeállítások
 Comment[is]=KSkipulag Tímastillignar
 Comment[it]=Configurazione data di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/configviews.desktop kdepim/korganizer/configviews.desktop
--- kdepim.orig/korganizer/configviews.desktop	Sun Feb 29 16:50:49 2004
+++ kdepim/korganizer/configviews.desktop	Sun Apr 25 01:31:07 2004
@@ -67,7 +67,7 @@
 Comment[fi]=KOrganizerin näkymäasetukset
 Comment[fr]=Configuration des vues de KOrganizer
 Comment[gl]=Configuración de Vista de KOrganizer
-Comment[hi]=के-आर्गेनाइज़र दृश्य कॉन्फिगरेशन
+Comment[hi]=के-आर्गेनाइज़र दृश्य कॉन्फ़िगरेशन
 Comment[hu]=KOrganizer nézeti beállítások
 Comment[is]=Stillinga birtingar fyrir KSkipulag
 Comment[it]=Configurazione viste di KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/history.cpp kdepim/korganizer/history.cpp
--- kdepim.orig/korganizer/history.cpp	Sun Sep  7 14:46:43 2003
+++ kdepim/korganizer/history.cpp	Wed Apr 14 21:57:00 2004
@@ -194,14 +194,16 @@
 void History::EntryEdit::undo()
 {
   Incidence *incidence = mCalendar->incidence( mNewIncidence->uid() );
-  mCalendar->deleteIncidence( incidence );
+  if( incidence )
+	  mCalendar->deleteIncidence( incidence );
   mCalendar->addIncidence( mOldIncidence->clone() );
 }
 
 void History::EntryEdit::redo()
 {
   Incidence *incidence = mCalendar->incidence( mOldIncidence->uid() );
-  mCalendar->deleteIncidence( incidence );
+  if( incidence )
+	  mCalendar->deleteIncidence( incidence );
   mCalendar->addIncidence( mNewIncidence->clone() );
 }
 
diff -urN -x CVS kdepim.orig/korganizer/knewstuff/engine.cpp kdepim/korganizer/knewstuff/engine.cpp
--- kdepim.orig/korganizer/knewstuff/engine.cpp	Sat Apr  3 00:32:03 2004
+++ kdepim/korganizer/knewstuff/engine.cpp	Sun Apr 11 13:00:05 2004
@@ -276,14 +276,14 @@
     } else {
       int result = KMessageBox::questionYesNo( mParentWidget, text, caption,
                                                i18n("Upload Info..."),
-                                               i18n("Close") );
+                                               KStdGuiItem::close() );
       if ( result == KMessageBox::Yes ) {
         kapp->invokeBrowser( noUploadUrl.url() );
       }
     }
   } else {
     int result = KMessageBox::questionYesNo( mParentWidget, text, caption,
-                                             i18n("Upload"), i18n("Cancel") );
+                                             i18n("Upload"), KStdGuiItem::cancel() );
     if ( result == KMessageBox::Yes ) {
       KURL destination = mUploadProvider->uploadUrl();
       destination.setFileName( fi.fileName() );
diff -urN -x CVS kdepim.orig/korganizer/koagenda.cpp kdepim/korganizer/koagenda.cpp
--- kdepim.orig/korganizer/koagenda.cpp	Sun Oct 26 14:48:49 2003
+++ kdepim/korganizer/koagenda.cpp	Mon Apr 19 12:58:54 2004
@@ -129,9 +129,10 @@
 
   mTimeBox->setText(KGlobal::locale()->formatTime(tim, KOPrefs::instance()->mMarcusBainsShowSeconds));
   mTimeBox->adjustSize();
+  if (y-mTimeBox->height()>=0) y-=mTimeBox->height(); else y++;
   agenda->moveChild(mTimeBox,
                     (int)(x+agenda->gridSpacingX()-mTimeBox->width()-1),
-                    y-mTimeBox->height());
+                    y);
   mTimeBox->raise();
   mTimeBox->setAutoMask(true);
 
diff -urN -x CVS kdepim.orig/korganizer/koeditorattachments.cpp kdepim/korganizer/koeditorattachments.cpp
--- kdepim.orig/korganizer/koeditorattachments.cpp	Sat Jan 10 03:38:17 2004
+++ kdepim/korganizer/koeditorattachments.cpp	Sun Apr 11 13:00:05 2004
@@ -111,7 +111,7 @@
 
   if ( KMessageBox::warningContinueCancel(this,
         i18n("This item will be permanently deleted."),
-        i18n("KOrganizer Confirmation"),i18n("Delete")) == KMessageBox::Continue )
+        i18n("KOrganizer Confirmation"),KGuiItem(i18n("Delete"),"editdelete")) == KMessageBox::Continue )
     delete item;
 }
 
diff -urN -x CVS kdepim.orig/korganizer/koeventeditor.cpp kdepim/korganizer/koeventeditor.cpp
--- kdepim.orig/korganizer/koeventeditor.cpp	Fri Feb 13 15:17:24 2004
+++ kdepim/korganizer/koeventeditor.cpp	Sun Apr 11 13:00:05 2004
@@ -204,7 +204,7 @@
                               const QString &attachment )
 {
   init();
-  
+
   mEvent = 0;
 
   loadDefaults();
@@ -372,7 +372,7 @@
 {
   return KMessageBox::warningContinueCancel(this,
       i18n("This item will be permanently deleted."),
-      i18n("KOrganizer Confirmation"),i18n("Delete"));
+      i18n("KOrganizer Confirmation"),KGuiItem(i18n("Delete"),"editdelete"));
 }
 
 void KOEventEditor::slotLoadTemplate()
diff -urN -x CVS kdepim.orig/korganizer/kotodoeditor.cpp kdepim/korganizer/kotodoeditor.cpp
--- kdepim.orig/korganizer/kotodoeditor.cpp	Fri Feb 13 15:17:24 2004
+++ kdepim/korganizer/kotodoeditor.cpp	Sun Apr 11 13:00:05 2004
@@ -41,7 +41,7 @@
 #include "koprefs.h"
 #include "koeditorattachments.h"
 #include "kodialogmanager.h"
- 
+
 #include "kotodoeditor.h"
 
 KOTodoEditor::KOTodoEditor( Calendar *calendar, QWidget *parent ) :
@@ -164,7 +164,7 @@
                               const QString &attachment )
 {
   init();
-  
+
   mTodo = 0;
 
   loadDefaults();
@@ -292,7 +292,7 @@
 {
   return KMessageBox::warningContinueCancel(this,
       i18n("This item will be permanently deleted."),
-      i18n("KOrganizer Confirmation"),i18n("Delete"));
+      i18n("KOrganizer Confirmation"),KGuiItem(i18n("Delete"),"editdelete"));
 }
 
 void KOTodoEditor::modified (int modification)
diff -urN -x CVS kdepim.orig/korganizer/plugindialog.cpp kdepim/korganizer/plugindialog.cpp
--- kdepim.orig/korganizer/plugindialog.cpp	Sun Jan  4 15:53:45 2004
+++ kdepim/korganizer/plugindialog.cpp	Sun Apr 11 13:00:05 2004
@@ -57,7 +57,7 @@
 */
 PluginDialog::PluginDialog( QWidget *parent )
   : KDialogBase( Plain, i18n("Configure Plugins"), Ok | Cancel | User1, Ok,
-                 parent, 0, false, false, i18n("Configure...") )
+                 parent, 0, false, false, KGuiItem( i18n("Configure..."), "configure") )
 {
   QFrame *topFrame = plainPage();
   QVBoxLayout *topLayout = new QVBoxLayout( topFrame, 0, spacingHint() );
diff -urN -x CVS kdepim.orig/korganizer/plugins/exchange/exchange.desktop kdepim/korganizer/plugins/exchange/exchange.desktop
--- kdepim.orig/korganizer/plugins/exchange/exchange.desktop	Sun Feb 29 16:51:07 2004
+++ kdepim/korganizer/plugins/exchange/exchange.desktop	Sat Apr 17 14:47:56 2004
@@ -20,6 +20,7 @@
 Name[gl]=Extensión de Microsoft Exchange 2000 para KOrganizer
 Name[he]=תוסף Microsoft Exchange 2000 ל-KOrganizer
 Name[hi]=के-आर्गेनाइज़र हेतु माइक्रोसॉफ्ट एक्सचेंज 2000  प्लगइन
+Name[hr]=Microsoft Exchange 2000 dodatak za KOrganizer
 Name[hu]=Microsoft Exchange 2000-bővítőmodul a KOrganizerhez
 Name[is]=Microsoft Exchange 2000 íforrit fyrir KSkipulag
 Name[it]=Plugin Microsoft Exchange 2000 per KOrganizer
diff -urN -x CVS kdepim.orig/korganizer/searchdialog.cpp kdepim/korganizer/searchdialog.cpp
--- kdepim.orig/korganizer/searchdialog.cpp	Sat Oct  4 18:04:43 2003
+++ kdepim/korganizer/searchdialog.cpp	Sun Apr 11 13:00:05 2004
@@ -42,7 +42,7 @@
 
 SearchDialog::SearchDialog(Calendar *calendar,QWidget *parent)
   : KDialogBase(Plain,i18n("Find Events"),User1|Close,User1,parent,0,false,false,
-                i18n("&Find"))
+                KGuiItem( i18n("&Find"), "find"))
 {
   mCalendar = calendar;
 
@@ -69,7 +69,7 @@
   mJournalsCheck = new QCheckBox( i18n("&Journal entries"), itemsGroup );
   mEventsCheck->setChecked( true );
   mTodosCheck->setChecked( true );
-  
+
   // Date range
   QGroupBox *rangeGroup = new QGroupBox( 1, Horizontal, i18n( "Date Range" ),
                                         topFrame );
@@ -81,7 +81,7 @@
   mStartDate = new KDateEdit( rangeWidget );
   rangeLayout->addWidget( new QLabel( mStartDate, i18n("&From:"), rangeWidget ) );
   rangeLayout->addWidget( mStartDate );
-  
+
   mEndDate = new KDateEdit( rangeWidget );
   rangeLayout->addWidget( new QLabel( mEndDate, i18n("&To:"), rangeWidget ) );
   mEndDate->setDate( QDate::currentDate().addDays( 365 ) );
@@ -101,7 +101,7 @@
   mSummaryCheck->setChecked( true );
   mDescriptionCheck = new QCheckBox( i18n("Desc&riptions"), subjectGroup );
   mCategoryCheck = new QCheckBox( i18n("Cate&gories"), subjectGroup );
-  
+
 
   // Results list view
   listView = new KOListView( mCalendar, topFrame );
@@ -177,28 +177,28 @@
 {
   QDate startDt = mStartDate->date();
   QDate endDt = mEndDate->date();
-  
+
   Event::List events = mCalendar->events( startDt, endDt,
                                           mInclusiveCheck->isChecked() );
   Todo::List todos;
-  if ( mIncludeUndatedTodos ) 
+  if ( mIncludeUndatedTodos )
     todos = mCalendar->todos();
   else {
-    QDate dt = startDt; 
+    QDate dt = startDt;
     while ( dt <= endDt ) {
       todos += mCalendar->todos( dt );
       dt = dt.addDays( 1 );
     }
   }
-  
-  Journal::List journals; 
-  QDate dt = startDt; 
+
+  Journal::List journals;
+  QDate dt = startDt;
   while ( dt <= endDt ) {
     Journal* j=mCalendar->journal( dt );
     if (j) journals.append( j );
     dt = dt.addDays( 1 );
   }
-  
+
   Incidence::List allIncidences = Calendar::mergeIncidenceList( events, todos, journals );
 
   mMatchedEvents.clear();
diff -urN -x CVS kdepim.orig/korganizer/statusdialog.cpp kdepim/korganizer/statusdialog.cpp
--- kdepim.orig/korganizer/statusdialog.cpp	Fri Mar  7 23:10:28 2003
+++ kdepim/korganizer/statusdialog.cpp	Sun Apr 11 13:00:05 2004
@@ -18,12 +18,13 @@
 */
 
 #include <qlabel.h>
-#include <qpushbutton.h>
 #include <qstringlist.h>
 #include <qlayout.h>
 
 #include <kdebug.h>
 #include <klocale.h>
+#include <kpushbutton.h>
+#include <kstdguiitem.h>
 
 #include "statusdialog.h"
 #include "statusdialog.moc"
@@ -48,11 +49,11 @@
 
   QBoxLayout *buttonLayout = new QHBoxLayout( topLayout );
 
-  QPushButton *ok = new QPushButton(i18n("&OK"), this);
+  QPushButton *ok = new KPushButton(KStdGuiItem::ok(), this);
   connect ( ok,SIGNAL(clicked()), this,SLOT(accept()) );
   buttonLayout->addWidget( ok );
 
-  QPushButton *cancel = new QPushButton(i18n("&Cancel"), this);
+  QPushButton *cancel = new KPushButton(KStdGuiItem::cancel(), this);
   connect ( cancel,SIGNAL(clicked()), this,SLOT(reject()) );
   buttonLayout->addWidget( cancel );
 }
diff -urN -x CVS kdepim.orig/korn/shell.cpp kdepim/korn/shell.cpp
--- kdepim.orig/korn/shell.cpp	Fri Feb 20 18:50:56 2004
+++ kdepim/korn/shell.cpp	Wed Apr 14 18:25:33 2004
@@ -1,7 +1,7 @@
 /*
 * shell.cpp -- Implementation of class KornShell.
 * Author:	Sirtaj Singh Kang
-* Version:	$Id$
+* Version:	$Id$
 * Generated:	Sun May  3 10:30:24 EST 1998
 */
 
@@ -382,7 +382,7 @@
 		i18n( "You do not appear to have used KOrn before.\n"
 		"Would you like a basic configuration created for you?" ),
 		i18n("Welcome to KOrn"),
-		i18n("Yes"), i18n( "No, Exit" ));
+		KStdGuiItem::yes(), i18n( "No, Exit" ));
 
 	if( status != 0 ) {
 		return false;
diff -urN -x CVS kdepim.orig/kpilot/conduits/docconduit/doc_conduit.desktop kdepim/kpilot/conduits/docconduit/doc_conduit.desktop
--- kdepim.orig/kpilot/conduits/docconduit/doc_conduit.desktop	Sun Feb 29 16:51:23 2004
+++ kdepim/kpilot/conduits/docconduit/doc_conduit.desktop	Sun Apr 25 01:31:09 2004
@@ -14,7 +14,7 @@
 Comment[fi]=Lisää tekstitiedostoja Pilottiisi, käyttökelpoinen DOC-lukijoille.
 Comment[fr]=Ajoute des fichiers texte à votre Palm, approprié pour les lecteurs de DOC.
 Comment[gl]=Engade ficheiros de texto ó seu Pilot,axeitado para lectores DOC.
-Comment[hi]=आपके पायलट में पाठ फाइलें जोड़ता है, डॉक पाठकों हेतु उपयुक्त.
+Comment[hi]=आपके पायलट में पाठ फ़ाइलें जोड़ता है, डॉक पाठकों हेतु उपयुक्त.
 Comment[hu]=Szöveges fájlok hozzáadása Pilot kéziszámítógépes DOC-olvasókhoz.
 Comment[is]=Setur texta í lófatölvuna þína, sem hægt er að lesa í DOC lesara.
 Comment[it]=Aggiunge fille di testo al tuo Pilot, adatti per lettori DOC.
diff -urN -x CVS kdepim.orig/kpilot/conduits/docconduit/kpalmdoc.desktop kdepim/kpilot/conduits/docconduit/kpalmdoc.desktop
--- kdepim.orig/kpilot/conduits/docconduit/kpalmdoc.desktop	Sun Feb 29 16:51:25 2004
+++ kdepim/kpilot/conduits/docconduit/kpalmdoc.desktop	Sun Apr 18 14:40:41 2004
@@ -19,6 +19,7 @@
 GenericName[et]=PalmDOC konverter
 GenericName[fi]=PalmDOC-muunnin
 GenericName[fr]=Convertisseur PalmDOC
+GenericName[gl]=Convertedor PalmDOC
 GenericName[hi]=पॉम-डॉक परिवर्तक
 GenericName[hu]=PalmDOC-konverter
 GenericName[is]=PalmDOC umbreytir
diff -urN -x CVS kdepim.orig/kpilot/conduits/popmail/passworddialog.cc kdepim/kpilot/conduits/popmail/passworddialog.cc
--- kdepim.orig/kpilot/conduits/popmail/passworddialog.cc	Sat Mar  8 02:43:05 2003
+++ kdepim/kpilot/conduits/popmail/passworddialog.cc	Sun Apr 11 13:00:05 2004
@@ -36,7 +36,7 @@
 // make up a binary distribution.
 //
 //
-static const char *passworddialog_id="$Id$";
+static const char *passworddialog_id="$Id$";
 
 
 
@@ -46,7 +46,6 @@
 #include <stdio.h>
 
 #include <qapplication.h>
-#include <qpushbutton.h>
 #include <qlabel.h>
 #include <qlayout.h>
 #include <qdialog.h>
@@ -59,6 +58,8 @@
 
 #include <klocale.h>
 #include <kapplication.h>
+#include <kstdguiitem.h>
+#include <kpushbutton.h>
 
 PasswordDialog::PasswordDialog(QString head, QWidget* parent, const char* name, bool modal, WFlags wflags)
    : QDialog(parent, name, modal, wflags)
@@ -108,10 +109,10 @@
    // Die Buttons "Ok" & "Cancel" erzeugen
    //
    QPushButton *b1, *b2;
-   b1 = new QPushButton(i18n("OK"), this);
+   b1 = new KPushButton(KStdGuiItem::ok(), this);
    b1->setGeometry( 10, 90, 80, 30 );
    
-   b2 = new QPushButton(i18n("Cancel"), this);
+   b2 = new KPushButton(KStdGuiItem::cancel(), this);
    b2->setGeometry( 110, 90, 80, 30 );
    
    // Buttons mit Funktionaliataet belegen
diff -urN -x CVS kdepim.orig/kpilot/conduits/sysinfoconduit/sysinfo_conduit.desktop kdepim/kpilot/conduits/sysinfoconduit/sysinfo_conduit.desktop
--- kdepim.orig/kpilot/conduits/sysinfoconduit/sysinfo_conduit.desktop	Sat Mar 27 14:14:22 2004
+++ kdepim/kpilot/conduits/sysinfoconduit/sysinfo_conduit.desktop	Sun Apr 25 01:31:12 2004
@@ -14,8 +14,9 @@
 Comment[fi]=Tämä yhdyskäytävä kirjoittaa tietoja mobiililaitteestasi ja synkronoi ne tiedostoon.
 Comment[fr]=Ce canal écrit des informations sur votre périphérique et la synchronisation dans un fichier.
 Comment[gl]=Este conducto escribe información sobre a súa axenda electrónica e a sincronización a un ficheiro.
-Comment[hi]=यह कन्ड्यूइट आपके हैंण्डहेल्ड के बारे में जानकारी लिखता है तथा एक फाइल में सिंक करता है.
+Comment[hi]=यह कन्ड्यूइट आपके हैंण्डहेल्ड के बारे में जानकारी लिखता है तथा एक फ़ाइल में सिंक करता है.
 Comment[hu]=Ezzel a csatolóval fájlba lehet kiíratni a kéziszámítógép és a szinkronizálás jellemzőit
+Comment[is]=Þessi rás skrifar upplýsingar um lófatölfuna þína og samstillinguna í skrá
 Comment[it]=QUesto conduit scrive in un file informazioni sul tuo palmare.
 Comment[ja]=本コンジットはコメント かつ to aをファイルに書き出し、 同期します。
 Comment[lt]=Šis kanalas rašo informaciją apie jūsų delninuką ir sinchronizaciją į bylą.
diff -urN -x CVS kdepim.orig/kpilot/lib/syncAction.cc kdepim/kpilot/lib/syncAction.cc
--- kdepim.orig/kpilot/lib/syncAction.cc	Tue Mar 11 00:37:01 2003
+++ kdepim/kpilot/lib/syncAction.cc	Wed Apr 14 18:25:34 2004
@@ -26,7 +26,7 @@
 ** Bug reports and questions can be sent to kde-pim@kde.org
 */
 static const char *syncAction_id =
-	"$Id$";
+	"$Id$";
 
 #include "options.h"
 
@@ -217,7 +217,7 @@
 		KDialogBase::Yes | KDialogBase::No,
 		KDialogBase::Yes, KDialogBase::No,
 		fParent, "questionYesNo", true, true,
-		i18n("Yes"), i18n("No"));
+		KStdGuiItem::yes(), KStdGuiItem::no());
 
 	// The following code is taken from KDialogBase.cc,
 	// part of the KDE 2.2 libraries. Copyright 2001
diff -urN -x CVS kdepim.orig/ktnef/gui/ktnef.desktop kdepim/ktnef/gui/ktnef.desktop
--- kdepim.orig/ktnef/gui/ktnef.desktop	Sun Feb 29 16:51:49 2004
+++ kdepim/ktnef/gui/ktnef.desktop	Sun Apr 25 01:31:14 2004
@@ -12,7 +12,8 @@
 Name[fi]=TNEF-tiedostonäyttäjä
 Name[fr]=Afficheur de fichiers TNEF
 Name[gl]=Visor de Ficheiros TNEF
-Name[hi]=टीएनईएफ फाइल प्रदर्शक
+Name[hi]=टीएनईएफ फ़ाइल प्रदर्शक
+Name[hr]=Preglednik TNEF datoteka
 Name[hu]=TNEF-fájlnézegető
 Name[is]=TNEF Skráarsýnir
 Name[it]=Visualizzatore file TNEF
@@ -53,7 +54,8 @@
 Comment[fi]=Näyttäjä/purkaja TNEF-tiedostoille
 Comment[fr]=Pour afficher / extraire des fichiers TNEF
 Comment[gl]=Un visor/extractor para ficheiros TNEF
-Comment[hi]=टीएनईएफ फाइलों हेतु एक प्रदर्शक/एक्सट्रेक्टर
+Comment[hi]=टीएनईएफ फ़ाइलों हेतु एक प्रदर्शक/एक्सट्रेक्टर
+Comment[hr]=Prikazivač/izdvajač za TNEF datoteke
 Comment[hu]=Nézegetőprogram TNEF-fájlokhoz
 Comment[is]=Sýnir/lesari fyrir TNEF skrár
 Comment[it]=Un visualizzatore/estrattore di file TNEF
diff -urN -x CVS kdepim.orig/ktnef/gui/ms-tnef.desktop kdepim/ktnef/gui/ms-tnef.desktop
--- kdepim.orig/ktnef/gui/ms-tnef.desktop	Sun Feb 29 16:51:51 2004
+++ kdepim/ktnef/gui/ms-tnef.desktop	Sun Apr 25 01:31:16 2004
@@ -17,7 +17,8 @@
 Comment[fi]=TNEF-tiedosto
 Comment[fr]=Fichier TNEF
 Comment[gl]=Ficheiro TNEF
-Comment[hi]=टीएनईएफ फाइल
+Comment[hi]=टीएनईएफ फ़ाइल
+Comment[hr]=TNEF datoteka
 Comment[hu]=TNEF-fájl
 Comment[is]=TNEF skrá
 Comment[it]=File TNEF
diff -urN -x CVS kdepim.orig/libkcal/icalformatimpl.cpp kdepim/libkcal/icalformatimpl.cpp
--- kdepim.orig/libkcal/icalformatimpl.cpp	Sat Feb  7 01:16:10 2004
+++ kdepim/libkcal/icalformatimpl.cpp	Wed Apr 14 00:32:52 2004
@@ -987,7 +987,7 @@
         icalcomponent_add_property(a,p);
       }
       icalcomponent_add_property(a,icalproperty_new_summary(alarm->mailSubject().utf8()));
-      icalcomponent_add_property(a,icalproperty_new_description(alarm->text().utf8()));
+      icalcomponent_add_property(a,icalproperty_new_description(alarm->mailText().utf8()));
       QStringList attachments = alarm->mailAttachments();
       if (attachments.count() > 0) {
         for (QStringList::Iterator at = attachments.begin();  at != attachments.end();  ++at) {
diff -urN -x CVS kdepim.orig/libkcal/local.desktop kdepim/libkcal/local.desktop
--- kdepim.orig/libkcal/local.desktop	Sun Feb 29 16:51:57 2004
+++ kdepim/libkcal/local.desktop	Sun Apr 25 01:31:18 2004
@@ -15,7 +15,7 @@
 Name[fi]=Kalenteri paikallisessa tiedostossa
 Name[fr]=Calendrier dans un fichier local
 Name[gl]=Calendario en Ficheiro Local
-Name[hi]=स्थानीय फाइल में तिथि पत्री
+Name[hi]=स्थानीय फ़ाइल में तिथि पत्री
 Name[hu]=Helyi fájlban tárolt naptár
 Name[is]=Dagatal í staðbundinni skrá
 Name[it]=Calendario in file locale
diff -urN -x CVS kdepim.orig/libkcal/remote.desktop kdepim/libkcal/remote.desktop
--- kdepim.orig/libkcal/remote.desktop	Sun Feb 29 16:52:01 2004
+++ kdepim/libkcal/remote.desktop	Sun Apr 25 01:31:20 2004
@@ -14,7 +14,7 @@
 Name[fi]=Kalenteri etätiedostossa
 Name[fr]=Calendrier dans un fichier distant
 Name[gl]=Calendario en Ficheiro Remoto
-Name[hi]=रिमोट फाइल में तिथि पत्री
+Name[hi]=रिमोट फ़ाइल में तिथि पत्री
 Name[hu]=Távoli fájlban tárolt naptár
 Name[is]=Dagatal í nettengdri möppu
 Name[it]=Calendario in file remoto
diff -urN -x CVS kdepim.orig/libkdenetwork/kscoringeditor.cpp kdepim/libkdenetwork/kscoringeditor.cpp
--- kdepim.orig/libkdenetwork/kscoringeditor.cpp	Mon Jul 21 17:45:48 2003
+++ kdepim/libkdenetwork/kscoringeditor.cpp	Wed Apr 14 12:28:04 2004
@@ -519,6 +519,8 @@
 void RuleEditWidget::slotAddGroup()
 {
   QString grp = groupsBox->currentText();
+  if( grp.isEmpty())
+	  return;
   QString txt = groupsEdit->text().stripWhiteSpace();
   if (txt == ".*") groupsEdit->setText(grp);
   else groupsEdit->setText(txt + ";" + grp);
diff -urN -x CVS kdepim.orig/libkdepim/kdateedit.cpp kdepim/libkdepim/kdateedit.cpp
--- kdepim.orig/libkdepim/kdateedit.cpp	Sun Mar 14 11:20:14 2004
+++ kdepim/libkdepim/kdateedit.cpp	Wed Apr 28 13:21:49 2004
@@ -162,15 +162,20 @@
   QRect desk = KGlobalSettings::desktopGeometry(this);
 
   QPoint popupPoint = mapToGlobal( QPoint( 0,0 ) );
-  if ( popupPoint.x() < desk.left() ) popupPoint.setX( desk.x() );
 
   int dateFrameHeight = mDateFrame->sizeHint().height();
-
   if ( popupPoint.y() + height() + dateFrameHeight > desk.bottom() ) {
     popupPoint.setY( popupPoint.y() - dateFrameHeight );
   } else {
     popupPoint.setY( popupPoint.y() + height() );
   }
+  int dateFrameWidth = mDateFrame->sizeHint().width();
+  if ( popupPoint.x() + dateFrameWidth > desk.right() ) {
+    popupPoint.setX( desk.right() - dateFrameWidth );
+  }
+
+  if ( popupPoint.x() < desk.left() ) popupPoint.setX( desk.left() );
+  if ( popupPoint.y() < desk.top() ) popupPoint.setY( desk.top() );
 
   mDateFrame->move( popupPoint );
 
diff -urN -x CVS kdepim.orig/libkdepim/kprefsdialog.cpp kdepim/libkdepim/kprefsdialog.cpp
--- kdepim.orig/libkdepim/kprefsdialog.cpp	Sun Nov  9 02:13:04 2003
+++ kdepim/libkdepim/kprefsdialog.cpp	Wed Apr 14 18:25:34 2004
@@ -57,7 +57,7 @@
   if ( stringItem ) {
     return new KPrefsWidString( stringItem, parent );
   }
-  
+
   KConfigSkeleton::ItemEnum *enumItem =
       dynamic_cast<KConfigSkeleton::ItemEnum *>( item );
   if ( enumItem ) {
@@ -74,13 +74,13 @@
       return radios;
     }
   }
-  
+
   KConfigSkeleton::ItemInt *intItem =
       dynamic_cast<KConfigSkeleton::ItemInt *>( item );
   if ( intItem ) {
     return new KPrefsWidInt( intItem, parent );
   }
-  
+
   return 0;
 }
 
@@ -526,11 +526,11 @@
 void KPrefsDialog::autoCreate()
 {
   KConfigSkeletonItem::List items = prefs()->items();
-  
+
   QMap<QString,QWidget *> mGroupPages;
   QMap<QString,QGridLayout *> mGroupLayouts;
   QMap<QString,int> mCurrentRows;
-  
+
   KConfigSkeletonItem::List::ConstIterator it;
   for( it = items.begin(); it != items.end(); ++it ) {
     QString group = (*it)->group();
@@ -555,7 +555,7 @@
     }
 
     KPrefsWid *wid = KPrefsWidFactory::create( *it, page );
-    
+
     if ( wid ) {
       QValueList<QWidget *> widgets = wid->widgets();
       if ( widgets.count() == 1 ) {
@@ -563,24 +563,24 @@
                                     currentRow, currentRow, 0, 1 );
       } else if ( widgets.count() == 2 ) {
         layout->addWidget( widgets[ 0 ], currentRow, 0 );
-        layout->addWidget( widgets[ 1 ], currentRow, 1 );      
+        layout->addWidget( widgets[ 1 ], currentRow, 1 );
       } else {
         kdError() << "More widgets than expected: " << widgets.count() << endl;
       }
-  
+
       if ( (*it)->isImmutable() ) {
         QValueList<QWidget *>::Iterator it2;
         for( it2 = widgets.begin(); it2 != widgets.end(); ++it2 ) {
           (*it2)->setEnabled( false );
         }
       }
-  
+
       addWid( wid );
-  
+
       mCurrentRows.replace( group, ++currentRow );
     }
   }
-  
+
   readConfig();
 }
 
@@ -625,7 +625,7 @@
   if (KMessageBox::warningContinueCancel(this,
       i18n("You are about to set all preferences to default values. All "
       "custom modifications will be lost."),i18n("Setting Default Preferences"),
-      i18n("Continue"))
+      KStdGuiItem::cont())
     == KMessageBox::Continue) setDefaults();
 }
 
