Index: akregator/src/akregator.desktop
===================================================================
--- akregator/src/akregator.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ akregator/src/akregator.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -58,7 +58,6 @@
 Comment[fi]=RSS-syötelukija
 Comment[fr]=Un lecteur de flux RSS pour KDE
 Comment[ga]=Comhbhailitheoir RSS le haghaidh KDE
-Comment[he]==קורא חדשות (RSS) עבור KDE
 Comment[hu]=KDE-s hírolvasó RSS hírcsatornákhoz
 Comment[is]=RSS fréttaforrit fyrir KDE
 Comment[it]=Un concentratore KDE per RSS
Index: kresources/egroupware/xmlrpciface.cpp
===================================================================
--- kresources/egroupware/xmlrpciface.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kresources/egroupware/xmlrpciface.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -200,7 +200,14 @@
   {
       case QVariant::String:
       case QVariant::CString:
-      return "<value><string>" + arg.toString() + "</string></value>\r\n";
+        {
+        QString result = arg.toString();
+        result = result.replace( "&", "&amp;" );
+        result = result.replace( "\"", "&quot;" );
+        result = result.replace( "<", "&lt;" );
+        result = result.replace( ">", "&gt;" );
+        return "<value><string>" + result + "</string></value>\r\n";
+        }
       case QVariant::Int:
       return "<value><int>" + QString::number( arg.toInt() ) + "</int></value>\r\n";
       case QVariant::Double:
Index: kresources/groupwise/soap/types.xsd
===================================================================
--- kresources/groupwise/soap/types.xsd	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kresources/groupwise/soap/types.xsd	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -524,7 +524,7 @@
 					<xs:element name="officialVersion" type="xs:unsignedInt" minOccurs="0"/>
 					<xs:element name="currentVersion" type="xs:unsignedInt" minOccurs="0"/>
 					<xs:element name="versionNumber" type="xs:unsignedInt" minOccurs="0"/>
-					<xs:element name="versionDescription" minOccurs="0"/>
+					<xs:element name="versionDescription" type="xs:string" minOccurs="0"/>
 					<xs:element name="fileSize" type="xs:unsignedInt" minOccurs="0"/>
 					<xs:element name="acl" type="tns:AccessControlList" minOccurs="0"/>
 				</xs:sequence>
Index: kmail/kmreaderwin.cpp
===================================================================
--- kmail/kmreaderwin.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kmail/kmreaderwin.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -1103,6 +1103,13 @@
           break;
         }
       }
+      if ( i == encodings.size() ) {
+        // the value of encoding is unknown => use Auto
+        kdWarning(5006) << "Unknown override character encoding \"" << encoding
+                        << "\". Using Auto instead." << endl;
+        mSelectEncodingAction->setCurrentItem( 0 );
+        mOverrideEncoding = QString::null;
+      }
     }
   }
   update( true );
Index: kmail/kmmessage.cpp
===================================================================
--- kmail/kmmessage.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kmail/kmmessage.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -706,6 +706,8 @@
                                            const QTextCodec*& codec,
                                            bool& isHTML ) const
 {
+  if ( !root ) return;
+
   isHTML = false;
   // initialy parse the complete message to decrypt any encrypted parts
   {
@@ -3819,7 +3821,7 @@
 //-----------------------------------------------------------------------------
 QCString KMMessage::charset() const
 {
-  if ( mMsg->Headers().HasContentType() ) {  
+  if ( mMsg->Headers().HasContentType() ) {
     DwMediaType &mType=mMsg->Headers().ContentType();
     mType.Parse();
     DwParameter *param=mType.FirstParameter();
Index: kmail/configuredialog.cpp
===================================================================
--- kmail/configuredialog.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kmail/configuredialog.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -2309,6 +2309,13 @@
     }
     i++;
   }
+  if ( i == encodings.size() ) {
+    // the current value of overrideCharacterEncoding is an unknown encoding => reset to Auto
+    kdWarning(5006) << "Unknown override character encoding \"" << currentOverrideEncoding
+                    << "\". Resetting to Auto." << endl;
+    mOverrideCharsetCombo->setCurrentItem( 0 );
+    GlobalSettings::self()->setOverrideCharacterEncoding( QString::null );
+  }
 }
 
 void AppearancePage::ReaderTab::doLoadFromGlobalSettings()
Index: kmail/configure.in.in
===================================================================
--- kmail/configure.in.in	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kmail/configure.in.in	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -6,8 +6,9 @@
 
 missing_indexlib=ignore
 AC_ARG_ENABLE(indexlib,
-	AC_HELP_STRING([--enable-indexlib],[Enable full-text indexing in KMail.]),
-	AC_DEFINE(HAVE_INDEXLIB,1,[Define if you want full-text indexing in KMail])
+	AC_HELP_STRING([--enable-indexlib],[Enable EXPERIMENTAL full-text indexing in KMail.
+ Use are your own risk!]),
+	AC_DEFINE(HAVE_INDEXLIB,1,[Define if you want EXPERIMENTAL full-text indexing in KMail])
 	missing_indexlib=no
 	)
 
Index: kaddressbook/xxport/kde2_xxport.desktop
===================================================================
--- kaddressbook/xxport/kde2_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/kde2_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -65,7 +65,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចូល​សៀវភៅ​អាសយដ្ឋាន KDE 2 ចាស់ៗ
 Comment[lt]=Priedas senosios KDE 2 adresų knygelės importui
 Comment[ms]=Plug masuk untuk import buku alamat lama KDE 2
-Comment[nb]=Tilleggsprogram for å importere adressebok fra KDE2
+Comment[nb]=Programtillegg for å importere adressebok fra KDE2
 Comment[nds]=Moduul för't Importeren vun KDE2-Adressböker
 Comment[nl]=Plugin voor het importeren van het oude KDE 2-adresboek
 Comment[nn]=Programtillegg for å importera den gamle KDE 2 adresseboka
Index: kaddressbook/xxport/csv_xxport.desktop
===================================================================
--- kaddressbook/xxport/csv_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/csv_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -64,7 +64,7 @@
 Comment[km]=កម្មវិធី ជំនួយ​ដើម្បី​នាំចូល និង នាំចេញ​ទំនាក់ទំនង​ក្នុង​ទ្រង់ទ្រាយ​ជា CSV ។
 Comment[lt]=Priedas, skirtas kontaktų eksportui ir importui CSV formatu
 Comment[ms]=Plug masuk untuk import dan eksport alamat perhubungan di dalam format CSV
-Comment[nb]=Tilleggsprogram for import/eksport av kontakter i CSV-format
+Comment[nb]=Programtillegg for import/eksport av kontakter i CSV-format
 Comment[nds]=Moduul för't Im- un Exporteren vun Kontakten in't CSV-Formaat
 Comment[nl]=Plugin voor het importeren en exporteren van contactpersonen in CSV-formaat
 Comment[nn]=Programtillegg for å importera og eksportera kontaktar i CSV-format
Index: kaddressbook/xxport/vcard_xxport.desktop
===================================================================
--- kaddressbook/xxport/vcard_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/vcard_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -67,7 +67,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចូល និង នាំចេញ​ទំនាក់ទំនង​ក្នុង​ទ្រង់ទ្រាយ​ជា vCard
 Comment[lt]=Įskiepis kontaktų importavimui ir eksportavimui vCard formatu
 Comment[ms]=Plugin untuk mengimpot dan mengekspot orang hubungan dalam format vCard 
-Comment[nb]=Tilleggsprogram for import og eksport av kontakter i vCard-format
+Comment[nb]=Programtillegg for import og eksport av kontakter i vCard-format
 Comment[nds]=Moduul för't Im-/Exporteren vun Kontakten in't vCard-Formaat
 Comment[nl]=Plugin voor het importeren en exporteren van contactpersonen in vCard-formaat
 Comment[nn]=Programtillegg for å importera og eksportera kontaktar i vCard-format
Index: kaddressbook/xxport/pab_xxport.desktop
===================================================================
--- kaddressbook/xxport/pab_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/pab_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -22,7 +22,7 @@
 Name[km]=កម្មវិធី​ជំនួយ KAB MS Exchange Personal Addressbook XXPort
 Name[lt]=KAB MS Exchange asmeninės adresų knygelės XXPort priedas
 Name[ms]=Plugin Buku Alamat Peribadi KAB MS Exchange  XXPort Plugin
-Name[nb]=KAB-tilleggsprogram for import fra MS Exchange
+Name[nb]=KAB-programtillegg for import fra MS Exchange
 Name[nds]=MSExchange-Importmoduul för KAdressbook
 Name[nl]=KAB MS Exchange Personal Addressbook XXPort-plugin
 Name[nn]=KAB MS Exchange personleg adressebok XXPort programtillegg
@@ -64,7 +64,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចូល​សៀវភៅ​អាសយដ្ឋាន​ផ្ទាល់​ខ្លួន​របស់ MS Exchange
 Comment[lt]=Priedas leidžiantis importuoti MS Exchange asmenines adresų knygeles
 Comment[ms]=Plugin untuk mengimpot Buku Alamat Peribadi MS Exchange 
-Comment[nb]=Tilleggsprogram for import av personlige adressebøker fra MS Exchange
+Comment[nb]=Programtillegg for import av personlige adressebøker fra MS Exchange
 Comment[nds]=Moduul för't Importeren vun persöönliche Adressböker ut MS Exchange
 Comment[nl]=Plugin voor het importeren van MS Exchange-adresboeken
 Comment[nn]=Programtillegg for å imortera MS Exchange personleg adressebok
Index: kaddressbook/xxport/opera_xxport.desktop
===================================================================
--- kaddressbook/xxport/opera_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/opera_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -65,7 +65,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចូល​ទំនាក់ទំនង​របស់ Opera
 Comment[lt]=Priedas Opera kontaktų importui
 Comment[ms]=Plug masuk untuk import alamat perhubungan Opera
-Comment[nb]=Tilleggsprogram for å importere Opera-kontakter
+Comment[nb]=Programtillegg for å importere Opera-kontakter
 Comment[nds]=Modul för't Importeren vun Opera-Kontakten
 Comment[nl]=Plugin voor het importeren an Opera's contactpersonen
 Comment[nn]=Programtillegg for å imortera Opera-kontaktar
Index: kaddressbook/xxport/gnokii_xxport.desktop
===================================================================
--- kaddressbook/xxport/gnokii_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/gnokii_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -23,7 +23,7 @@
 Name[km]=កម្មវិធី​ជំនួយ KAB Mobile Phone XXPort
 Name[lt]=KAB mobilaus telefono XXPort priedas
 Name[ms]=Plug masuk KAB Fon Mudah Alih XXPort
-Name[nb]=KAB-tilleggsprogram for mobiltelefon
+Name[nb]=KAB-programtillegg for mobiltelefon
 Name[nds]=Mobiltelefoon-Im-/Exportmoduul för KAdressbook
 Name[nl]=KAB Mobiele Telefoon XXPort-plugin
 Name[nn]=KAB Mobiltelefon XXPort programtillegg
@@ -64,7 +64,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ទូរស័ព្ទ​ចល័ត​ដើម្បី​នាំចូល និង នាំចេញ​ធាតុ​សៀវភៅ​អាសយដ្ឋាន
 Comment[lt]=Priedas skirtas importuoti ir eksportuoti adresų knygelės įrašus į mobiliuosius telefonus
 Comment[ms]=Plug masuk Fon Mudah Alih untuk Import dan Eksport Input Buku Alamat
-Comment[nb]=Tilleggsprogram for import/eksport av adressebok fra/til mobiltelefon
+Comment[nb]=Programtillegg for import/eksport av adressebok fra/til mobiltelefon
 Comment[nds]=Mobiltelefoon-Moduul för't Im- un Exporteren vun KAdressbook-Indrääg
 Comment[nl]=Mobiele-telefoon-plugin voor het importeren en exporteren van adresboekitems
 Comment[nn]=Programtillegg for å importera og eksportera adressebokoppføringar i mobiltelefon
Index: kaddressbook/xxport/bookmark_xxport.desktop
===================================================================
--- kaddressbook/xxport/bookmark_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/bookmark_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -64,7 +64,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចេញ​អាសយដ្ឋាន​បណ្ដាញ​របស់​ទំនាក់ទំនង ជា​ចំណាំ
 Comment[lt]=Priedas skirtas žiniatinklio adresų kontaktuose eksportavimui į žymeles
 Comment[ms]= Plug masuk untuk eksport alamat web untuk perhubungan sebagai tanda laman
-Comment[nb]=Tilleggsprogram som eksporterer kontaktenes nett-addresser som bokmerker
+Comment[nb]=Programtillegg som eksporterer kontaktenes nett-addresser som bokmerker
 Comment[nds]=Moduul för't Exporteren vun Kontakt-Nettadressen as Leestekens
 Comment[nl]=Plugin voor het exporteren van de webadressen van de contactpersonen als bladwijzers
 Comment[nn]=Programtillegg for å eksportera nettadresser av kontaktar som bokmerker
Index: kaddressbook/xxport/ldif_xxport.desktop
===================================================================
--- kaddressbook/xxport/ldif_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/ldif_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -66,7 +66,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចូល និង នាំចេញ​ទំនាក់ទំនង​ក្នុង​ទ្រង់ទ្រាយ​ជា Netscape និង Mozilla LDIF
 Comment[lt]=Priedas, skirtas kontaktų importavimui ir eksportavimui Netscape ir Mozilla LDIFF formatu
 Comment[ms]=Plug masuk untuk import dan eksport alamat perhubungan di dalam format LDIF Netscape dan Mozilla
-Comment[nb]=Tilleggsprogram for import/eksport av kontakter i Netscape og Mozillas LDIF-format
+Comment[nb]=Programtillegg for import/eksport av kontakter i Netscape og Mozillas LDIF-format
 Comment[nds]=Moduul för't Im- un Exporteren vun Kontakten in't LDIF-Formaat vun Netscape und Mozilla
 Comment[nl]=Plugin voor het importeren en exporteren van contactpersonen in Netscape's en Mozilla's LDIF-formaat
 Comment[nn]=Programtillegg for å importera og eksportera kontaktar i Netscape og Mozilla LDIF-format
Index: kaddressbook/xxport/eudora_xxport.desktop
===================================================================
--- kaddressbook/xxport/eudora_xxport.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/xxport/eudora_xxport.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -65,7 +65,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​នាំចូល និង នាំចេញ​ទំនាក់ទំនង​របស់ Eudora
 Comment[lt]=Priedas Eudora kontaktų importui ir eksportui
 Comment[ms]=Plug masuk untuk import dan eksport alamat perhubungan Eudora
-Comment[nb]=Tilleggsprogram for import/eksport av Eudora-kontakter
+Comment[nb]=Programtillegg for import/eksport av Eudora-kontakter
 Comment[nds]=Moduul för't Im- un Exporteren vun Eudora-Kontakten
 Comment[nl]=Plugin voor het importeren en exporteren van Eudora-contactpersonen
 Comment[nn]=Programtillegg for å importera og eksportera Eudora-kontaktar
Index: kaddressbook/features/distributionlist.desktop
===================================================================
--- kaddressbook/features/distributionlist.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/features/distributionlist.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -26,7 +26,7 @@
 Name[km]=កម្មវិធី​ជំនួយ​បញ្ជី​ចែកចាយ​របស់ KAB
 Name[lt]=KAB platinimo sąrašo priedas
 Name[ms]=Plugin Senarai Agihan KAB 
-Name[nb]=KAB tilleggsprogram for distribusjonslister
+Name[nb]=KAB programtillegg for distribusjonslister
 Name[nds]=Verdeellist-Moduul för KAdressbook
 Name[nl]=KAB Distributielijst-plugin
 Name[nn]=Programtillegg for KAB distribusjonsliste
@@ -72,7 +72,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​គ្រប់គ្រង​បញ្ជី​ចែកចាយ
 Comment[lt]=Priedas platinimo sąrašų tvarkymui
 Comment[ms]=Plug masuk untuk pengurusan senarai edaran
-Comment[nb]=Tilleggsprogram for å håndtere distribusjonslister
+Comment[nb]=Programtillegg for å håndtere distribusjonslister
 Comment[nds]=Moduul för't Plegen vun Verdeellisten
 Comment[nl]=Plugin voor het beheren van distributielijsten
 Comment[nn]=Programtillegg for handtering av distibusjonsliste
Index: kaddressbook/features/resourceselection.desktop
===================================================================
--- kaddressbook/features/resourceselection.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/features/resourceselection.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -68,7 +68,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​ដើម្បី​គ្រប់គ្រង​សៀវភៅ​អាសយដ្ឋាន
 Comment[lt]=Priedas adresų knygelių tvarkymui
 Comment[ms]=Plugin untuk pengurusan buku alamat
-Comment[nb]=Tilleggsprogram for å håndtere adressebøker
+Comment[nb]=Programtillegg for å håndtere adressebøker
 Comment[nds]=Moduul för't Plegen vun Adressböker
 Comment[nl]=Plugin voor het beheren van adresboeken
 Comment[nn]=Programtillegg for handtering av adressebøker
Index: kaddressbook/features/resourceselection.cpp
===================================================================
--- kaddressbook/features/resourceselection.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kaddressbook/features/resourceselection.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -191,6 +191,7 @@
   }
 
   resource->setResourceName( i18n( "%1 address book" ).arg( type ) );
+  resource->setAddressBook(core()->addressBook());
 
   KRES::ConfigDialog dlg( this, QString( "contact" ), resource );
 
Index: kitchensync/multisynk/konnectorpair.cpp
===================================================================
--- kitchensync/multisynk/konnectorpair.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kitchensync/multisynk/konnectorpair.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -29,6 +29,7 @@
   : mManager( new KonnectorManager ), mConfig( 0 )
 {
   mUid = KApplication::randomString( 10 );
+  mStrategy = ResolveManually;
 }
 
 KonnectorPair::~KonnectorPair()
Index: libkcal/resourcecachedconfig.cpp
===================================================================
--- libkcal/resourcecachedconfig.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ libkcal/resourcecachedconfig.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -52,7 +52,7 @@
            SLOT( slotIntervalStateChanged( int ) ) );
   QHBox *intervalBox = new QHBox( mGroup );
   new QLabel( i18n("Interval in minutes"), intervalBox );
-  mIntervalSpin = new QSpinBox( intervalBox );
+  mIntervalSpin = new QSpinBox( 1,900, 1,intervalBox );
   mIntervalSpin->setEnabled( false );
 }
 
@@ -92,7 +92,7 @@
            SLOT( slotIntervalStateChanged( int ) ) );
   QHBox *intervalBox = new QHBox( mGroup );
   new QLabel( i18n("Interval in minutes"), intervalBox );
-  mIntervalSpin = new QSpinBox( intervalBox );
+  mIntervalSpin = new QSpinBox( 1,900, 1,intervalBox );
   mIntervalSpin->setEnabled( false );
 
   new QRadioButton( i18n("Delayed after changes"), mGroup );
Index: libkcal/libical/zoneinfo/America/Argentina/Jujuy.ics
===================================================================
--- libkcal/libical/zoneinfo/America/Argentina/Jujuy.ics	(.../tags/KDE/3.5.4/kdepim)	(revision 0)
+++ libkcal/libical/zoneinfo/America/Argentina/Jujuy.ics	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -0,0 +1,14 @@
+BEGIN:VCALENDAR
+PRODID:-//Ximian//NONSGML Evolution Olson-VTIMEZONE Converter//EN
+VERSION:2.0
+BEGIN:VTIMEZONE
+TZID:/softwarestudio.org/Olson_20011030_5/America/Argentina/Jujuy
+X-LIC-LOCATION:America/Argentina/Jujuy
+BEGIN:STANDARD
+TZOFFSETFROM:-0300
+TZOFFSETTO:-0300
+TZNAME:ART
+DTSTART:19700101T000000
+END:STANDARD
+END:VTIMEZONE
+END:VCALENDAR
Index: libkcal/libical/zoneinfo/America/Argentina/Cordoba.ics
===================================================================
--- libkcal/libical/zoneinfo/America/Argentina/Cordoba.ics	(.../tags/KDE/3.5.4/kdepim)	(revision 0)
+++ libkcal/libical/zoneinfo/America/Argentina/Cordoba.ics	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -0,0 +1,14 @@
+BEGIN:VCALENDAR
+PRODID:-//Ximian//NONSGML Evolution Olson-VTIMEZONE Converter//EN
+VERSION:2.0
+BEGIN:VTIMEZONE
+TZID:/softwarestudio.org/Olson_20011030_5/America/Argentina/Cordoba
+X-LIC-LOCATION:America/Argentina/Cordoba
+BEGIN:STANDARD
+TZOFFSETFROM:-0300
+TZOFFSETTO:-0300
+TZNAME:ART
+DTSTART:19700101T000000
+END:STANDARD
+END:VTIMEZONE
+END:VCALENDAR
Index: libkcal/libical/zoneinfo/America/Argentina/Catamarca.ics
===================================================================
--- libkcal/libical/zoneinfo/America/Argentina/Catamarca.ics	(.../tags/KDE/3.5.4/kdepim)	(revision 0)
+++ libkcal/libical/zoneinfo/America/Argentina/Catamarca.ics	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -0,0 +1,14 @@
+BEGIN:VCALENDAR
+PRODID:-//Ximian//NONSGML Evolution Olson-VTIMEZONE Converter//EN
+VERSION:2.0
+BEGIN:VTIMEZONE
+TZID:/softwarestudio.org/Olson_20011030_5/America/Argentina/Catamarca
+X-LIC-LOCATION:America/Argentina/Catamarca
+BEGIN:STANDARD
+TZOFFSETFROM:-0300
+TZOFFSETTO:-0300
+TZNAME:ART
+DTSTART:19700101T000000
+END:STANDARD
+END:VTIMEZONE
+END:VCALENDAR
Index: libkcal/libical/zoneinfo/America/Argentina/Mendoza.ics
===================================================================
--- libkcal/libical/zoneinfo/America/Argentina/Mendoza.ics	(.../tags/KDE/3.5.4/kdepim)	(revision 0)
+++ libkcal/libical/zoneinfo/America/Argentina/Mendoza.ics	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -0,0 +1,14 @@
+BEGIN:VCALENDAR
+PRODID:-//Ximian//NONSGML Evolution Olson-VTIMEZONE Converter//EN
+VERSION:2.0
+BEGIN:VTIMEZONE
+TZID:/softwarestudio.org/Olson_20011030_5/America/Argentina/Mendoza
+X-LIC-LOCATION:America/Argentina/Mendoza
+BEGIN:STANDARD
+TZOFFSETFROM:-0300
+TZOFFSETTO:-0300
+TZNAME:ART
+DTSTART:19700101T000000
+END:STANDARD
+END:VTIMEZONE
+END:VCALENDAR
Index: libkcal/libical/zoneinfo/America/Argentina/Buenos_Aires.ics
===================================================================
--- libkcal/libical/zoneinfo/America/Argentina/Buenos_Aires.ics	(.../tags/KDE/3.5.4/kdepim)	(revision 0)
+++ libkcal/libical/zoneinfo/America/Argentina/Buenos_Aires.ics	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -0,0 +1,14 @@
+BEGIN:VCALENDAR
+PRODID:-//Ximian//NONSGML Evolution Olson-VTIMEZONE Converter//EN
+VERSION:2.0
+BEGIN:VTIMEZONE
+TZID:/softwarestudio.org/Olson_20011030_5/America/Argentina/Buenos_Aires
+X-LIC-LOCATION:America/Argentina/Buenos_Aires
+BEGIN:STANDARD
+TZOFFSETFROM:-0300
+TZOFFSETTO:-0300
+TZNAME:ART
+DTSTART:19700101T000000
+END:STANDARD
+END:VTIMEZONE
+END:VCALENDAR
Index: libkcal/libical/zoneinfo/zones.tab
===================================================================
--- libkcal/libical/zoneinfo/zones.tab	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ libkcal/libical/zoneinfo/zones.tab	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -55,6 +55,11 @@
 +0181200 -0630400 America/Anguilla
 +0170300 -0614800 America/Antigua
 -0071200 -0481200 America/Araguaina
+-0343600 -0582700 America/Argentina/Buenos_Aires
+-0282800 -0654700 America/Argentina/Catamarca
+-0312400 -0641100 America/Argentina/Cordoba
+-0241100 -0651800 America/Argentina/Jujuy
+-0325300 -0684900 America/Argentina/Mendoza
 +0123000 -0685800 America/Aruba
 -0251600 -0574000 America/Asuncion
 +0130600 -0593700 America/Barbados
@@ -220,19 +225,17 @@
 +0013300 +1102000 Asia/Kuching
 +0292000 +0475900 Asia/Kuwait
 +0221400 +1133500 Asia/Macao
++0221400 +1133500 Asia/Macau
 +0593400 +1504800 Asia/Magadan
--0050700 +1192400 Asia/Makassar
 +0143500 +1210000 Asia/Manila
 +0233600 +0583500 Asia/Muscat
 +0351000 +0332200 Asia/Nicosia
 +0550200 +0825500 Asia/Novosibirsk
 +0550000 +0732400 Asia/Omsk
-+0511300 +0512100 Asia/Oral
 +0113300 +1045500 Asia/Phnom_Penh
 +0000200 +1092000 Asia/Pontianak
 +0390100 +1254500 Asia/Pyongyang
 +0251700 +0513200 Asia/Qatar
-+0444800 +0652800 Asia/Qyzylorda
 +0164700 +0961000 Asia/Rangoon
 +0243800 +0464300 Asia/Riyadh
 +0104500 +1064000 Asia/Saigon
Index: libkcal/libical/zoneinfo/Asia/Macau.ics
===================================================================
--- libkcal/libical/zoneinfo/Asia/Macau.ics	(.../tags/KDE/3.5.4/kdepim)	(revision 0)
+++ libkcal/libical/zoneinfo/Asia/Macau.ics	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -0,0 +1,14 @@
+BEGIN:VCALENDAR
+PRODID:-//Ximian//NONSGML Evolution Olson-VTIMEZONE Converter//EN
+VERSION:2.0
+BEGIN:VTIMEZONE
+TZID:/softwarestudio.org/Olson_20011030_5/Asia/Macau
+X-LIC-LOCATION:Asia/Macau
+BEGIN:STANDARD
+TZOFFSETFROM:+0800
+TZOFFSETTO:+0800
+TZNAME:CST
+DTSTART:19700101T000000
+END:STANDARD
+END:VTIMEZONE
+END:VCALENDAR
Index: kpilot/conduits/vcalconduit/todo-conduit.desktop
===================================================================
--- kpilot/conduits/vcalconduit/todo-conduit.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kpilot/conduits/vcalconduit/todo-conduit.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -69,7 +69,7 @@
 Name[nds]=Opgaven (KOrganizer)
 Name[nl]=Taken (KOrganizer)
 Name[nn]=Oppgåveliste (KOrganizer)
-Name[pl]=Do zrobienia (Organizator)
+Name[pl]=Do zrobienia (Korganizer)
 Name[pt]=Por Fazer (KOrganizer)
 Name[pt_BR]=Tarefas (KOrganizer)
 Name[ru]=Задачи (KOrganizer)
Index: kpilot/conduits/vcalconduit/vcal-conduit.desktop
===================================================================
--- kpilot/conduits/vcalconduit/vcal-conduit.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kpilot/conduits/vcalconduit/vcal-conduit.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -29,7 +29,7 @@
 Name[nds]=Kalenner (KOrganizer)
 Name[nl]=Agenda (KOrganizer)
 Name[nn]=Kalender (KOrganizer)
-Name[pl]=Kalendarz (Organizator)
+Name[pl]=Kalendarz (Korganizer)
 Name[pt]=Calendário (KOrganizer)
 Name[pt_BR]=Calendário (KOrganizer)
 Name[ru]=Календарь (KOrganizer)
Index: kontact/plugins/newsticker/kcmkontactknt.cpp
===================================================================
--- kontact/plugins/newsticker/kcmkontactknt.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ kontact/plugins/newsticker/kcmkontactknt.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -79,7 +79,7 @@
 
   mTitle->setText( title );
   mURL->setText( url );
-
+  mTitle->setFocus();
   connect( mTitle, SIGNAL( textChanged( const QString& ) ),
            this, SLOT( modified() ) );
   connect( mURL, SIGNAL( textChanged( const QString& ) ),
Index: korganizer/kogroupwareprefspage.ui
===================================================================
--- korganizer/kogroupwareprefspage.ui	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/kogroupwareprefspage.ui	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -84,6 +84,9 @@
                                 <property name="name">
                                     <cstring>publishDelay</cstring>
                                 </property>
+                                <property name="minValue">
+                                    <number>1</number>
+                                </property>
                                 <property name="whatsThis" stdset="0">
                                     <string>Configure minimum the interval of time in minutes between each upload here. This configuration is only effective in case you choose to publish your information automatically.</string>
                                 </property>
Index: korganizer/plugins/projectview/projectview.desktop
===================================================================
--- korganizer/plugins/projectview/projectview.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/plugins/projectview/projectview.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -77,7 +77,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​នេះ​ផ្ដល់​នូវ​ទិដ្ឋភាព​រៀបចំ​គម្រោង​សម្រាប់ korganizer (ដូចជា ទិដ្ឋភាព​ការងារ​ត្រូវ​ធ្វើ ឬ ទិដ្ឋភាព​ខែ​ជាដើម) ។ បើ​អ្នក​ធ្វើ​ឲ្យ​កម្មវិធី​ជំនួយ​នេះ​ប្រើ​បាន អ្នក​នឹង​អាច​ប្ដូរ​ទិដ្ឋភាព​គម្រោង ហើយ​មើល​ព្រឹត្តិការណ៍​របស់​អ្នក​ដូចជា​នៅ​ក្នុងកម្មវិធី​រៀបចំ​គម្រោង​អញ្ចឹង​ដែរ ។
 Comment[lt]=Šis priedas pateikia KOrganizaer projekto planavimo vaizdą (panašiai kaip darbų ar mėnesio vaizdai). Jei įjungsite šį įskiepį, galėsite peršokti į projekto vaizdą ir peržiūrėti savo darbų sąrašą kaip projektą.
 Comment[ms]=Plugin ini menyediakan paparan perancangan projek untuk KOrganizer (seperti tugasan atau paparan bulan). Jika plugin ini diaktifkan, anda boleh beralih ke paparan projek dan paparkan senarai tugasan seperti dalam perancang projek. 
-Comment[nb]=Dette tilleggsprogrammet lager en prosjektplan-visning for KOrganizer (som i gjørelister og månedsvisninger). Slår du på dette tilleggsprogrammet kan du bytte til prosjektvisning og se på gjørelista som i en prosjektplan.
+Comment[nb]=Dette programtilleggmet lager en prosjektplan-visning for KOrganizer (som i gjørelister og månedsvisninger). Slår du på dette programtilleggmet kan du bytte til prosjektvisning og se på gjørelista som i en prosjektplan.
 Comment[nds]=Mit dit Moduul kannst Du in KOrganizer en Projektplaan-Ansicht opropen (jüst as de Opgaven- oder Maandansichten). Wenn Du dit Moduul aktiveerst, kannst Du na de Projektansicht wesseln un Dien Opgavenlist as mit en Projektplaner ankieken.
 Comment[nl]=Deze plugin biedt een projectweergave voor KOrganizer (net zoals een takenlijst of maandweergave). Wanneer u deze plugin aanzet, kunt u deze weergave aanzetten om uw takenlijst te bekijken als een projectplanner.
 Comment[nn]=Dette programtillegget lagar ei prosjektplanvising for KOrganizer (slik som oppgåve- eller månadsvisingane). Dersom du brukar dette programtillegget, kan du visa oppgåvelista som i ein prosjektplanleggjar.
Index: korganizer/plugins/printing/whatsnext/whatsnextprint.desktop
===================================================================
--- korganizer/plugins/printing/whatsnext/whatsnextprint.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/plugins/printing/whatsnext/whatsnextprint.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -56,7 +56,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​នេះ​អនុញ្ញាត​ឲ្យ​អ្នក​បោះពុម្ព​បញ្ជី​នៃ​ព្រឹត្តិការណ៍ និង ការងារ​ត្រូវ​ធ្វើ​បន្តបន្ទាប់
 Comment[lt]=Šis priedas leidžia spausdinti sąrašą visų artėjančių įvykių ir darbų.
 Comment[ms]=Plugin ini membenarkan anda mencetak senarai semua peristiwa dan tugasan mendatang. 
-Comment[nb]=Med dette tilleggsprogrammet kan du skrive ut en liste over kommende hendelser og ting som skal gjøres.
+Comment[nb]=Med dette programtilleggmet kan du skrive ut en liste over kommende hendelser og ting som skal gjøres.
 Comment[nds]=Mit dit Moduul kann en List vun de direktemang nakamen Begeefnissen un Opgaven utdruckt warrn.
 Comment[nl]=Deze plugin maakt het mogelijk om een lijst van alle komende evenementen en taken uit te printen.
 Comment[nn]=Dette programtillegget lèt deg skriva ut ei liste over alle dei komande hendingane og oppføringane i hugselista.
Index: korganizer/plugins/printing/journal/journalprint.desktop
===================================================================
--- korganizer/plugins/printing/journal/journalprint.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/plugins/printing/journal/journalprint.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -56,7 +56,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​នេះ​អនុញ្ញាត​ឲ្យ​អ្នក​បោះពុម្ព​ធាតុ​ទិនានុប្បវត្តិ (ធាតុ​កំណត់ហេតុ​ប្រចាំ​ថ្ងៃ) ។
 Comment[lt]=Šis priedas leidžia spausdinti dienyno įrašus.
 Comment[ms]=Plugin ini membenarkan anda mencetak entri jurnal  (entri diari).
-Comment[nb]=Med dette tilleggsprogrammet kan du skrive ut oppføringer i dagboka.
+Comment[nb]=Med dette programtilleggmet kan du skrive ut oppføringer i dagboka.
 Comment[nds]=Mit dit Moduul köönt Daagbook-Indrääg utdruckt warrn.
 Comment[nl]=Deze plugin maakt het mogelijk om journalen uit te printen (dagboekitems).
 Comment[nn]=Dette programtillegget lèt deg skriva ut dagboksoppføringar.
Index: korganizer/plugins/printing/list/listprint.desktop
===================================================================
--- korganizer/plugins/printing/list/listprint.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/plugins/printing/list/listprint.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -56,7 +56,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​នេះ​អនុញ្ញាត​ឲ្យ​អ្នក​បោះពុម្ព​ព្រឹត្តិការណ៍ និង ការងារ​ត្រូវ​ធ្វើក្នុង​ទម្រង់​ជា​បញ្ជី ។
 Comment[lt]=Šis priedas leidžia spausdinti įvykius ir darbus sąrašo forma.
 Comment[ms]=Plugin ini membenarkan anda mencetak peristiwa dan tugasan dalam borang senarai.
-Comment[nb]=Med dette tilleggsprogrammet kan du skrive ut hendelser og gjørelister på liste-form.
+Comment[nb]=Med dette programtilleggmet kan du skrive ut hendelser og gjørelister på liste-form.
 Comment[nds]=Mit dit Moduul köönt Begeefnissen un Opgaven as Listen utdruckt warrn.
 Comment[nl]=Deze plugin maakt het mogelijk om evenementen en taken in lijstvorm uit te printen.
 Comment[nn]=Dette programtillegget lèt deg skriva ut hendingar og hugselister på listeform.
Index: korganizer/plugins/timespanview/timespanview.desktop
===================================================================
--- korganizer/plugins/timespanview/timespanview.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/plugins/timespanview/timespanview.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -56,7 +56,7 @@
 Comment[km]=កម្មវិធី​ជំនួយ​នេះ​ផ្ដល់​នូវ​ទិដ្ឋភាព​រយៈពេល​សម្រាប់ korganizer (ដូចជា ទិដ្ឋភាព​ការងារ​ត្រូវ​ធ្វើ ឬ ទិដ្ឋភាព​ខែ​ជាដើម) ។ បើ​អ្នក​ធ្វើ​ឲ្យ​កម្មវិធី​ជំនួយ​នេះ​ប្រើ​បាន អ្នក​នឹង​អាច​ប្ដូរ​ទិដ្ឋភាព​រយៈពេល ហើយ​មើល​ព្រឹត្តិការណ៍​របស់​អ្នក​ដូចជា​នៅ​ក្នុងដ្យាក្រាម Gantt អញ្ចឹង​ដែរ ។
 Comment[lt]=Šis priedas korganizer programoje sudaro galimybę apžvelgti laiko tarpą (pvz., darbų arba mėnesio peržiūra). Įgalinus šį priedą galėsite persijungti į laiko tarpo peržiūrą ir žiūrėti įvykius tarsi Gantt diagramoje.
 Comment[ms]=Plugin ini menyediakan paparan kitar waktu  untuk korganizer (seperti tugasan atau paparan bulan). Jika plugin ini diaktifkan,anda boleh beralih ke paparan kitar masa dan paparkan peristiwa seperti dalam diagram Gantt. 
-Comment[nb]=Dette tilleggsprogrammet lager en periodevisning for korganizer (slik som gjøreliste og månedsvisning). Hvis du slår på dette tilleggsprogrammet kan du bytte til periodevisning og se hendelser som i et Gantt-diagram.
+Comment[nb]=Dette programtilleggmet lager en periodevisning for korganizer (slik som gjøreliste og månedsvisning). Hvis du slår på dette programtilleggmet kan du bytte til periodevisning og se hendelser som i et Gantt-diagram.
 Comment[nds]=Mit dit Moduul kannst Du in KOrganizer en Tietbruuk-Ansicht opropen (jüst as de Opgaven- oder Maandansichten). Wenn Du dit Moduul aktiveerst, kannst Du na de Tietbruukansicht wesseln un Dien Begeefnissen as Ganntt-Diagramm ankieken.
 Comment[nl]=Deze plugin biedt een weergave voor KOrganizer (zoals de takenlijst en de maandweergave). Wanneer u deze plugin inschakelt kunt u kiezen voor een weergave waarin u uw evenementen in een Gantt-diagram kunt bekijken.
 Comment[pl]=Ta wtyczka tworzy widok zakresu czasu w KOrganizerze (tak jak widok zadań do zrobienia lub widok miesiąca). Po włączeniu tej wtyczki możliwe jest przełączenie się na widok zakresu czasu i przeglądanie zdarzeń jak na diagramie Gantta.
Index: korganizer/plugins/hebrew/configdialog.cpp
===================================================================
--- korganizer/plugins/hebrew/configdialog.cpp	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/plugins/hebrew/configdialog.cpp	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -26,7 +26,7 @@
 #include <kstandarddirs.h>
 #include <ksimpleconfig.h>
 
-ConfigDialog::ConfigDialog(QWidget * parent):KDialogBase(Plain, i18n("Configure Holidays"), Ok, Ok,
+ConfigDialog::ConfigDialog(QWidget * parent):KDialogBase(Plain, i18n("Configure Holidays"), Ok|Cancel, Ok,
             parent)
 {
   QFrame *topFrame = plainPage();
Index: korganizer/korganizer_configfreebusy.desktop
===================================================================
--- korganizer/korganizer_configfreebusy.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/korganizer_configfreebusy.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -39,7 +39,7 @@
 Name[nds]=Free/Bunnen
 Name[nl]=Vrij/bezet
 Name[nn]=Ledig/opptatt
-Name[pl]=Wolny/Zajęty
+Name[pl]=Wolny/zajęty
 Name[pt]=Livre/Ocupado
 Name[pt_BR]=Livre/Ocupado
 Name[ru]=Занятое время
Index: korganizer/korganizer.desktop
===================================================================
--- korganizer/korganizer.desktop	(.../tags/KDE/3.5.4/kdepim)	(revision 570934)
+++ korganizer/korganizer.desktop	(.../branches/KDE/3.5/kdepim)	(revision 570934)
@@ -35,7 +35,7 @@
 Comment[nb]=Et kalender- og tidsplanleggingsprogram
 Comment[nds]=Kalenner un Tietplaner
 Comment[nl]=Agenda- en afsprakenprogramma
-Comment[nn]=Kalender- og planleggjingsprogram
+Comment[nn]=Kalender- og planleggingsprogram
 Comment[nso]=Lenaneo la Peakanyo ya Tshupamabaka
 Comment[pl]=Kalendarz i terminarz
 Comment[pt]=Calendário e Programa de Escalonamento

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


