--- kdepim-3.4.2/knode/knnntpclient.cpp.old	2005-07-20 12:03:03.000000000 +0200
+++ kdepim-3.4.2/knode/knnntpclient.cpp	2005-09-04 02:35:04.000000000 +0200
@@ -478,7 +478,7 @@
     if ((code == 430) || (code == 423))  // 430 no such article found || 423 no such article number in this group
       job->setErrorString(
              errorPrefix + getCurrentLine() +
-             i18n("<br><br>The article you requested is not available on your news server;<br>you could try to get it from <a href=\"http://groups.google.com/groups?q=msgid:%1&ic=1\">groups.google.com</a>.")
+             i18n("<br><br>The article you requested is not available on your news server;<br>you could try to get it from <a href=\"http://groups.google.pl/groups?as_msgid=&selm=%1\">groups.google.pl</a>.")
                   .arg(target->messageID()->as7BitString(false)));
     return;
   }
