Index: Makefile.am
===================================================================
RCS file: /home/kde/kdepim/libkdepim/Makefile.am,v
retrieving revision 1.39
diff -u -5 -d -p -r1.39 Makefile.am
--- Makefile.am	11 Apr 2004 18:11:29 -0000	1.39
+++ Makefile.am	4 May 2004 23:22:50 -0000
@@ -27,11 +27,11 @@ MailTransportServiceIface_DCOPIDLNG = tr
 MailTransportServiceIface_DIR = $(srcdir)/interfaces
 
 AddressBookServiceIface_DIR = $(srcdir)/interfaces
 
 libkdepim_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 1:0:0
-libkdepim_la_LIBADD  = $(LIB_KIO) $(LIB_KABC) $(LIB_KDEUI)
+libkdepim_la_LIBADD  = $(LIB_KIO) $(LIB_KABC) $(LIB_KDEUI) -lkimproxy
 #               $top_builddir)/libkdepim/resources/libkpimresources.la
 
 METASOURCES = AUTO
 
 messages: rc.cpp
Index: addresseeview.cpp
===================================================================
RCS file: /home/kde/kdepim/libkdepim/addresseeview.cpp,v
retrieving revision 1.33
diff -u -5 -d -p -r1.33 addresseeview.cpp
--- addresseeview.cpp	3 May 2004 20:49:24 -0000	1.33
+++ addresseeview.cpp	4 May 2004 23:22:50 -0000
@@ -39,18 +39,20 @@
 #include <kmessagebox.h>
 #include <krun.h>
 #include <kstringhandler.h>
 #include <ktempfile.h>
 
+#include <kdebug.h>
+
 #include "addresseeview.h"
 
 using namespace KPIM;
 
 AddresseeView::AddresseeView( QWidget *parent, const char *name,
                               KConfig *config )
   : KTextBrowser( parent, name ), mDefaultConfig( false ), mImageJob( 0 ),
-    mLinkMask( AddressLinks | EmailLinks | PhoneLinks | URLLinks )
+    mLinkMask( AddressLinks | EmailLinks | PhoneLinks | URLLinks | IMLinks )
 {
   setWrapPolicy( QTextEdit::AtWordBoundary );
   setLinkUnderline( false );
   setVScrollBarMode( QScrollView::AlwaysOff );
   setHScrollBarMode( QScrollView::AlwaysOff );
@@ -85,10 +87,13 @@ AddresseeView::AddresseeView( QWidget *p
     mDefaultConfig = true;
   } else
     mConfig = config;
 
   load();
+  
+  mKIMProxy = new ::KIMProxy( kapp->dcopClient() );
+  //connect( mKIMProxy, SIGNAL( sigContactStatusChanged( QString ) ), this, SLOT( statusChanged( ) ) );
 }
 
 AddresseeView::~AddresseeView()
 {
   if ( mDefaultConfig )
@@ -98,10 +103,12 @@ AddresseeView::~AddresseeView()
   delete mActionShowBirthday;
   delete mActionShowAddresses;
   delete mActionShowEmails;
   delete mActionShowPhones;
   delete mActionShowURLs;
+  
+  delete mKIMProxy;
 }
 
 void AddresseeView::setAddressee( const KABC::Addressee& addr )
 {
   mAddressee = addr;
@@ -119,14 +126,15 @@ void AddresseeView::setAddressee( const 
 void AddresseeView::enableLinks( int linkMask )
 {
   mLinkMask = linkMask;
 }
 
-QString AddresseeView::vCardAsHTML( const KABC::Addressee& addr, int linkMask,
+QString AddresseeView::vCardAsHTML( const KABC::Addressee& addr, ::KIMProxy *proxy, int linkMask,
                                     bool internalLoading,
                                     bool showBirthday, bool showAddresses,
-                                    bool showEmails, bool showPhones, bool showURLs )
+                                    bool showEmails, bool showPhones, bool showURLs,
+                                    bool showIMAddresses )
 {
   QString name = ( addr.formattedName().isEmpty() ?
                    addr.assembledName() : addr.formattedName() );
 
   QString dynamicPart;
@@ -294,11 +302,48 @@ QString AddresseeView::vCardAsHTML( cons
   if ( !addr.organization().isEmpty() && name.isEmpty() ||
        addr.formattedName() == addr.organization() ) {
     name = addr.organization();
     organization = QString::null;
   }
+ 
+  if ( showIMAddresses )
+  {
+    // fill in IM status
+    // check the addressee has IM presence
+    QStringList imUids = proxy->imAddresseeUids();
+    QStringList::Iterator it = qFind( imUids.begin(), imUids.end(), addr.uid() );
+    
+    // 
+    if ( it != imUids.end() )
+    {
+      // set image source to either a QMimeSourceFactory key or a data:/ URL
+      QString imgSrc;
+      if ( internalLoading )
+      {
+        imgSrc = QString::fromLatin1( "im_status_%1_image").arg( addr.uid() );
+        QMimeSourceFactory::defaultFactory()->setPixmap( imgSrc, proxy->statusIcon( addr.uid() ) );
+      }
+      else
+        imgSrc = pixmapAsDataUrl( proxy->statusIcon( addr.uid() ) );
 
+      // make the status a link, if required
+      QString imStatus;
+      if ( linkMask & IMLinks )
+        imStatus = QString::fromLatin1( "<a href=\"im:\"><img src=\"%1\"> (%2)</a>" );
+      else
+        imStatus = QString::fromLatin1( "<img src=\"%1\"> (%2)" );
+      
+      // append our status to the rest of the dynamic part of the addressee
+      dynamicPart += rowFmtStr
+              .arg( i18n( "IM Status" ) )
+              .arg( imStatus
+                        .arg( imgSrc )
+                        .arg( proxy->statusString( addr.uid() ) )
+                  );
+    }
+  }
+   
   QString strAddr = QString::fromLatin1(
     "<div>"
     "<table width=\"100%\">"
     "<tr>"
     "<td align=\"right\" valign=\"top\" width=\"30%\" rowspan=\"3\">"
@@ -350,11 +395,11 @@ void AddresseeView::updateView()
     mImageJob = 0;
 
     mImageData.truncate( 0 );
   }
 
-  QString strAddr = vCardAsHTML( mAddressee, mLinkMask, true,
+  QString strAddr = vCardAsHTML( mAddressee, mKIMProxy, mLinkMask, true,
                                  mActionShowBirthday->isChecked(),
                                  mActionShowAddresses->isChecked(),
                                  mActionShowEmails->isChecked(),
                                  mActionShowPhones->isChecked(),
                                  mActionShowURLs->isChecked() );
@@ -388,11 +433,11 @@ void AddresseeView::updateView()
     } else {
       QMimeSourceFactory::defaultFactory()->setPixmap( imageURL,
         KGlobal::iconLoader()->loadIcon( "personal", KIcon::Desktop, 128 ) );
     }
   }
-
+  
   // at last display it...
   setText( strAddr );
 }
 
 KABC::Addressee AddresseeView::addressee() const
@@ -438,10 +483,15 @@ void AddresseeView::faxNumberClicked( co
 
   commandLine.replace( "%N", number );
   KRun::runCommand( commandLine );
 }
 
+void AddresseeView::imAddressClicked()
+{
+  mKIMProxy->chatContactById( mAddressee.uid() );
+}
+
 QPopupMenu *AddresseeView::createPopupMenu( const QPoint& )
 {
   QPopupMenu *menu = new QPopupMenu( this );
   mActionShowBirthday->plug( menu );
   mActionShowAddresses->plug( menu );
@@ -463,10 +513,12 @@ void AddresseeView::slotUrlClicked( cons
     phoneNumberClicked( strippedNumber( url.mid( 8 ) ) );
   else if ( url.startsWith( "fax:" ) )
     faxNumberClicked( strippedNumber( url.mid( 6 ) ) );
   else if ( url.startsWith( "addr:" ) )
     emit addressClicked( url.mid( 7 ) );
+  else if ( url.startsWith( "im:" ) )
+    imAddressClicked();
   else
     urlClicked( url );
 }
 
 void AddresseeView::slotHighlighted( const QString &link )
@@ -489,10 +541,13 @@ void AddresseeView::slotHighlighted( con
   } else if ( link.startsWith( "addr:" ) ) {
     emit highlightedMessage( i18n( "Show address on map" ) );
   } else if ( link.startsWith( "http:" ) ) {
     emit urlHighlighted( link );
     emit highlightedMessage( i18n( "Open URL %1" ).arg( link ) );
+  } else if ( link.startsWith( "im:" ) ) {
+    emit highlightedMessage( i18n( "Chat with %1" ).arg( ( mAddressee.formattedName().isEmpty() ?
+                   mAddressee.assembledName() : mAddressee.formattedName() ) ) );
   } else
     emit highlightedMessage( "" );
 }
 
 void AddresseeView::configChanged()
Index: addresseeview.h
===================================================================
RCS file: /home/kde/kdepim/libkdepim/addresseeview.h,v
retrieving revision 1.11
diff -u -5 -d -p -r1.11 addresseeview.h
--- addresseeview.h	3 May 2004 20:49:24 -0000	1.11
+++ addresseeview.h	4 May 2004 23:22:51 -0000
@@ -24,24 +24,26 @@
 
 #include <qcstring.h>
 
 #include <kabc/addressee.h>
 #include <ktextbrowser.h>
+#include <kimproxy.h>
 
 namespace KIO {
 class Job;
 }
 class KToggleAction;
 
 class QPopupMenu;
 
+
 namespace KPIM {
 
+
 class AddresseeView : public KTextBrowser
 {
   Q_OBJECT
-
   public:
     /**
       Constructor.
  
       @param config The config object where the settings are stored
@@ -67,11 +69,11 @@ class AddresseeView : public KTextBrowse
 
     /**
       This enums are used by enableLinks to set which kind of links shall
       be enabled.
      */
-    enum LinkMask { AddressLinks = 1, EmailLinks = 2, PhoneLinks = 4, URLLinks = 8 };
+    enum LinkMask { AddressLinks = 1, EmailLinks = 2, PhoneLinks = 4, URLLinks = 8, IMLinks = 16 };
 
     /**
       Sets which parts of the contact shall be presented as links.
       The mask can be OR'ed LinkMask. By default all links are enabled.
      */
@@ -92,22 +94,23 @@ class AddresseeView : public KTextBrowse
                       The links looks like this:
                         "addr://<addr id>" for addresses
                         "mailto:<email address>" for emails
                         "phone://<phone number>" for phone numbers
                         "http://<url>" for urls
+                        "im:<im addrss>" for instant messaging addresses
       @param internalLoading If true, the loading of internal pictures is done automatically.
       @param showBirthday If true, the birthday is included.
       @param showAddresses If true, the addresses are included.
       @param showEmails If true, the emails are included.
       @param showPhones If true, the phone numbers are included.
       @param showURLs If true, the urls are included.
      */
-    static QString vCardAsHTML( const KABC::Addressee& addr, int linkMask = true,
+    static QString vCardAsHTML( const KABC::Addressee& addr, ::KIMProxy *proxy, int linkMask = true,
                                 bool internalLoading = true,
                                 bool showBirthday = true, bool showAddresses = true,
                                 bool showEmails = true, bool showPhones = true,
-                                bool showURLs = true );
+                                bool showURLs = true, bool showIMAddresses = true );
     /**
      * Encodes a QPixmap as a PNG into a data: URL (rfc2397), readable by the data kio protocol
      * @param pixmap the pixmap to encode
      * @return a data: URL
      */
@@ -126,11 +129,12 @@ class AddresseeView : public KTextBrowse
   protected:
     virtual void urlClicked( const QString &url );
     virtual void emailClicked( const QString &mail );
     virtual void phoneNumberClicked( const QString &number );
     virtual void faxNumberClicked( const QString &number );
-
+    virtual void imAddressClicked();
+    
     virtual QPopupMenu *createPopupMenu( const QPoint& );
 
   private slots:
     void slotMailClicked( const QString&, const QString& );
     void slotUrlClicked( const QString& );
@@ -163,10 +167,11 @@ class AddresseeView : public KTextBrowse
     KABC::Addressee mAddressee;
     int mLinkMask;
 
     class AddresseeViewPrivate;
     AddresseeViewPrivate *d;
+    ::KIMProxy *mKIMProxy;
 };
 
 }
 
 #endif
